<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
      "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type"
  content="application/xhtml+xml; charset=UTF-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=9" />
  <title>Analysis Power Tools: MANUAL: apt-copynumber-axiom-safer
  (2.10.2.1)</title>
  <link href="tabs.css" rel="stylesheet" type="text/css" />
  <link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>

<body>

<div id="top">
<!-- do not remove this div, it is closed by doxygen! -->
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->

<div class="navigation" id="top">

<div class="tabs">
<ul class="tablist">
  <li><a href="index.html"><span>Main Page</span></a></li>
  <li><a href="CHANGE.html"><span>Change Log</span></a></li>
  <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
  <li><a href="FAQ.html"><span>FAQs</span></a></li>
  <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
  <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
</ul>
</div>
</div>
</div>
<!-- top -->

<div class="header">

<div class="headertitle">
<h1>Analysis Power Tools: MANUAL: apt-copynumber-axiom-safer (2.10.2.1) </h1>
</div>
</div>
<!--header-->

<div class="contents">

<div class="textblock">
<h2>Contents </h2>
<ul>
  <li><a href="#intro">Introduction.</a></li>
  <li><a href="#quickstart">Quick Start</a></li>
  <li><a href="#options">Options</a></li>
</ul>

<p><a class="anchor" id="intro"></a> </p>

<h2>Introduction </h2>

<p><b>apt-copynumber-axiom-safer</b> is a program for detecting copy number
variations (calculating copy number states) in specified genes.</p>

<p><a class="anchor" id="quickstart"></a> </p>

<h2>Quick Start </h2>

<p>The basic requirements for a run of apt-copynumber-axiom-safer are:</p>
<ul>
  <li>The .summary.a5 (or .summary.txt) and .report.txt files from
    apt-copynumber-axiom-safer. Note that for a large number of samples the
    program will run much faster using the summary.a5 file.</li>
  <li>The .cn_models reference file.</li>
  <li>The .cnbins.txt regions file.</li>
  <li>The .controls.txt plate controls file.</li>
</ul>

<p>Additional parameters may be specified on the command line or by means of a
parameter XML file.</p>

<p>The example command below runs the program:</p>
<pre class="fragment">  apt-copynumber-axiom-safer \
      --analysis-files-path /library/file/path \
      --arg-file AxiomCN.xml \
      --out-dir out \
      --log-file ./apt-copynumber-axiom-safer.log</pre>

<p>Alternatively, parameters can be set using command line parameters
directly:</p>
<pre class="fragment">/nfs/swbuildtmp/devel/jbiela/Axiom/release_build/output/x86_64-intel-linux/bin/apt-copynumber-axiom-safer \
      --analysis-files-path /library/file/path \
      --reference-file PharmacoScan.20160620.cn_models \
      --summary-file /path/to/AxiomGT1.summary.a5 \
      --cn-regions-file /path/to/PharmacoScan.9.cnbins.txt \
      --report-file /path/to/AxiomGT1.report.txt \
      --controls-file /path/to/Axiom_ADME.v1.controls.txt \
      --waviness-sd-max 0.5 \
      --mapd-max 1.425 \
      --analysis-name AxiomGT1 \
      --out-dir /path/to/output_directory \
      --log-file /path/to/log_file.txt
      --penn-cnv-output true \
      --penn-cnv-output-plate-corrected true</pre>

<p><a class="anchor" id="options"></a> </p>

<h2>Options: </h2>
<pre class="fragment">   'apt-copynumber-axiom-safer'
   Version: 2.10.2.1 stash.amer.thermo.com/ma-alg/apt2-genotyping.git/release/APT_2.10.2.1@4574d9db5d(553ea994bc) 2018-11-09-19:37
   
USER HELP ----------

  --version (flag) = Display the version info for this program.

  --user-help (flag) = Display help intended for end-users.
    Aliases: --help, -h

  --log-file (string8) = The name of the log file.  Defaults to 'PROGRAMNAME-NUMBER.log'.
    Aliases: --lf

  --console-add-select (string8) = Add selectors for console messages.
    e.g. to include all debug messages: --console-add-select debug
         to include all messages of level 3 or higher: --console-add-select '*:3'


  --console-add-neg-select (string8) = Add selectors to be excluded from console messages.
    e.g. to exclude all warning messages and errors summary: --console-add-neg-select WARNING,summary


  --arg-file (filename) = Read arguments from this file.
    File should be a valid XML file, and should contain elements of the form
    &lt;Parameter name="PARAMETER-NAME" analysis="NODE-NAME" currentValue="VALUE" /&gt;.

  --check-param (flag) = Stop the program after the command parameters have been checked.

  --out-dir (string8) = Output directory for result files

  --temp-dir (string8) = The output directory for temporary files

  --analysis-files-path (string8) = Default directory to search for analysis library files

  --summary-file (filename) = Summary file generated by apt-genotype-axiom (used as input)

  --reference-file (filename) = Reference file

  --analysis-name (string8) = Name to prepend to output files
    default value: 'AxiomCN'

  --chip-type (vec_string8) = Text identifier for chip.
    Aliases: --array-type

  --mapd-max (float64) = Samples with MAPD greater than this threshold will fail quality control
    default value: '0.400000'

  --waviness-sd-max (float64) = Samples with waviness standard deviation greater than this threshold will fail quality control
    default value: '0.300000'

  --penn-cnv-output-plate-corrected (bool) = Output plate-corrected log2 ratios and BAFs in Penn CNV format files
    default value: 'False'

  --use-text-format-summary-file (bool) = Use Axiom summary file in text format instead of a5
    default value: 'False'

  --write-cnv-a5 (bool) = Write the .cnv.a5 file.
    default value: 'True'

  --cn-regions-file (filename) = File that lists the regions for the CN calculation

  --report-file (filename) = Report file generated by apt-genotype-axiom (used as input)

  --controls-file (filename) = The file containing list of control sample file names used for per-plate signal correction

  --loh-batch-folder (filename) = The batch folder generated by apt-genotype-axiom (used as input). Set either the loh-batch-folder or loh-calls-file and loh-confidences-file to get LOH segments.

  --loh-calls-file (filename) = The calls file generated by apt-genotype-axiom (used as input)

  --loh-confidences-file (filename) = The confidences file generated by apt-genotype-axiom (used as input)

  --loh-samples-per-block (int32) = Number of samples per block to process from LOH batch folder or calls file. Only supports up to 200.
    default value: '200'

  --loh-error-rate (float64) = Error rate to use in LOH processing
    default value: '0.050000'

  --loh-beta (float64) = Beta to use in LOH processing
    default value: '0.001000'

  --loh-alpha (float64) = Alpha to use in LOH processing
    default value: '0.010000'

  --loh-separation (int32) = Separation to use in LOH processing
    default value: '1000000'

  --loh-min-marker-count (int32) = Minimum marker count to use in LOH processing
    default value: '10'

  --loh-no-call-threshold (float64) = No call threshold to use in LOH processing
    default value: '0.050000'

  --loh-min-genomic-span (int32) = Minimum genomic span to use in LOH processing
    default value: '1000000'

  --ap-baf-step (int32) = Step to use in calculating allele peaks
    default value: '25'
    Aliases: --baf-step

  --ap-baf-window (int32) = Window to use in calculating allele peaks
    default value: '150'
    Aliases: --baf-window

  --ap-baf-point-count (int32) = Point count to use in calculating allele peaks
    default value: '129'
    Aliases: --baf-point-count

  --ap-baf-bandwidth (float64) = Bandwidth to use in calculating allele peaks
    default value: '0.300000'
    Aliases: --baf-bandwidth

  --ap-baf-cutoff (float64) = Cutoff to use in calculating allele peaks
    default value: '0.000000'
    Aliases: --baf-cutoff

  --ap-baf-threshold (float64) = Clean threshold to use in calculating allele peaks
    default value: '0.300000'
    Aliases: --baf-threshold

  --ap-baf-height-threshold (float64) = Height threshold factor to use in calculating allele peaks
    default value: '3.000000'
    Aliases: --baf-height-threshold

  --ap-baf-height-threshold-bound (float64) = Bounds between which the height threshold is applied
    default value: '1.000000'
    Aliases: --baf-height-threshold-bound

  --ap-baf-symmetry (bool) = Whether to symmetrize the data when calculating allele peaks
    default value: 'True'
    Aliases: --baf-symmetry

  --shrink-factor-3 (float64) = shrink-factor-3
    default value: '0.500000'

  --shrink-factor-4 (float64) = shrink-factor-4
    default value: '0.600000'

  --fld-weights-value (float64) = Value for the fld weights array
    default value: '100.000000'

  --transform-data (bool) = Whether to transform the data
    default value: 'True'

  --penn-cnv-output (bool) = Output log2 ratios and BAFs in Penn CNV format files
    default value: 'False'

  --@help       = print developer help</pre>
</div>
</div>
<!-- contents -->
<!-- start footer part -->
<hr class="footer" />
<address class="footer">
  <small>Generated on Thu Sep 06 2018 02:38:43 for Analysis Power Tools by  <a
  href="http://www.doxygen.org/index.html"><img class="footer"
  src="doxygen.png" alt="doxygen" /> </a> 1.8.5 </small> 
</address>
</body>
</html>
