<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Analysis Power Tools: MANUAL: apt-copynumber-axiom-hmm (2.10.2.1)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>MANUAL:  MANUAL: apt-copynumber-axiom-hmm (2.10.2.1) </h1>  </div>
</div>
<div class="contents">
<h2><a class="anchor" id="Contents"></a>
Contents</h2>
<ul>
<li>
<a href="#intro">Introduction.</a> </li>
<li>
<a href="#quickstart">Quick Start</a> </li>
<li>
<a href="#options">Options</a> </li>

</ul>
<p><a class="anchor" id="intro"></a> </p>
<h2><a class="anchor" id="intro"></a>
Introduction</h2>
<p>__apt-copynumber-axiom-hmm__ implements copy number analysis
for the Axiom family of arrays. The program reports copy number and loss of
heterozygosity (LOH) states using a hidden Markov model.</p>
<p><a class="anchor" id="quickstart"></a> </p>
<h2><a class="anchor" id="QuickStart"></a>
Quick Start</h2>
<p>The basic requirements for a run of apt-copynumber-axiom-hmm are:</p>
<div class="fragment"><pre class="fragment">
The .summary.a5 (or .summary.txt) and .report.txt files from apt-genotype-axiom.

The .cn_models reference file.

The .hmm_regions.txt file 
      </div>
	  
<p><a class="anchor" id="options"></a> </p>
<h2><a class="anchor" id="options"></a>
<h3>Options:</h3>
<div class="fragment"><pre class="fragment">
'apt-copynumber-axiom-hmm'
Version: 2.10.2.1 stash.amer.thermo.com/ma-alg/apt2-genotyping.git/release/APT_2.10.2.1@4574d9db5d(553ea994bc) 2018-11-09-19:37

options:
Common Options (not used by all programs)
--out-dir .\out_dir 
--chip-type .\Axiom 
--use-text-format-summary-file False 
--summary-file .\AxiomGT1.summary.a5 
--report-file .\AxiomGT1.report.txt 

Command-line Parameters:

apt-copynumber-axiom-hmm.exe 

--out-dir .\out_dir 
--chip-type .\Axiom 
--use-text-format-summary-file False 
--summary-file .\AxiomGT1.summary.a5 
--report-file .\AxiomGT1.report.txt 
--arg-file .AxiomHMM.apt2.xml
--analysis-files-path \library file path
--cn-regions-file .hmm_regions.txt 
--reference-file .cn_models 
--log-file .\logfile.out


USER HELP:

--version (flag) = Display the version info for this program.
--user-help (flag) = Display help intended for end-users.
Aliases: --help, -h

--log-file (string8) = The name of the log file. Defaults to 'PROGRAMNAME-NUMBER.log'.
Aliases: --lf</p>



--console-add-select (string8) = Add selectors for console messages.

e.g. to include all debug messages:
--console-add-select debug

to include all messages of level 3 or higher: --console-add-select '*:3'

--console-add-neg-select (string8) = Add selectors to be excluded from console messages.

e.g. to exclude all warning messages and errors 
summary: --console-add-neg-select WARNING,summary

--arg-file (filename) = Read arguments from this file.
File should be a valid XML file, and should contain elements of the form

&lt;Parameter name=&quot;PARAMETER-NAME&quot;
analysis=&quot;NODE-NAME&quot; currentValue=&quot;VALUE&quot; /&gt;.

--check-param (flag) = Stop the program after the command parameters have been checked.

--out-dir (string8) = Output directory for result files

--analysis-name (string8) = Name to prepend to output files
default value: 'AxiomHMM'

--analysis-files-path (string8) = Default directory to search for analysis library files

--summary-file (filename) = Summary file generated by apt-genotype-axiom (used as input)

--reference-file (filename) = Reference file

--mapd-max (float64) = Samples with MAPD greater than this threshold will fail quality control
default value: '0.400000'

--waviness-sd-max (float64) = Samples with waviness standard deviation greater than this threshold will fail quality control
default value: '0.300000'

--chip-type (vec_string8) = Text identifier for chip.
Aliases: --array-type

--use-text-format-summary-file (bool) = Use Axiom summary file in text format instead of a5
default value: 'False'

--cn-regions-file (filename) = File that lists the regions for the CN calculation

--report-file (filename) = Report file generated by apt-genotype-axiom (used as input)

--controls-file (filename) = The file containing list of control sample file names used for per-plate signal correction

--regions-file (filename,required) = The file that defines regions and HMM parameters

--loh-batch-folder (filename) = The batch folder generated by apt-genotype-axiom (used as input). Set either the
loh-batch-folder or loh-calls-file and loh-confidences-file to get LOH segments.

--loh-calls-file (filename) = The calls file generated by apt-genotype-axiom (used as input)

--loh-confidences-file (filename) = The confidences file generated by apt-genotype-axiom (used as input)

--loh-samples-per-block (int32) = Number of samples per block to process fromLOH batch folder or calls file. Only supports up to 200.
default value: '200'

--loh-error-rate (float64) = Error rate to use in LOH processing
default value: '0.050000'

 --loh-beta (float64) = Beta to use in LOH processing
default value: '0.001000'

--loh-alpha (float64) = Alpha to use in LOH processing
default value: '0.010000'

--loh-separation (int32) = Separation to use in LOH processing
default value: '1000000'

--loh-min-marker-count (int32) = Minimum marker count to use in LOH processing
default value: '10'

--loh-no-call-threshold (float64) = No call threshold to use in LOH processing
default value: '0.050000'

--loh-min-genomic-span (int32) = Minimum genomic span to use in LOH processing
default value: '1000000'

--ap-baf-step (int32) = Step to use in calculating allele peaks
default value: '25'
Aliases: --baf-step

--ap-baf-window (int32) = Window to use in calculating allele peaks
default value: '150'
Aliases: --baf-window

--ap-baf-point-count (int32) = Point count to use in calculating allele peaks
default value: '129'
Aliases: --baf-point-count

--ap-baf-bandwidth (float64) = Bandwidth to use in calculating allele peaks
default value: '0.300000'
Aliases: --baf-bandwidth

--ap-baf-cutoff (float64) = Cutoff to use in calculating allele peaks
default value: '0.000000'
Aliases: --baf-cutoff

--ap-baf-threshold (float64) = Clean threshold to use in calculating allele peaks
default value: '0.300000'
Aliases: --baf-threshold

--ap-baf-height-threshold (float64) = Height threshold factor to use in calculating allele peaks
default value: '3.000000'
Aliases: --baf-height-threshold

--ap-baf-height-threshold-bound (float64) = Bounds between which the height threshold is applied
default value: '1.000000'
Aliases: --baf-height-threshold-bound

--ap-baf-symmetry (bool) = Whether to symmetrize the data when calculating allele peaks
default value: 'True'
Aliases: --baf-symmetry

--shrink-factor-3 (float64) = shrink-factor-3
default value: '0.500000'

--shrink-factor-4 (float64) = shrink-factor-4
default value: '0.600000'

--fld-weights-value (float64) = Value for the fld weights array
default value: '100.000000'

--transform-data (bool) = Whether to transform the data
default value: 'True'

--special-snps-file (filename) = (unused) The special-snps file for male/female copy numbers

--seg-min-bases-CN-zero (int32) = Minimum size in bases for segments that report copy number 0
default value: '0'

--seg-min-bases-CN-oneormore (int32) = Minimum size in bases for segments that report copy number greater than 0
default value: '0'

--seg-min-probesets-CN-zero (int32) = Minimum size in probesets for segments that report copy number 0
default value: '0'

--seg-min-probesets-CN-oneormore (int32) = Minimum size in probesets for segments that report copy number greater than 0
default value: '0'

--kernel-sigma-span (float64) = Probes spanned by one standard deviation
default value: '50.000000'

--@help       = print developer help

--arg-file .AxiomHMM.apt2.xml 

--analysis-files-path \library file path

--cn-regions-file .hmm_regions.txt 

--reference-file .cn_models

--log-file .\logfile.out


USER HELP:
--version (flag) = Display the version info for this program.

--user-help (flag) = Display help intended for end-users.
Aliases: --help, -h

--log-file (string8) = The name of the log file. 
Defaults to 'PROGRAMNAME-NUMBER.log'.
Aliases: --lf

--console-add-select (string8) = Add selectors for console messages.
e.g. to include all debug messages:
--console-add-select debug

to include all messages of level 3 or higher: --console-add-select '*:3'
--console-add-neg-select (string8) = Add selectors to be excluded from console messages.

e.g. to exclude all warning messages and errors summary: --console-add-neg-select WARNING,summary

--arg-file (filename) = Read arguments from this file.
File should be a valid XML file, and should contain elements of the form

&lt;Parameter name=&quot;PARAMETER-NAME&quot;
analysis=&quot;NODE-NAME&quot; currentValue=&quot;VALUE&quot; /&gt;.

--check-param (flag) = Stop the program after the command parameters have been checked.

--out-dir (string8) = Output directory for result files

--analysis-name (string8) = Name to prepend to output files
default value: 'AxiomHMM'

--analysis-files-path (string8) = Default directory to search for analysis library files

--summary-file (filename) = Summary file generated by apt-genotype-axiom (used as input)

--reference-file (filename) = Reference file

--mapd-max (float64) = Samples with MAPD greater than this threshold will fail quality control
default value: '0.400000'

--waviness-sd-max (float64) = Samples with waviness standard deviation greater than this threshold will fail quality control
default value: '0.300000'

--chip-type (vec_string8) = Text identifier for chip.
Aliases: --array-type

--use-text-format-summary-file (bool) = Use Axiom summary file in text format instead of a5
default value: 'False'

--cn-regions-file (filename) = File that lists the regions for the CN calculation

--report-file (filename) = Report file generated by apt-genotype-axiom (used as input)

--controls-file (filename) = The file containing list of control sample file names used for per-plate signal correction

--regions-file (filename,required) = The file that defines regions and HMM parameters

--loh-batch-folder (filename) = The batch folder generated by apt-genotype-axiom (used as input). Set either the
loh-batch-folder or loh-calls-file and loh-confidences-file to get LOH segments.

--loh-calls-file (filename) = The calls file generated by apt-genotype-axiom (used as input)

--loh-confidences-file (filename) = The confidences file generated by apt-genotype-axiom (used as input)

--loh-samples-per-block (int32) = Number of samples per block to process fromLOH batch folder or calls file. Only supports up to 200.
default value: '200'

--loh-error-rate (float64) = Error rate to use in LOH processing
default value: '0.050000'

--loh-beta (float64) = Beta to use in LOH processing
default value: '0.001000'

--loh-alpha (float64) = Alpha to use in LOH processing
default value: '0.010000'

--loh-separation (int32) = Separation to use in LOH processing
default value: '1000000'

--loh-min-marker-count (int32) = Minimum marker count to use in LOH processing
default value: '10'

--loh-no-call-threshold (float64) = No call threshold to use in LOH processing
default value: '0.050000'

--loh-min-genomic-span (int32) = Minimum genomic span to use in LOH processing
default value: '1000000'

--ap-baf-step (int32) = Step to use in calculating allele peaks
default value: '25'
Aliases: --baf-step

--ap-baf-window (int32) = Window to use in calculating allele peaks
default value: '150'
Aliases: --baf-window

--ap-baf-point-count (int32) = Point count to use in calculating allele peaks
default value: '129'
Aliases: --baf-point-count

--ap-baf-bandwidth (float64) = Bandwidth to use in calculating allele peaks
default value: '0.300000'
Aliases: --baf-bandwidth

--ap-baf-cutoff (float64) = Cutoff to use in calculating allele peaks
default value: '0.000000'
Aliases: --baf-cutoff

--ap-baf-threshold (float64) = Clean threshold to use in calculating allele peaks
default value: '0.300000'
Aliases: --baf-threshold

--ap-baf-height-threshold (float64) = Height threshold factor to use in calculating allele peaks
default value: '3.000000'
Aliases: --baf-height-threshold

--ap-baf-height-threshold-bound (float64) = Bounds between which the height threshold is applied
default value: '1.000000'
Aliases: --baf-height-threshold-bound

 --ap-baf-symmetry (bool) = Whether to symmetrize the data when calculating allele peaks
default value: 'True'
Aliases: --baf-symmetry

--shrink-factor-3 (float64) = shrink-factor-3
default value: '0.500000'

--shrink-factor-4 (float64) = shrink-factor-4
default value: '0.600000'

--fld-weights-value (float64) = Value for the fld weights array
default value: '100.000000'

--transform-data (bool) = Whether to transform the data
default value: 'True'

--special-snps-file (filename) = (unused) The special-snps file for male/female copy numbers

--seg-min-bases-CN-zero (int32) = Minimum size in bases for segments that report copy number 0
default value: '0'

--seg-min-bases-CN-oneormore (int32) = Minimum size in bases for segments that report copy number greater than 0
default value: '0'

--seg-min-probesets-CN-zero (int32) = Minimum size in probesets for segments that report copy number 0
default value: '0'

--seg-min-probesets-CN-oneormore (int32) = Minimum size in probesets for segments that report copy number greater than 0
default value: '0'

--kernel-sigma-span (float64) = Probes spanned by one standard deviation
default value: '50.000000'

--@help       = print developer help

</pre></div> </div>
<hr class="footer"/><address class="footer"><small>Generated for Analysis Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
