<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Analysis Power Tools: MANUAL: apt-genotype-eureka (2.10.0)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div class="header">
  <div class="headertitle">
<div class="title"><h1>MANUAL: apt-genotype-eureka (2.10.0)</h1></div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><h2>Contents </h2>
<ul>
<li><a href="#intro">Introduction.</a></li>
<li><a href="#quickstart">Quick Start</a></li>
<li><a href="#options">Options</a></li>
<li><a href="#references">References</a></li>
</ul>
<p><a class="anchor" id="intro"></a> </p>
<h2>Introduction </h2>
<p><b>apt-genotype-eureka</b> is a program for making genotyping calls on the Affymetrix Eureka&trade; platform. <code>apt-genotype-eureka</code> uses the BRLMM-P algorithm for genotype calling. BRLMM-P is a model-based approach which performs 1-dimensional clustering by fitting a Gaussian mixture model.</p>
<p><a class="anchor" id="quickstart"></a> </p>
<h2>Quick Start </h2>
<p>The basic requirements for a run of apt-genotype-eureka are</p>
<ul>
<li>a binning file generated by the <a href="nibls.html">nibls</a> application;</li>
<li>a priors file.</li>
</ul>
<p>Additional parameters may be specified on the command line or by means of a parameter XML file.</p>
<p>The command below runs the basic genotyping analysis on a Eureka binning file:</p>
<pre class="fragment">  apt-genotype-eureka \
      --chip-type eureka \
      --binning-file ./input/Binning.txt \
      --snp-priors-input-file ./inputs/referencePriors_heavy_CES.txt \
      --out-dir out \
      --cel-files cel_file_list.txt \
      --log-file ./apt-genotype-eureka.log
</pre><p><a class="anchor" id="options"></a> </p>
<h2>Options: </h2>
<pre class="fragment">
   'apt-genotype-eureka'
   2.10.0 stash.amer.thermo.com/ma-alg/apt2-genotyping.git/release/APT_2.10.0@321b13f822(38a8486d9a) 2018-04-20-06:58

USER HELP ----------

  --version (flag) = Display the version info for this program.

  --user-help (flag) = Display help intended for end-users.
    Aliases: --help, -h

  --log-file (string8) = The name of the log file.  Defaults to 'PROGRAMNAME-NUMBER.log'.
    Aliases: --lf

  --console-add-select (string8) = Add selectors for console messages.
    e.g. to include all debug messages: --console-add-select debug
         to include all messages of level 3 or higher: --console-add-select '*:3'


  --console-add-neg-select (string8) = Add selectors to be excluded from console messages.
    e.g. to exclude all warning messages and errors summary: --console-add-neg-select WARNING,summary


  --arg-file (filename) = Read arguments from this file.
    File should be a valid XML file, and should contain elements of the form
    &lt;Parameter name="PARAMETER-NAME" analysis="NODE-NAME" currentValue="VALUE" /&gt;.

  --check-param (flag) = Stop the program after the command parameters have been checked.

  --out-dir (string8) = The output directory for result files.

  --temp-dir (string8) = The output directory for temporary files.

  --analysis-files-path (dirname) = Default directory to search for analysis library files

  --batch-folder (string8) = Full path to analysis data batch folder. All genotyping data will be streamed into this folder as'all_genotypes_by_snps.CHP.bin' and 'all_genotypes_by_snps.CHP.index.txt'.

  --batch-folder-data-dir (string8) = Name of directory inside batch-folder where the analysis suite data is written.
    default value: 'AnalysisSuiteData'

  --delete-temp-files (bool) = Delete the temporary files after the run completes.
    default value: 'True'

  --chip-type (vec_string8,required) = Text identifier for chip.
    Aliases: --array-type

  --analysis-name (string8) = Name to prepend to output files.

  --binning-file (filename,required) = Temporarily reading input file, later it can be deligated to library-file-node

  --report (bool) = Print report.txt file of summary and genotyping statistics.
    default value: 'True'

  --report-file (string8) = Optional full path of report.txt file.

  --summaries-only (bool) = Skip the genotyping step.
    default value: 'False'

  --probeset-ids (string8) = Tab delimited file with column 'probeset_id' specifying probesets to genotype.

  --sample-list (string8) = List of samples indexes to be processed.  Other samples in the binning file will be excluded from all analysis.

  --inbred-file (string8) = File specifying inbred het penalty for each CEL file.
    File must have columns cel_files and inbred_het_penalty.
    Aliases: --read-inbred

  --hints-file (string8) = File specifying seed genotypes instead of DM algorithm
    File must have first column 'probeset_id' and one additional column per CEL file, with file basename as column name.
    Rows must have probeset id for first column and integers corresponding to calls for remaining columns.
    Calls represented by 'No Call' = -1, 'AA' = 0, 'AB' = 1, 'BB' = 2.
    Aliases: --genotypes

  --sample-multiplier (int32) = Used for generating fake blob with large number of samples. Ex: 10 samples x 500 multiplier = 5000 fake samples.
    default value: '1'

  --allele-summaries (bool) = Print file of intensities summarized by allele.
    default value: 'False'
    Aliases: --summaries

  --allele-summaries-file (string8) = Optional fully-specified path of allele-summaries.txt file.
    Aliases: --summaries-file

  --trim-percent (int32) = Percentage amount to trim from each tail of the distribution.
    default value: '5'

  --signal-scale-factor (int32) = Normalized results are multiplied by this number to make them look more like Axiom summary signals.
    default value: '550'

  --trustcheck-threshhold (int32) = Threshhold to filter out blemished probesets
    default value: '10'

  --trimmed-mean-output-trustcheck (bool) = Output text file of blemished probesets with cel index
    default value: 'False'

  --snp-priors-input-file (string8,required) = The string that identifies the snp priors to use when genotyping.

  --snp-posteriors-output (bool) = Write snp posteriors to output when genotyping.
    default value: 'False'
    Aliases: --write-models

  --snp-posteriors-output-file (string8) = Optional full path of snp-posteriors.txt file.

  --table-output (bool) = Output matching matrices of tab delimited genotype calls and confidences. [default 'true']
    default value: 'True'

  --calls-file (string8) = Optional full path of calls.txt file.

  --confidences-file (string8) = Optional full path of confidences.txt file.

  --brlmmp-HARD (int32) = brlmmp hardshell
    default value: '3'

  --brlmmp-SB (float64) = brlmmp shellbarrier
    default value: '0.450000'

  --brlmmp-CM (int32) = brlmmp callmethod (1=batch mode, 2=single sample mode)
    default value: '1'

  --brlmmp-bins (int32) = brlmmp bins
    default value: '100'

  --brlmmp-mix (int32) = brlmmp mix
    default value: '1'

  --brlmmp-bic (float64) = brlmmp bic
    default value: '2.000000'

  --brlmmp-CSepPen (float64) = brlmmp CSepPen
    default value: '0.000000'

  --brlmmp-CSepThr (float64) = brlmmp CSepThr
    default value: '16.000000'

  --brlmmp-lambda (float64) = brlmmp lambda
    default value: '1.000000'

  --brlmmp-wobble (float64) = brlmmp wobble
    default value: '0.050000'

  --brlmmp-copyqc (float64) = brlmmp copyqc. 0 = no test for copy qc
    default value: '0.000000'

  --brlmmp-copytype (int32) = brlmmp copytype. 0 = standard copy qc method
    default value: '0'

  --brlmmp-MS (float64) = brlmmp MS. confidence threshold
    default value: '0.050000'

  --brlmmp-ocean (float64) = brlmmp ocean
    default value: '0.000000'

  --brlmmp-clustertype (int32) = brlmmp clustertype. 1 = standard 1-d clustering only
    default value: '1'

  --brlmmp-transform (string8) = brlmmp transform. (mva, rvt, ssf, ces, assf, ccs)
    default value: 'mva'

  --brlmmp-hints (int32) = brlmmp hints
    default value: '0'

  --brlmmp-IsoHetY (float64) = brlmmp IsoHetY
    default value: '0.000000'

  --brlmmp-inflatePRA (float64) = brlmmp inflatePRA. 0 = no increase in uncertainty
    default value: '0.000000'

  --brlmmp-CP (float64) = brlmmp contradiction penalty. 0 = no penalty
    default value: '16.000000'

  --brlmmp-comvar (bool) = boolean to indicate if common variance is used in all clusters.
    default value: 'True'

  --brlmmp-K (float64) = Scaling Parameter to use in transformations.
    default value: '1.000000'

  --@help       = print developer help

</pre><p><a class="anchor" id="faq"></a> </p>
<h2>Frequently Asked Questions </h2>
<p><a class="anchor" id="references"></a> </p>
<h2>References </h2>
<ul>
<li>BRLMM-P - <a href="http://www.affymetrix.com/support/technical/whitepapers/brlmmp_whitepaper.pdf">http://www.affymetrix.com/support/technical/whitepapers/brlmmp_whitepaper.pdf</a></li>
<li>PLIER - <a href="http://media.affymetrix.com/support/technical/technotes/plier_technote.pdf">http://media.affymetrix.com/support/technical/technotes/plier_technote.pdf</a> </li>
</ul>
</div></div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Sun Jun 24 2018 13:29:25 for Analysis Power Tools by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
