<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
      "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="content-type"
  content="application/xhtml+xml; charset=UTF-8" />
  <title>Analysis Power Tools: CHANGE LOG</title>
  <link href="tabs.css" rel="stylesheet" type="text/css" />
  <link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>

<body>
<!-- Generated by Doxygen 1.7.1 -->

<div class="navigation" id="top">

<div class="tabs">
<ul class="tablist">
  <li><a href="index.html"><span>Main Page</span></a></li>
  <li class="current"><a href="change`.html"><span>Change Log</span></a></li>
  <li><a href="vignette.html"><span>Vignettes</span></a></li>
  <li><a href="faq.html"><span>FAQs</span></a></li>
  <li><a href="file-formats.html"><span>File Formats</span></a></li>
  <li><a href="platforms.html"><span>Platforms</span></a></li>
</ul>
</div>
</div>

<div class="header">

<div class="headertitle">
<h1>CHANGE LOG </h1>
</div>
</div>

<div class="contents">
<dl class="version">
  <dt>Version:</dt>
    <dd>apt-2.10.2.2 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Thursady January 03, 2019</dd>
</dl>

<p>The 2.10.2.2 release of APT is a patch release to fix a critical defect</p>
<p><b>Changed Applications:</b></p>

<p><strong> apt-format-result.exe</strong></p> 

<h3>Bug Fixes </h3>
<ul>
  <li>APTNG-2180: Add option to sort PLINK .tped and .ped order by chromosome and position.
  <li>APTNG-2183: Remove unnecessary serialize files
</ul>


<dl class="version">
  <dt>Version:</dt>
    <dd>apt-2.10.2.1 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Wednesday December 05, 2018</dd>
</dl>

<p>The 2.10.2.1 release of APT is a patch release to support copy number analysis of selected AgBio arrays in Axiom Analysis Suite
Highlights/Features and improvements include:</p>

<p><b>Changed Applications:</b></p>

<p><strong> apt-copynumber-axiom-hmm:</strong></p> 
<p><strong> apt-copynumber-axiom-safer:</strong></p> 
<p><strong> apt-copynumber-axiom-cnvmix:</strong></p>
<ul>
  <li>The current release of APT 2.10.2 does not allow for non gender arrays to run the copy number workflows. If attempted, the analysis will error out. This is because the copy number workflows require a fixed column in the report.txt file for the plate barcode. This restriction has been removed in In AxAS 2.10.2.1 in the above mentioned applications</li>
</ul>
<h3>Bug Fixes </h3>
<ul>
  <li>APTNG-2193: Copy Number Engines errors out when the input report-file has scrambled columns
</ul>


<dl class="version">
  <dt>Version:</dt>
    <dd>apt-2.10.2 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Saturday August 25, 2018</dd>
</dl>

<p>The 2.10.2 release of APT is a patch release to support copy number analysis of selected AgBio arrays in Axiom Analysis Suite
Highlights/Features and improvements include:</p>

<p><b>Changed Applications:</b></p>

<p><strong> apt-copynumber-axiom-hmm:</strong></p> 
<p><strong> apt-copynumber-axiom-safer:</strong></p> 
<p><strong> apt-copynumber-axiom-cnvmix:</strong></p>
<ul>
  <li>supports analysis of  non human diploid arrays</li>
</ul>
<p><strong> apt-copynumber-cyto-ssa:</strong></p> 
<p><strong> apt-copynumber-onco-ssa:</strong></p>
<ul>
  <li>Allows the user to suppress any gender reporting or data generated on Chromosomes X and Y through a newly introduced parameters.</li>
</ul>

<p><strong> apt-format-result:</strong> </
  <li>updates for PLINK export include</li>
  <ul>
      <li>option to sort by chromosome and position.</li>
  </ul>
  <li>updates for VCF export include</li>
  <ul>
      <li> rule added: all exported IDs must be unique.</li>
	  <li> if ps.performance.txt is supplied:</li>
	  <li> For probesets with duplicate reported ID always have apt-format-result select probeset where BestProbeset=1.</li>
	  <li> If there are still multiple probesets per ID, pick sorted first probeset</li>
	  <li> if not supplied: probeset list, ps.performance, recommended.ps: there will be a warning: "WARNING: Duplicate probeset for ID [value] not written to VCF"</li>
  </ul>
  <li> format output: metadata section and INFO column will contain information from ps.performance.txt</li>
  <ul>
      <li> INFO column will contain CR:probeset_id;ConversionType;</li>
	  <ul>
		<li>If probeset_id is NOT selected as the annotation to use for the ID field, the INFO field shall include the probeset_id</li>
	  </ul>
	  <li> metadata section will include:</li>
	  <ul> ##INFO=&ltID=CR,Number=1,Type=Float,Description="Call Rate" </ul>
	  <ul> ##INFO=&ltID=probeset_id,Number=1,Type=String,Description="Measured probeset" </ul>
	  <ul> ##INFO=&ltID=ConversionType,Number=1,Type=String,Description="Probeset category" </ul>
  </ul>
</ul>
</ul>

<h3>Bug Fixes </h3>
<ul>
  <li>APTNG-1414: Require set of CN regions to be consistent between header and table of AxiomCN.cnregioncalls.txt.</li>
  <li>APTNG-2104: when adding params to RootGdh for cychp and cnchp, filter out ones that start with '@'.</li>
  <li>APTNG-2057: apt-format-result: Add option to sort PLINK map order by chromosome and position..</li>
  <li>APTNG-2058: apt-format-result: Move all gtest/gold and gtest/input data to apt-regression-data repository.</li>
  <li>APTNG-2083: add _&keyvals to summary.a5 output when --summary-a5-output option is set.</li>
  <li>APTNG-2067: HMM: enable non-human chromosome, beyond Chr 22</li>
  <li>APTNG-1958: apt-format-result: VCF Format output: exported ID must be unique, no duplicate markers exported.</li>
  <li>APTNG-1953: apt-format-result: VCF format output: metadata section and INFO column will contain information from ps.performance.txt.</li>
  <li>APTNG-1983: apt-format-result--export-vcf-export: update to include in FILTER column PASS or FAIL if probeset in ps.performance.txt in bestandrecommended column.</li>
</ul>



<dl class="version">
  <dt>Version:</dt>
    <dd>apt-2.10.0 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Sunday June 24, 2018</dd>
</dl>

<p>The 2.10.0 release of APT supports the Axiom Analysis Suite 4.0 release.
Highlights/Features and improvements include:</p>

<p><b>New Applications:</b></p>
<ul>
  <li><strong>apt-copynumber-axiom-ref</strong>: a program to create CN
    reference model files for copy number enabled arrays</li>
  <li><strong>apt-copynumber-axiom-hmm</strong>: a program for that utilizes a
    hidden Markov model to report copy number and LOH</li>
  <li><strong>apt-copynumber-axiom-cnvmix</strong>: a program that reports copy
    number and LOH in predefined regions</li>
</ul>

<p></p>

<p><b>Changed Applications:</b></p>

<p><strong>apt-copynumber-axiom-ssa:</strong> </p>
<ul>
  <li>is now apt-copynumber-axiom-safer</li>
  <li>LOH segment generation and BAF output has been enabled</li>
  <li>Option to write to a .cnv.a5 file which includes BAF, L2R, sample,
    probeset and chromosome information</li>
</ul>

<p><strong>apt-format-results:</strong> </p>
<ul>
  <li>include export to IGV option for all CN batches</li>
  <li>support use of alternate sample names</li>
  <li>support exporting genotype data for only a subset of samples</li>
  <li>ability to export to VCF for apt-copynumber-axiom-hmm runs</li>
  <li>updates for PLINK export include
    <ul>
      <li>export short chromosome names.</li>
      <li>export abstract call codes.</li>
    </ul>
  </li>
</ul>
<ul>
  <li>updates for VCF export include
    <ul>
      <li>export short names instead of Chr_id.</li>
      <li>format output: chromosome in chronological order.</li>
    </ul>
  </li>
</ul>
<ul>
  <li>new option: --export-basecall-otv to distinguish between NoCall and OTV
    when exporting with forward-strand base call using AxiomGT1.calls.txt</li>
</ul>

<p><strong>apt-package-util: </strong></p>
<ul>
  <li>include copy number command line output</li>
  <li>include all command line output for viewing in AxAS</li>
</ul>

<p><strong>apt-cel-convert: </strong></p>
<ul>
  <li>add option to extract header key/vale pairs only</li>
  <li>add option to convert a list of cel files from a given directory</li>
  <li>supports output formats of calvin, hdf5 and tsv</li>
</ul>

<h3>Bug Fixes </h3>
<ul>
  <li>APTNG-2000: Small HMM segment filtering with four new parameters</li>
  <li>APTNG-1952:apt-format-result: VCF format output; chromosome should be in
    chronological order.</li>
  <li>APTNG-1967:Small HMM segment filtering with two new parameters.</li>
  <li>APTNG-1993:Change default analysis_name to "AxiomCNVMix".</li>
  <li>APTNG-1803:cn_models File: when users create their own reference file,
    create a log of run details.</li>
  <li>APTNG-1698:Add qc node to apt-copynumber-axiom-hmm-ssa.</li>
  <li>APTNG-1872:Change the probeset in the adjusted summary file to be a
    calculated fixed length string instead of variable length.</li>
  <li>APTNG-1701:LOH qc metric for Axiom CNV.</li>
  <li>APTNG-1930:remove warning when more than two plate control samples are
    used for fixed region CN calling.</li>
  <li>APTNG-1882:CN models and cnv.a5 headers should include apt parameters
    used to create it.</li>
  <li>APTNG-1548:apt-format-result: TXT base_call option: don't export CN=1
    genotype codes as "---".</li>
  <li>APTNG-1905:rename -chimera-celfiles-by-regions-file.</li>
  <li>APTNG-1906:delete temp files after each iteration of fixed region CN
    reference generation.</li>
  <li>APTNG-1933:Additional sort requirement for *cnregioncalls.txt.</li>
  <li>APTNG-1939:Make VCF containing only results from cn-region-calls-file TXT
    format.</li>
  <li>APTNG-1526:Reference Model File Creation for AxAS.</li>
  <li>APTNG-1859:Include Export to IGV option to all CN batches in AxAS.</li>
  <li>APTNG-1864:PSSW-CNVX-0030103 The software will convert CN segment data
    into IGV compatible format.</li>
  <li>APTNG-1904:Export Nexus Copy Number Format.</li>
  <li>APTNG-1860:PSSW-CNVX-0030104 The software will convert LOH data into IGV
    compatible format.</li>
  <li>APTNG-1861:PSSW-CNVX-0030106 The software will convert log2ratio data
    from Axiom into IGV format.</li>
  <li>APTNG-1862:PSSW-CNVX-0030102 The software will convert smooth signal data
    from Axiom into IGV format.</li>
  <li>APTNG-1863:PSSW-CNVX-0030105 The software will convert BAF data into IGV
    format.</li>
  <li>APTNG-1704:apt-format-results will include VCF export option for HMM.</li>
  <li>APTNG-1765:New output file for SAFER: hfd5 file which will include BAF
    and log2ratio, sample, probeset and chromosome information.</li>
  <li>APTNG-1766:HMM will be outputted in hdf5 format.</li>
  <li>APTNG-1773:LOH: include option to start from blob.</li>
  <li>APTNG-1865:PSSW-CNVX-0020101 The software will generate CN Segments, BAF,
    Log2Ratio, Smooth Signal and LOH Segments for Axiom data..</li>
  <li>APTNG-1866:There will be a default flag set to true for Safer:
    AxiomCN.cnv.a5 output option.</li>
  <li>APTNG-1878:add chip-type parameter to apt-copynumber-axiom-hmm.</li>
  <li>APTNG-1899:Reference Creation: Probeset-ID to mask probeset that don’t
    want used. option-hidden currently, can be unhidden: show option.</li>
  <li>APTNG-1887:update help text for chimera-celfiles-by-regions-file.</li>
  <li>APTNG-1888:read gender calls from input report-file, then pass to output
    AxiomHMM.report.txt.</li>
  <li>APTNG-1740:APT-package-util: Include all command line output for viewing
    in AxAS.</li>
  <li>APTNG-1731:APT-Package-Util: include copy number command line output.</li>
  <li>APTNG-1983:apt-format-result--export-vcf-export: update to include in
    FILTER column PASS or FAIL if probeset in ps.performance.txt in
    bestandrecommended column.</li>
  <li>APTNG-1572:Reference Model File: cn_models file creation.</li>
  <li>APTNG-1538:APT2-Core: Improve handling by using aptErrorPrintf and
    aptWarnPrintf to immediately flush out errors and warnings to log file.</li>
  <li>APTNG-1360:DRSW-AGD007 apt-genotype-axiom shall support multiallele
    genotyping.</li>
  <li>APTNG-1341:Multiallele posterior header additions to describe field
  order.</li>
</ul>
</div>

<div class="contents">
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.21.0 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Wednesday November 15, 2017</dd>
</dl>

<p>The 1.21.0 release of APT supports the Chromosome Analysis Suite 3.3
release. Highlights include:</p>
<ul>
  <li>Support for the CytoScan XON array</li>
  <li>Export probe level and segment level data into IGV format.</li>
</ul>

<h3>New Features and Improvements </h3>
<ul>
  <li>To provide support for CytoScan XON arrays. Details can be found in the
    CytoScan XON projects.</li>
</ul>

<h3>Bug Fixes </h3>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/CHAS-17055">CHAS-17055</a>] -
    CHAS-17055: apt-copynumber-cyto-ssa: BAF is not generated for the Y
    chromosome for CytoScan HD arrays</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1678">APTNG-1678</a>] -
    apt-copynumber-cyto-ssa: XON calling algorithm PAR handling for genomes
    other than hg19.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1272">APTNG-1272</a>] -
    apt-copynumber-cyto-ssa: Run time exception when running 3000 samples with
    hdf5 summary file.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-849">APTNG-849</a>] -
    apt-copynumber-cyto-ssa: ERROR MainNode::standardMain(): Unhandled
    Exception! (Unknown) running cyto apt2 reference.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1597">APTNG-1597</a>] -
    apt-copynumber-cyto-ssa: App does not parse parameter string correctly:
    'crf-init-prob 0.1;0.8,0.1'.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1611">APTNG-1611</a>] -
    apt-copynumber-cyto-ssa: Data types in ExonRegion and values in Chromosome
    column are not correct.</li>
</ul>
</div>

<div class="contents">
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.20.6 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Friday August 11, 2017</dd>
</dl>

<p>The 1.20.6 release of APT supports the Axiom Analysis Suite 3.1 release.
Highlights include:</p>
<ul>
  <li>SNP QC for non-PAR X probesets is handled differently, and so the
    ‘CategoryType’, ‘BestProbeset’, and ‘BestandRecommended’ values
    for a probeset may change:</li>
  <li>a. Some probesets that should be categorized as ‘PolyHighResolution’
    were mis-assigned to another ‘CategoryType’. This has been fixed.</li>
  <li>b. Previously, non-PAR X call rates for males and for females were both
    compared to the call rate threshold, and if either failed then the probeset
    might be classified as ‘CallRateBelowThreshold’. If all samples were
    assigned to a single gender, the probeset would fail the call rate
    threshold. The change for this release is that now total call rate for all
    samples is compared against the call rate threshold. Note that the
    displayed call rate in the SNP Summary Table for non-PAR X probesets
    continues to be reported only for female samples.</li>
  <li>c. New SNP QC parameter ‘het-so-XChr-offset’ has been added for
    non-PAR X regions.</li>
  <li>Arrays with multi-allele probesets now properly save multi-allele
    posterior information for probesets with more than 6 alleles..</li>
</ul>

<h3>New Features and Improvements </h3>
<ul>
  <li>To provide support for Carrier Screening. Details can be found in Axiom
    Console and Carrier Reporter projects</li>
</ul>

<h3>Bug Fixes </h3>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1658">APTNG-1658</a>] -
    apt-genotype-axiom: Calls are not correct, based on trust-check file.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1652">APTNG-1652</a>] -
    apt-genotype-axiom: output calls file contains unexpected call codes for
    multi-allele probesets when has been ran without copy-number call
  codes.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1579">APTNG-1579</a>] -
    apt-genotype-axiom: multi-allele snp-posteriors file has incorrect
    pre-header information and covariance data for probesets with 7 and 8
    alleles.</li>
</ul>
</div>

<div class="contents">
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.20.5 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Wednesday March 21, 2017</dd>
</dl>

<p>The 1.20.5 release of APT supports the Axiom Analysis Suite 3.0 release.
Highlights include:</p>
<ul>
  <li>Apt-genotype-axiom can execute genotyping of probesets with more than two
    alleles, if supplied with Axiom Analysis library files for multi-allele
    processing. Please note that support of the multi-allelic genotyping
    workflow on Axiom custom arrays is not standard practice. The request must
    be communicated at the time of designing the custom array to verify the
    workflow can be supported. Execution of best practices multiallelic
    genotyping workflows require additional steps that are not automatically
    supported by APT </li>
  <li>Metrics now supports MA and non-MA probesets and produces one metrics
    file for each probeset type (MA and non-MA).</li>
  <li>Metrics now supports CN probesets, including CN+MA probesets</li>
  <li>Supplemental and SSP metrics added to the metrics function.</li>
  <li>New metric MMD added to metrics.</li>
  <li>Classification now supports CA, MA, and CN+MA probesets. One performance
    file is produced from the two separate MA and non-MA metrics files.</li>
  <li>Classification updated to handle supplemental and SSP workflows.
    Classification now includes the supplemental variance categories.</li>
  <li>Special SNP handling added to metrics and classification: Y/MT probesets
    have their own set of metrics and categorization cut-offs, and non-PAR X
    probesets have their own set of metrics and categorization cut-offs.</li>
</ul>

<h3>New Features and Improvements </h3>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-929">APTNG-929</a>] -
    apt-package-util: Update to include the affymetrix-plate-barcode and
    affymetrix-plate-peg-wellposition .</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1057">APTNG-1057</a>] -
    apt-package-util: Should be able to create genotyping blobs (v1 and v2)
    from SNP-6 CHP files..</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1078">APTNG-1078</a>] -
    apt-genotype-axiom: refactor GenotypeNodeRmaSummarization to accept more
    than 2 alleles.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1098">APTNG-1098</a>] -
    apt-genotype-axiom: enable genotyping node to perform pairwise genotyping
    on a probeset.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1099">APTNG-1099</a>] -
    apt-genotype-axiom: create pipeline for handling mutli-allelic
  probesets.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1281">APTNG-1281</a>] -
    apt-genotype-axiom: create reader and writer for multiallelic priors
  file.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1282">APTNG-1282</a>] -
    apt-genotype-axiom: allow priors file to have an allelePair field for
    pairwise multiallele genotyping.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1292">APTNG-1292</a>] -
    modify SpfFileReader to read probe_seq_id field, if present.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1293">APTNG-1293</a>] -
    output posteriors file with allelePair field if priors file has that
  field.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1300">APTNG-1300</a>] -
    apt-genotype-axiom: Combine biallelic calls into a single multiallelic
  call.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1308">APTNG-1308</a>] -
    apt-blob-extract: Add ability to extract allele signals, contrast and size
    from batch folder.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1311">APTNG-1311</a>] -
    apt-format-result: Add support for PLINK export of multialleles and
    CN-specific genotypes.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1312">APTNG-1312</a>] -
    apt-format-result: Add support for TEXT export of multiallelic genotyping
    results.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1313">APTNG-1313</a>] -
    apt-format-result: Add support for VCF export of multiallelic genotyping
    results.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1324">APTNG-1324</a>] -
    apt-genotype-axiom: Set maximum # of alleles for multiallele genotyping
    blob.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1340">APTNG-1340</a>] -
    Change default multiallele posteriors filename.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1341">APTNG-1341</a>] -
    Multiallele posterior header additions to describe field order.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1360">APTNG-1360</a>] -
    DRSW-AGD007 apt-genotype-axiom shall support multiallele genotyping.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1363">APTNG-1363</a>] -
    apt-blob-extract: Write unique call codes and the number of occurrences for
    each.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1422">APTNG-1422</a>] -
    Do we need to have AxiomGT1.multi-snp-posteriors.txt in the package as
    output from apt-package-util if input has it.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1427">APTNG-1427</a>] -
    Add ability to exclude a set of probes from the OncoScan MAPD and ndSNPQC
    calculations.</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1134">APT-1134</a>]
    - Blemish info output.</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1272">APT-1272</a>]
    - apt-probeset-summarize - DABG probe aggregation improvement.</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1364">APT-1364</a>]
    - Build script should be able to determine location for appropriate
    platform and APT version to send binaries to.</li>
</ul>

<h3>Bug Fixes </h3>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-928">APTNG-928</a>] -
    apt-package-util fails in case if UNC path is too long.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-962">APTNG-962</a>] -
    apt-package-util: returns wrong All_genotypes_by_snps.CHP.bin files for
    some Eureka-specific datasets.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1070">APTNG-1070</a>] -
    apt-format-result: progress not displayed in the console while operating
    with big files.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1253">APTNG-1253</a>] -
    apt-format-result: Marker ID not converted from Chr_id 24 -&gt; X -&gt; Chr
    23 when Chr_id is used as SNP identifier.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1260">APTNG-1260</a>] -
    apt-format-result: Missing identifier warnings not include in final count
    in run summary.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1262">APTNG-1262</a>] -
    Differences in Log Ratio, Strength fields of Genotype dataset between
    apt1.17 and 1.18.2 releases.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-960">APTNG-1269</a>] -
    apt-package-util: different error handling comparing to the APT 1.18.2
  .</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1270">APTNG-1270</a>] -
    apt-package-util creates blob version 1 when --blob-format-version 2.0 is
    used.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1274">APTNG-1274</a>] -
    Version number for apt2 application for win only is not correct.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1286">APTNG-1286</a>] -
    apt-package-util creates wrong call table in the
    All_genotypes_by_snps.CHP.index.txt.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1289">APTNG-1289</a>] -
    apt-copynumber-cyto-ssa: Disable shrinkage v2 node and use log2 ratio (no
    smoothing) instead.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1303">APTNG-1303</a>] -
    apt2-copynumber-axiom: Investigate segmentation fault for CentOS-7 with gcc
    4.8.5.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1307">APTNG-1307</a>] -
    apt-blob-extract: Missing header information when extracting call codes
    from batch folder.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1310">APTNG-1310</a>] -
    Error: Missing summaries necessary for genotyping with parameter:
    --summaries-only true.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1315">APTNG-1315</a>] -
    apt crash when not using --use-copynumber-call-codes option.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1317">APTNG-1317</a>] -
    Runtime exception in apt-genotype-axiom: Unhandled std::exception
    (vector::_M_range_check).</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1318">APTNG-1318</a>] -
    apt-format-result: Investigate export issue with multialleles. Not reading
    multiallele genotyping blob correctly.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1319">APTNG-1319</a>] -
    apt-format-result: Numeric call codes not behaving properly when exporting
    from multiallele blob data.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1320">APTNG-1320</a>] -
    genotyping complains that a GENERIC prior has not been provided, when
    actually it has.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1327">APTNG-1327</a>] -
    apt-genotype-axiom: Application crashes when running with multiallele data
    on Windows platform.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1330">APTNG-1330</a>] -
    apt-format-result: Must dynamically and accurately calculate # of bytes per
    sample block for legacy and multiallele blob formats.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1331">APTNG-1331</a>] -
    apt-genotype-axiom multi-allele truncated posterior output.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1332">APTNG-1332</a>] -
    apt-format-result: Multiallele Plink not exporting properly when using a
    list of ordered alleles.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1333">APTNG-1333</a>] -
    Blob index header file is not correct.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1342">APTNG-1342</a>] -
    apt-package-util: legacy call codes are used inside blob when
    --create-multiallelic-blob option is set to True and --blob-format-version
    option is 2.0.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1343">APTNG-1343</a>] -
    Difference in the call files for run using binary or ascii cdf files.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1344">APTNG-1344</a>] -
    Differences between java and apt2 results after background and pairwise
    nodes.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1345">APTNG-1345</a>] -
    export plink and export plinkt file are not transposed.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1346">APTNG-1346</a>] -
    modes() incorrectly handles a vector of unique items.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1349">APTNG-1349</a>] -
    apt-package-util: call codes in the blob and in the AxiomGT1.calls.txt
    mismatch for some probesets.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1350">APTNG-1350</a>] -
    Call values for male samples is outside of the call values list allows by
    call code table.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1351">APTNG-1351</a>] -
    Windows build script contains unescaped spaces in the MSBUILD pathname.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1354">APTNG-1354</a>] -
    Differences in calls between java and apt2 output afther background and
    pairwise nodes.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1356">APTNG-1356</a>] -
    apt-copynumber-axiom-ssa limited to 128 copy number regions.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1358">APTNG-1358</a>] -
    Call file header is inconsistent with call table implemented and warning in
    the log file.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1359">APTNG-1359</a>] -
    hom_rate total_hom_rate are different in th ereport file if using ascii or
    binary cdf files.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1361">APTNG-1361</a>] -
    Call and index files header have different call tables.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1362">APTNG-1362</a>] -
    Difference in calls between java and apt2 for special snps data.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1367">APTNG-1367</a>] -
    apt-format-result: Call codes were not translated properly for multiallele
    markers in standard PLINK export.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1370">APTNG-1370</a>] -
    500 samples Axiom_CARRIER array run error message ASSERT:
    calvin_databuf_fpos==tmp_fpos: File position index is not at start of
    dataset.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1372">APTNG-1372</a>] -
    apt-package-util does not work with multi-allele packages.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1378">APTNG-1378</a>] -
    apt-genotype-axiom: SVN version extraction is not correct if CYGWIN_HOME is
    set.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1380">APTNG-1380</a>] -
    Errror about wrong file version if input has output bi-allelic
    snp-posteriors file as input.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1381">APTNG-1381</a>] -
    No specific or GENERIC prior provided for probeset: AFFX-SP-000001 but it
    is exist in the ../default/AxiomGT1.snp-posteriors.txt \ .</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1382">APTNG-1382</a>] -
    Segmentation fault.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1383">APTNG-1383</a>] -
    Unknown exception occurred in multiallele genotyping node.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1384">APTNG-1384</a>] -
    Pairwise results shouldn't be written to blob.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1388">APTNG-1388</a>] -
    Call code is not correct for --use-copynumber-call-codes false.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1390">APTNG-1390</a>] -
    apt-genotype-axiom: Not able to run with debug mode on Windows platform due
    to deletion of a null reference.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1391">APTNG-1391</a>] -
    apt-genotype-axiom: Prevent unit tests from registering same node multiple
    times. Caused memory leaks and segmentation fault on Linux with C++11.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1395">APTNG-1395</a>] -
    APT2-Core: PackageSDK: IndexHeader not initializing properly when parsing
    sample names with '=' in debug mode on Mac-OS.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1396">APTNG-1396</a>] -
    Differences is not correct for calls value between allele-code true/false
    runs.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1398">APTNG-1398</a>] -
    Could not output probabilities file using --probabilities-output true.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1399">APTNG-1399</a>] -
    Stange errors if I remove one bi-allelic snp from input snp-specific prior
    file.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1401">APTNG-1401</a>] -
    Call for some male samples is not in the array of possible calls from
    call-code table(it was fixed, but I see it again).</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1402">APTNG-1402</a>] -
    Output for freqFlag=True has differences between apt2 and gold data.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1403">APTNG-1403</a>] -
    Error when running axiom cn app on mac OS from summary file created with
    --process-multi-alleles true(2000 samples run).</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1404">APTNG-1404</a>] -
    Differences between java and apt2 for multi-allele-pairwise-node:brlmmp-MS
    0.85.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1405">APTNG-1405</a>] -
    Differences between java and apt2 for
    --multi-allele-pairwise-node:brlmmp-CSepPen 0.1&amp;
    --multi-allele-pairwise-node:brlmmp-CSepThr 4 \.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1406">APTNG-1406</a>] -
    Unexpected error and run interruption when using posteriors files as as
    input.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1407">APTNG-1407</a>] -
    Cannot run eureka design with apt1.20 executable.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1408">APTNG-1408</a>] -
    apt-package-util: add description for the --create-multiallelic-blob option
    to the help output.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1409">APTNG-1409</a>] -
    apt-package-util not fails when inconsistent array type detected in tsv
    file header.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1410">APTNG-1410</a>] -
    Probeset count reporting in the blob index file is not correct.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1412">APTNG-1412</a>] -
    Differences in hetRate, totalhetRate, homRate, and total_homRate in the
    report file with allele-code =False.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1413">APTNG-1413</a>] -
    Call table in the call file under condition '--use-copynumber-call-codes'
    false is wrong.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1415">APTNG-1415</a>] -
    If snp is multi-allelic, then l2r and strength in th blob should be
  '-1'.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1417">APTNG-1417</a>] -
    bamboo build version of apt-genotype-eureka crashes, but local checkout
    doesn't.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1418">APTNG-1418</a>] -
    Inconsistency between calls and call table when blob is created from axiom
    output with copynumber-allele code set to False.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1419">APTNG-1419</a>] -
    Segmentation fault under condirion --use-copynumber-call-codes false.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1420">APTNG-1420</a>] -
    apt-genotype-eureka: Several fields in eureka.report.txt were not being
    populated correctly.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1424">APTNG-1424</a>] -
    Apt2_summary_genotype: columns in report file: 'call_rate',
    'total_call_rate', 'het_rate', 'total_het_rate', hom_rate',
    'total_hom_rate' does not have the values.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1425">APTNG-1425</a>] -
    Default blob version for eureka must be 1.0.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1426">APTNG-1426</a>] -
    Application creates output with blob-version 2.0 even input contains
    blob-version 1.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1427">APTNG-1427</a>] -
    Add ability to exclude a set of probes from the OncoScan MAPD and ndSNPQC
    calculations.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1428">APTNG-1428</a>] -
    Data in the blob and call table is inconsistent for ma array blob version
    2.0(default for ma).</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1429">APTNG-1429</a>] -
    apt-package-util: different call tables in .CHP.index.txt when
    --create-multiallelic-blob True and False used for legacy data.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1430">APTNG-1430</a>] -
    apt-package-util: different call tables in .CHP.index.txt when the same
    input data the same, but created in AxAS 1.1 and AxAS 2.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1431">APTNG-1431</a>] -
    Current version of the application does not report error is posterior for
    the snp could not be found in the input.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1432">APTNG-1432</a>] -
    apt-package-util: unable to create batch folder based on multiallele
    genotyping data (Axiom_Mayo_ECS.v2 library).</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1433">APTNG-1433</a>] -
    Index file call table is not correct under conditions:
    --blob-format-version=2.0 ^ --create-multiallelic-blob false ^.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1450">APTNG-1450</a>] -
    Plink exported incorrectly when only text file is provide for call codes
    like "A", "B".</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1453">APTNG-1453</a>] -
    Segmentation fault only on Linux system running
  apt-copynumber-axiom-ssa.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1455">APTNG-1455</a>] -
    Run-time exception for Carrier design .</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1325">APTNG-1325</a>] -
    apt-dmet-translation: inconsistent behavior of --base-name-prefix default
    value parameter.</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1347">APT-1347</a>]
    - apt-dmet-translation: Disable --calls-file option for translation tool
    now that we have full support for --batch-folder.</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1355">APT-1355</a>]
    - DMET Translation doesn't properly handle NoCalls for CN=1 samples.</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1363">APT-1363</a>]
    - apt-dmet-translation: Add Sample Attributes option fails if sample
    filenames have periods in them.</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1365">APT-1365</a>]
    - apt-dmet-translation: The following special characters are problematic in
    filename on Mac-OS platform: ` and !.</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1366">APT-1366</a>]
    - apt-dmet-translation: BlockReader failed when sample names in header
    section contains these special characters: = and space.</li>
</ul>

<div class="contents">
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.19.0 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Mon August 22, 2016</dd>
</dl>

<p>The 1.19.0 release of APT supports the Axiom Analysis Suite 2.0 release.
Highlights include:</p>
<ul>
  <li>New apt-copynumber-axiom-ssa engine reports copy number states in
    pre-defined regions for Axiom library packages that support it.</li>
  <li>Added limited ability to genotype samples in copy number variable
    regions. If samples are grouped for a probeset into ZeroCN, haploid, or
    diploid groups, each group is genotyped by apt-genotype-axiom using the
    appropriate model. Refer to optional arguments use-copynumber-call-codes
    and copynumber-probeset-calls-file.</li>
  <li>Ps-metrics and ps-classification support the new copy-number aware
    genotype call codes.</li>
  <li>Apt-format-results has made some changes to output formats.</li>
  <li>Apt-dmet-translation accepts input from the Axiom Analysis Suite batch
    folder.</li>
  <li>Note that creating CHP files using apt-genotype-axiom is no longer
    supported. The option --chp-a5-output has been deprecated to a developer
    option, and may be removed in a future release.</li>
</ul>

<h3>New Features and Improvements </h3>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-151">APTNG-151</a>] -
    Logging Enhancements.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-911">APTNG-911</a>] -
    AxiomDMET: Add self-describing allele table to genotyping blob index.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-940">APTNG-940</a>] -
    AxiomDMET: Integrate updated blob writer/reader APIs into SNP Polisher
    engines.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1030">APTNG-1030</a>] -
    apt-genotype-axiom: Embed a dynamic self-describing table for allele codes
    in header section of AxiomGT1.calls.txt.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1036">APTNG-1036</a>] -
    Add --inbred-file feature to apt2-summary-genotype.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1051">APTNG-1051</a>] -
    Inbred genotype at command line.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1052">APTNG-1052</a>] -
    apt-format-result: Must support SNP6-style chp files as inputs. (leave
    calls-file option intact).</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1054">APTNG-1054</a>] -
    apt-package-util: Add support for convert SNP 6 CHP files into
    CallsFile.txt for HLA export tool.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1073">APTNG-1073</a>] -
    add DoubleVectorVector class to CommonBboardData_t types.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1054">APTNG-1095</a>] -
    apt-format-result: Add option to convert PLINK format to VCF using the
    annotation file.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1100">APTNG-1100</a>] -
    APT2-Core: Create generator for multiallelic call codes with copy-number
    awareness.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1107">APTNG-1107</a>] -
    in ProbesetsSummaryGenotype object, first partition ProbeSetVector into
    bi-allelic and multi-allelic sections, then define chunks.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1114">APTNG-1114</a>] -
    add --@multi-allele-gentoype option to main node.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1121">APTNG-1121</a>] -
    add developer parameter --@process-multi-alleles.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1130">APTNG-1130</a>] -
    add option to provide genotyping node with probeset/sample copynumber calls
    and use them when genotyping.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1133">APTNG-1133</a>] -
    add option to convert brlmmp calls to new copynumber-aware call codes.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1136">APTNG-1136</a>] -
    create apt2-zerocn-summaries-extract application.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1154">APTNG-1154</a>] -
    APT2-Core: Improve the summary logging system for APT2 framework.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1155">APTNG-1155</a>] -
    apt-format-result: Convert haploid calls to either AA or BB for VCF export
    (for Axiom DMET 2.0).</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1168">APTNG-1168</a>] -
    remove --chip-type requirement from apt2-cdf-to-spf.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1169">APTNG-1169</a>] -
    apt-blob-extract: Check validity of all call codes and make sure they exist
    in self-describing table.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1204">APTNG-1204</a>] -
    apt-genotype-axiom to write summary.a5.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1231">APTNG-1231</a>] -
    deprecate chp-a5-output.</li>
</ul>

<h3>Bug Fixes </h3>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-287">APTNG-287</a>] -
    Resume does not work after pre-ascat serialization file was converted to
    ascii and converted back without any modification.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-373">APTNG-373</a>] -
    Parameter parsing does not handle parameters with "no-" prefix
  correctly.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-572">APTNG-572</a>] -
    APT2 Cyto --@report-probe-marker-usage parameter causes error (need fix for
    flags testing).</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-877">APTNG-877</a>] -
    probeset object should have a 'char m_ProbeSetType' member, instead of
    hijacking position member.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-931">APTNG-931</a>] -
    apt-format-result - Warning messages that we are using.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-947">APTNG-947</a>] -
    Apt release: apt-package-util has diff revision number between application
    log and '--help' option. Should be fixed in the next release.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-960">APTNG-960</a>] - ND
    algorithm does not appear to be shutting off when &gt;1.5 million markers
    are detected.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-982">APTNG-982</a>] -
    ERROR setErr called and m_abort_on_err is set. Field index is out of
  bounds.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1026">APTNG-1026</a>] -
    apt-genotype-axiom: Need to send progress back to UI when scanning and
    checking cel files. (doCheckParam).</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1048">APTNG-1048</a>] -
    apt-genotype-axiom reported progress can exceed 100%.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1055">APTNG-1055</a>] -
    Segmentation fault (core dumped).</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1061">APTNG-1061</a>] -
    apt-genotype-axiom: Need send back to UI progress updates if checking
    params is slow.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1063">APTNG-1063</a>] -
    apt-genotype-axiom: Fs::fileSize is not computing size of file (&gt;4GB)
    correctly on Windows.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1066">APTNG-1066</a>] -
    Apt release: apt-package-util has diff revision number between application
    log and '--help' option. Should be fixed in the next release.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1072">APTNG-1072</a>] -
    cdf.html page does not have new formats of the file.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1081">APTNG-1081</a>] -
    IndexWriter works incorrect in update mode.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1085">APTNG-1085</a>] -
    setErr called and m_abort_on_err is set. Field index is out of bounds: 0 if
    cnbins file is empty.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1086">APTNG-1086</a>] -
    Application accept allele code in the summary that is not A or B.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1087">APTNG-1087</a>] -
    Segmentation fault if summary file cel file name is different from report
    file or one cel name from summary file is missing from report file.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1090">APTNG-1090</a>] -
    Segmentation fault if report file has a cel with diff name compare with
    summary file.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1091">APTNG-1091</a>] -
    No any errors or warning in log file if report file does not have a cel
    from summary file.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1092">APTNG-1092</a>] -
    No error or warning in the application if cnbins file has the following
    header line: Key_test Value.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1093">APTNG-1093</a>] -
    No errors in the log file if input summary file does not have 'probeset_id'
    in the header line.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1096">APTNG-1096</a>] -
    fix bamboo failure in axiom serialize-resume test.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1103">APTNG-1103</a>] -
    ERROR\tMainNode::standardMain(): Unhandled std::exception
    (vector::_M_range_check).</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1109">APTNG-1109</a>] -
    APT2-Core: Fix error handling for reading blob index file..</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1110">APTNG-1110</a>] -
    APT2-Core: BlobReader: Assign both blob and index file names before
    attempting to open batch folder for accessing.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1111">APTNG-1111</a>] -
    APT2-Core: Exception occurred while loading buffer data for blob
  reader.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1112">APTNG-1112</a>] -
    apt-format-result returns empty output files in case if
    snp-identifier-column parameter set to Affy_SNP_ID, Chr_id, or
  dbSNP_RS_ID.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1118">APTNG-1118</a>] -
    BlobWriter (IndexWriter) update Index file in batch folder incorectly.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1123">APTNG-1123</a>] -
    occaisional crash when deleting Dset::File for ReportData.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1126">APTNG-1126</a>] -
    apt-genotype-axiom returns calculating probe set chunks error for some
    datasets.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1128">APTNG-1128</a>] -
    apt-genotype-axiom output AxiomGT1.report.txt file doesn't match Gold
  Data.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1132">APTNG-1132</a>] -
    Index file and Blob file have a unsynchronized call code values after
    creation in batch folder.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1135">APTNG-1135</a>] -
    Dummy calls exported and must be able to handle haploid calls.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1138">APTNG-1138</a>] -
    LigationBases_ps additional L2R adjustment does not have any effect on the
    output results..</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1140">APTNG-1140</a>] -
    Creation of blob file should allow for possibility of serialize-resume run
    of apt-genotype-axiom.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1142">APTNG-1142</a>] -
    Export Genotyping Data txt Numeric Call Codes cannot be done: unable to set
    non-numeric field as int..</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1144">APTNG-1144</a>] -
    apt-format-result: Transpose plink format not exporting correctly using
    multiallelic blob.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1149">APTNG-1149</a>] -
    apt-format-result: Incorrect warnings regarding to mismatches between
    ref/alt and A/B alleles.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1151">APTNG-1151</a>] -
    apt-format-result could not recognise new calls, i.e. 4 or 5.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1153">APTNG-1153</a>] -
    apt-genotype-axiom: call codes in the AxiomGT1.calls.txt and in the blob
    doesn't match.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1156">APTNG-1156</a>] -
    For snp with chr='MT' and gender='female' call still is '-1', but should be
    -3.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1157">APTNG-1157</a>] -
    apt-format-result crashes when --export-txt-file parameter is used with
    --calls-file and --snp-list-file.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1162">APTNG-1162</a>] -
    apt-format-result: duplicate rows are missing when PLINKT format converts
    to VCF.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1164">APTNG-1164</a>] -
    Numeric difference in the report files between 1.19.0 and current release
    for case when new allele code is false, no special_snps and no cn file.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1166">APTNG-1166</a>] -
    L2r output after plate correction possible is not correct.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1167">APTNG-1167</a>] -
    Haploid ALT calls written incorrectly to VCF.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1172">APTNG-1172</a>] -
    If parameter file does have cn file input, application still creates old
    allele code behavior.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1175">APTNG-1175</a>] -
    if --blob-format-version=2.0, and --use-copynumber-call-codes=false, then
    blob should get brlmmp call codes instead of CHP call codes.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1176">APTNG-1176</a>] -
    Strange handlig of the MT snps with overwite option: false, true.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1178">APTNG-1178</a>] -
    Different header in the blob index files for cn default test and test
    contains cn+--use-copynumber-call-codes true +
    --copynumber-override-special-snps false.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1180">APTNG-1180</a>] -
    apt-format-result: wrong values for samples in VCF file when annotation
    database has single slash as Alt_Allele delimiter.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1183">APTNG-1183</a>] -
    apt-format-result: VCF file contains wrong call values in case if blob
    version one is used as an input data.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1184">APTNG-1184</a>] -
    apt-format-result: Text export is using GTC call codes {6,7,8,11,12}
    instead of translated call codes.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1186">APTNG-1186</a>] -
    apt-format-result: wrong application name in the help output.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1189">APTNG-1189</a>] -
    cn file reader has warning that causes seg fault.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1192">APTNG-1192</a>] -
    apt-format-result: POS field in VCF could contains negative value for some
    probesets.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1193">APTNG-1193</a>] -
    No any warnings or errors from application if 1 row in the reference file
    parameter dataset has been altered.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1194">APTNG-1194</a>] -
    No any warnings or errors from application if second row in the reference
    file parameter dataset has been altered.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1196">APTNG-1196</a>] -
    When writing hdf5 files with a cursor, the _&amp;keyval datasets are not
    being written.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1197">APTNG-1197</a>] -
    No errors or warnings if 'order, parameter has been altered.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1198">APTNG-1198</a>] -
    No errors or warnings if 'bin-count', parameter has been altered.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1199">APTNG-1199</a>] -
    No errors or warnings if 'SuperGC, parameter has been altered.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1200">APTNG-1200</a>] -
    No errors or warnings if 'mediansig' parameter has been altered.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1201">APTNG-1201</a>] -
    Shall we also use 0 to instead 2147483648 in SNP Identifier for PLINK
    format.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1203">APTNG-1203</a>] -
    VCF export crashes if reading AxiomCN.cnregioncalls.txt with "NoCall" for
    CN state.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1206">APTNG-1206</a>] -
    crash in windows when using fewer cels than samples in cnpscalls.txt.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1210">APTNG-1210</a>] -
    apt-format-result: TXT file has wrong column names when Contrast and Size
    has been exported from the CES batch folder.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1212">APTNG-1212</a>] -
    Exception when using '--summary-file' for ascii summary file input..</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1213">APTNG-1213</a>] -
    The same warning on two different rows without any reason. '\n' is
  used.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1214">APTNG-1214</a>] -
    App does not report anything if bin file region is empty.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1215">APTNG-1215</a>] -
    App does not report anything if bin file lower threshold is empty.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1216">APTNG-1216</a>] -
    App does not report anything if bin file upper threshold is empty.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1217">APTNG-1217</a>] -
    App does not report anything if bin file upper threshold is less than lower
    threshold.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1218">APTNG-1218</a>] -
    App does not report anything if duplication of the data in the bin
  file.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1219">APTNG-1219</a>] -
    App does not report anything if bin file has region thresholds
  overlapping.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1221">APTNG-1221</a>] -
    If controls file is empty, then app stops executuib with and error, but it
    should proceed with the warning.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1222">APTNG-1222</a>] -
    crash in windows when cleaning up GenotypeProbeMarkerUsage object.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1224">APTNG-1224</a>] -
    If there is a gap in the bin file between cn regions, then application does
    not report correct CN state.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1226">APTNG-1226</a>] -
    ERROR setErr called and m_abort_on_err is set. Dataset::getDataPtr(-1,0)
    out of range! (0--33334).</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1227">APTNG-1227</a>] -
    ERROR MainNode::standardMain(): Unhandled Exception! (Unknown) in
    apt2-dset-compare.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1229">APTNG-1229</a>] -
    App reports the message that does not explain anything: ERROR '' not found
    in report file..</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1233">APTNG-1233</a>] -
    Change use-copynumber-call-codes 'false' to 'true' does not change call
    file if cn=1 for all entries in the file.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1241">APTNG-1241</a>] -
    Application does report an error without a reason: Could not read data in
    the summary file.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1244">APTNG-1244</a>] -
    apt-genotype-axiom errors out when it can't find a prior that isn't
    supposed to be there anyway.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1245">APTNG-1245</a>] -
    'batch-folder' should be '@batch-folder' in all the getParam() calls.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1248">APTNG-1248</a>] -
    Blob index file does not have allele-code table if
    use-copynumber-call-codes false.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1250">APTNG-1250</a>] -
    No enough information in the message if ReportedPS table is not fomatted
    correctly.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1255">APTNG-1255</a>] -
    An error message about empty control file was accidentally removed!.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1261">APTNG-1261</a>] -
    Call files differences after 07/15 between apt1.19.0 and current
    release(revision 1523 is fine).</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1263">APTNG-1263</a>] -
    Unhandled Exception for ProbeSetSummarization.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1264">APTNG-1264</a>] -
    message is wrong when cn file has column count different from summary file
    for CN_GT.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1266">APTNG-1266</a>] -
    Version string is not correct - only for Windows os.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-1272">APTNG-1272</a>] -
    Run time exception when running 3000 samples with hdf5 summry file.</li>
</ul>

<div class="contents">
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.18.2 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Fri March 18, 2016</dd>
</dl>

<p>The 1.18.2 release of APT is branched from the 1.18.1 instead of trunk. This
is because the only necessary change from 1.18.1.</p>
<ul>
  <li>enabled the reading of new CDF version (APTv5 and GCv7)</li>
  <li>apt-genotype-axiom now handles probesets with wobble, and can gracefully
    ignore multi-allelic probesets (multi-allelic = more than 2 alleles)</li>
  <li>introducing apt-genotype-eureka</li>
</ul>

<h3>New Features and Improvements </h3>
<ul>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1283">APT-1283</a>]
    - modify apt-geno-qc (, apt-probeset-genotype, etc?) to accept new cdf
    format with probe_seq_id</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-948">APTNG-948</a>] -
    add feature to apt-genotype-axiom to ignore multi-allelic probesets and
    still properly process bi-allelic ones</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-974">APTNG-974</a>] -
    add functionality to artifact-reduction to group by probe_seq_id, if it
    exists, instead of by allele+wobble code</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-981">APTNG-981</a>] -
    Utilitiy for extracting calls used by apt-dmet-translate from Axiom
  package</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-740">APTNG-740</a>] -
    Wobble grouping integration into axiom-genotyping</li>
</ul>

<h3>Bug Fixes </h3>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-972">APTNG-972</a>] -
    Segmentation fault when probeset from hints does not exist in cdf</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-978">APTNG-978</a>] -
    enable sketch nodes to serialize output</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-979">APTNG-979</a>] -
    apt-genotype-axiom does not clean up temporary file in the output directory
    on Windows</li>
</ul>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.18.1 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Mon November 30, 2015</dd>
</dl>

<p>The 1.18.1 release of APT is a minor release in support of ChAS Dx 1.1 and
the Eureka platform. Highlights include:</p>
<ul>
  <li>Trio analysis (apt-copynumber-familial) adapted for use in ChAS Dx.</li>
  <li>Addition of genotyping analysis pipeline for Eureka
  (apt-genotype-eureka).</li>
  <li>Modifications to SNP Polisher and AAS package utilities in support of
    Eureka.</li>
</ul>

<h3>New Features and Improvements </h3>
<ul>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1278">APT-1278</a>]
    - apt-copynumber-familial - add option to turn off RelatednessTest table in
    output</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-689">APTNG-689</a>] -
    apt-genotype-eureka - change default value for brlmmp-CP parameter to be
    the same as apt1</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-869">APTNG-869</a>] -
    APT2 - Refactor LibraryFile Nodes</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-910">APTNG-910</a>] -
    apt-format-result, apt-extract-suitcase - add option to set name of data
    directory used by batch file</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-923">APTNG-923</a>] -
    apt-package-util: Support for eureka data</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-865">APTNG-865</a>] -
    apt-genotype-eureka - create app to normalize and genotype from a Eureka
    binning file</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-934">APTNG-934</a>] -
    Include executable and documentation for nibls binning script.</li>
</ul>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.18.0 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Mon August 10, 2015</dd>
</dl>

<p>The 1.18.0 release of APT supports the ChAS 3.1 and Axiom Analysis Suite 1.1
releases. Highlights include:</p>
<ul>
  <li>Support for manual recentering in OncoScan single-sample analysis
    (apt-copynumber-onco-ssa).</li>
  <li>Various improvements to the TuScan algorithm when reporting non-integer
    copynumbers.</li>
  <li>New wavelet shrinkage algorithm in apt-copynumber-cyto-ssa.</li>
  <li>Support for new Agatha folder format to replace suitcase files in
    apt-genotype-axiom, SNP Polisher and related applications.</li>
</ul>

<h3>New Features and Improvements </h3>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-619">APTNG-619</a>] -
    APT2 - Added total time summary for each node to log file</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-751">APTNG-751</a>] -
    APT2 - Switched to C++11.</li>
</ul>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-757">APTNG-757</a>] -
    apt-suitcase-extract: Add pHW and MAF columns from
    Apt_snp_summary_Report.txt to Ps.performance.txt</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-758">APTNG-758</a>] -
    apt-suitcase-extract: Improved progress meter</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-748">APTNG-748</a>] -
    apt-suitcase-extract: Add Suitcase Exploding</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-786">APTNG-786</a>] -
    apt-package-util: Support for Agatha 1.1 folder structure</li>
</ul>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-746">APTNG-746</a>] -
    apt-genotype-axiom: Replace suitcase file with regular folder</li>
</ul>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-813">APTNG-813</a>] -
    apt-copynumber-cyto-ssa: Wavelet Shrinkage v2 algorithm improvements</li>
</ul>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-753">APTNG-753</a>] -
    apt2-dset-util: add option to dump names of objects in file</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-762">APTNG-762</a>] -
    apt2-dset-util: add option to dump GDH to stdout</li>
</ul>
<ul>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-419">OSC-419</a>] -
    apt-copynumber-onco-ssa: Add manual recentering to TuScan</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-413">OSC-413</a>] -
    apt-copynumber-onco-ssa: Implement improved version of wavelet shrinkage
    algorithm</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-423">OSC-423</a>] -
    apt-copynumber-onco-ssa: Update default values to match current recommended
    defaults</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-422">OSC-422</a>] -
    apt-copynumber-onoc-ssa: Set default value for --report-int-ploidy to
  false</li>
</ul>
<ul>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-418">OSC-418</a>] -
    apt-copynumber-onco-som-ssa: Remove SOM probe count restriction</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-424">OSC-424</a>] -
    apt-copynumber-onco-som-ssa: Add minimum median mutscore to
    apt-copynumber-onco-som-ssa</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-417">OSC-417</a>] -
    apt-copynumber-onco-som-ssa: Add new adjustments for Somatic Mutation
  scores</li>
</ul>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/SNPOL-615">SNPOL-615</a>] -
    otv-caller: Improve progress indications in OTV caller</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/SNPOL-616">SNPOL-616</a>] -
    ps-metrics: Improve progress indications in ps-metrics</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/SNPOL-617">SNPOL-617</a>] -
    ps-classification-supplemental: Improve progress indication in
    ps-classification-supplemental</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/SNPOL-593">SNPOL-593</a>] -
    otv-caller, ps-classification-supplemental: Replace suitcase file support
    with regular folders</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/SNPOL-623">SNPOL-623</a>] -
    ps-metrics, ps-classification: Refactor to use APT2 framework</li>
</ul>
<ul>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1271">APT-1271</a>]
    - apt-probeset-summarize: SST - GainLock added</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1272">APT-1272</a>]
    - apt-probeset-summarize: DABG probe aggregation improvement</li>
</ul>

<h3>Bug Fixes </h3>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-517">APTNG-517</a>] -
    apt-copynumber-cyto-ssa: Added default values and ranges for parameters
    agr-denominator-source and agr-denominator-percentile</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-693">APTNG-693</a>] -
    apt-genotype-axiom: fixed memory issues when running greater than 1000
    samples</li>
</ul>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-729">APTNG-729</a>] -
    apt-genotype-axiom: File Existence Check error for snp-priors-input-file
    and sketch-target-input-file when using --analysis-files-path</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-814">APTNG-814</a>] -
    apt-genotype-axiom: Change output of error regarding insufficient disk
    space from Log to Error</li>
</ul>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-755">APTNG-755</a>] -
    apt-format-result: Need to output warning for empty ref/alt alleles in
    annotation file.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-760">APTNG-760</a>] -
    apt-format-result: Crashed when index/data files are not in expected batch
    subfolder, e.g. AxiomAnalysisSuiteData.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-774">APTNG-774</a>] -
    apt-suitcase-extract: Error occurs if output dir is a drive root</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-800">APTNG-800</a>] -
    apt-suitcase-extract: unexpectedly stop working when UNC path is used</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-795">APTNG-795</a>] -
    apt-package-util: unexpected folder structure created on the root of the
    local drive in case if UNC path has been used as a value for --batch-folder
    parameter</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-818">APTNG-818</a>] -
    apt-package-util: Log file summary does not count warnings in the file.</li>
</ul>
<ul>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-397">OSC-397</a>] -
    apt-copynumber-onco-ssa: Signature SNP markers sometimes end up in
    different TuScan segments</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-398">OSC-398</a>] -
    apt-copynumber-onco-ssa: TuScan - a tumor NaN BAF marker can incorrectly
    acquire a HET flag from the normal step</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-399">OSC-399</a>] -
    apt-copynumber-onco-ssa: TuScan - centromere breaks not added to alt mode
    segments</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-402">OSC-402</a>] -
    apt-copynumber-onco-ssa: TuScan PAR breaks do not work properly when there
    are no markers on a chromosome</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-403">OSC-403</a>] -
    apt-copynumber-onc-ssa: Centromere breaks do not work correctly when the
    segmentation is trivial</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-405">OSC-405</a>] -
    apt-copynumber-onc-ssa: Allele peaks bandwidth is computed incorrectly for
    sparse arrays</li>
</ul>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.17.0 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Fri March 6, 2015</dd>
</dl>

<p>The 1.17.0 release of APT includes the following highlights:</p>
<ul>
  <li>New applications <em>apt-copynumber-cyto-ssa</em> and
    <em>apt-copynumber-cyto-ref</em> for single-sample analysis and reference
    model creation for CytoScan family of arrays. apt-copynumber-cyto-ssa
    includes support for Normal Diploid detection and automatic
  recentering.</li>
  <li>New <em>apt-genotype-axiom</em> application implementing recommended
    genotyping analysis pipeline for Axiom arrays. Supports new features such
    as SNP-specific parameters, and improved performance for large sample
  sets.</li>
  <li>APT ports of the SNP Polisher applications <em>otv-caller</em> and
    <em>ps-classification-supplemental</em> providing improved performance over
    R equivalents.</li>
  <li>Inclusion of <em>apt-copynumber-familial</em> application, implementing
    trio-analysis on CytoScan family of arrays.</li>
  <li>Support for GC Count Normalization and Signal Space Transformation
    algorithms in apt-probeset-summarize and apt-cel-transformer
  applications.</li>
  <li>Various enhancecments and bug fixes (see below).</li>
</ul>

<h4>Notes</h4>
<ol type="1">
  <li>The applications apt-copynumber-cyto-ssa and apt-copynumber-cyto-ref
    replace the previous apt-copynumber-cyto application for CytoScan family of
    arrays, but do not support analysis of 2.7M arrays. For 2.7M it is
    recommended that you use the 1.16.1 version of apt-copynumber-cyto.</li>
  <li>Due to a change in the underlying HDF5 format, reference model files
    generated with the previous apt-copynumber-cyto application can not be read
    by apt-copynumber-cyto-ssa, and reference model files generated with
    apt-copynumber-cyto-ref can not be used with earlier versions of
    apt-copynumber-cyto. If using a custom reference model file, you will need
    to regenerate the reference model file in order to use the new
    single-sample analysis application.</li>
</ol>

<h4>New features and Improvements </h4>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-359">APTNG-359</a>] -
    apt-genotype-axiom: create APT2 Axiom genotyping pipeline</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-387">APTNG-387</a>] -
    apt-genotype-axiom: add support for hints file.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-388">APTNG-388</a>] -
    apt-genotype-axiom: add support for inbred penalty.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-389">APTNG-389</a>] -
    apt-genotype-axiom: add support for gender files.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-391">APTNG-391</a>] -
    apt-genotype-axiom: new suitcase output format.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-392">APTNG-392</a>] -
    apt-genotype-axiom: add VCF format output.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-393">APTNG-393</a>] -
    apt-genotype-axiom: add decoded base calls format.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-394">APTNG-394</a>] -
    apt-genotype-axiom: add forward strand base calls.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-395">APTNG-395</a>] -
    apt-genotype-axiom: add support for snp-specific analysis parameters.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-382">APTNG-382</a>] -
    apt-genotype-axiom: Improve performance for large sample sets</li>
</ul>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-404">APTNG-404</a>] -
    apt-param-convert: Implement utility for converting argument xml files</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-453">APTNG-453</a>] -
    axiom_param_conversion.txt - parameters for conversion between
    apt-probeset-genotype and apt-genotype-axiom.</li>
</ul>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-330">APTNG-330</a>] -
    Refactor apt2-dset-util</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-428">APTNG-428</a>] -
    APT2: Add multi-threading support.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-529">APTNG-529</a>] -
    APT2: Add ability to generate HTML documentation for workflow and node
    classes</li>
</ul>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/SNPOL-374">SNPOL-374</a>] -
    Ps_Classification: update the way conversion is called for the different
    species</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/SNPOL-394">SNPOL-394</a>] -
    otv-caller: EM algorithm corrections</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/SNPOL-414">SNPOL-414</a>] -
    otv-caller: option to update output files</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/SNPOL-423">SNPOL-423</a>] -
    Ps_Classification: Updated criteria for selecting best probeset for
    hemizygous</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/SNPOL-424">SNPOL-424</a>] -
    Ps_Classification: Update recommended/coverted probeset list</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/SNPOL-425">SNPOL-425</a>] -
    otv-caller: option to output files with only OTV SNPs</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/SNPOL-378">SNPOL-378</a>] -
    ps-classification-supplemental: defaults nMinorHom=0 and HetvMAF=10</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/SNPOL-386">SNPOL-386</a>] -
    ps-classification-supplemental node project was added to the solution</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/SNPOL-429">SNPOL-429</a>] -
    otv-caller: allow otv-caller to read inputs directly from "suitcase"
  file</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/SNPOL-430">SNPOL-430</a>] -
    ps-classification-supplemental: Allow ps-classification-supplemental to
    read inputs directly from "suitcase" file</li>
</ul>
<ul>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1198">APT-1198</a>]
    - apt-probeset-summarize: Add gcc1 (GC Count Correction) algorithm</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1222">APT-1222</a>]
    - apt-probeset-summarize: Add gc-sst-rma-sketch command line option</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1224">APT-1224</a>]
    - apt-cel-transformer: Add support for QCC files</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1225">APT-1225</a>]
    - apt-cel-transformer: Write appropriate header information when creating
    transformed cel files</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1227">APT-1227</a>]
    - apt-cel-transformer, apt-probeset-summarize: Don't allow reprocessingcel
    files with GC4 and SST</li>
</ul>

<h4>Bug fixes </h4>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-465">APTNG-465</a>] -
    APT2: Displaying default values for all optional parameters in help.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-378">APTNG-378</a>] -
    APT2: Garbled numbers (extremely long negative integer) in error message
    when lower limit exceeded for input parameter in APT2 applications.</li>
</ul>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-519">APTNG-519</a>] -
    apt-genotype-axiom: validating file format and content of the input
  files.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-632">APTNG-632</a>] -
    apt-genotype-axiom: Add plate barcode to the report file</li>
</ul>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/SNPOL-436">SNPOL-436</a>] -
    Ps_Classification: “recommended.ps” file is not matching between R and
    APT when output.recommended=TRUE</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/SNPOL-476">SNPOL-476</a>] -
    Ps_Metrics: Handling ":1" probesets in metrics</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/SNPOL-547">SNPOL-547</a>] -
    Ps_Classification: num-minor-allele-cutoff min and max should be from 1 to
    10,0000 and give error if set to 0A</li>
</ul>
<ul>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1228">APT-1228</a>]
    - apt-summary-genotype: --write-models does not consistently produce a
    snp-posteriors file</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1232">APT-1232</a>]
    - apt-copynumber-familial: Trio analysis add support for 750k array and
    other CytoScan arrays</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1238">APT-1238</a>]
    - apt-copynumber-familial: Support for CYCHP files generated by
    apt-copynumber-cyto-ssa</li>
</ul>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.16.1 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Fri June 23, 2014</dd>
</dl>

<p>The 1.16.1 release of APT addresses performance issues with the
Ps_Classification application when used with an optional ps2snp mapping
file.</p>

<p>Details:</p>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/SNPOL-357">SNPOL-357</a>] -
    Ps_Classification performance improvementswhen running with ps2snp-file
    option</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/SNPOL-359">SNPOL-359</a>] -
    Ps_Classification - added warning if Polyploid=TRUE and HomRO=TRUE</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/SNPOL-358">SNPOL-358</a>] -
    Ps_Classification - changed column snpid to affy_snp_id</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1212">APT-1212</a>]
    - Added new vignette for SNP6 arrays using Axiom analysis</li>
</ul>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.16.0 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Fri May 9, 2014</dd>
</dl>

<p>The 1.16.0 release of APT includes the following highlights:</p>
<ul>
  <li>Support for automatic recentering for samples with ploidy not equal to 2
    in the OncoScan analysis pipeline.</li>
  <li>Support for probabilities file output in apt-probeset-genotype.</li>
  <li>Addition of the applications Ps_Classification and Ps_Metrics (ported
    from the SNP Polisher R package).</li>
  <li>Addition of the apt2-dset-util applications for converting between OSCHP
    and other HDF5 based file formats and text files.</li>
  <li>Various performance and other improvements, minor features and minor bug
    fixes.</li>
</ul>

<p>Details below:</p>

<h4>New features and bug fixes </h4>
<ul>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1114">APT-1114</a>]
    - Minor documentation corrections from 1.15.0.</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-414">APT-414</a>] -
    Updated documentation for pgf, clf and bgp files to clarify row major
    instead of column major is the implementation.</li>
</ul>
<ul>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1038">APT-1038</a>]
    - apt-copynumber-cyto - no warnings or errors when running Ref Model file
    creation with missing probe_tab file.</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1184">APT-1184</a>]
    - apt-copynumber-cyto - mosaicism analysis could create overlapping
    MosaicGain segments on very noisy samples.</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1187">APT-1187</a>]
    - apt-copynumber-cyto - mosaicism algorithm creating no-call segments with
    non-neutral copynumber and mixture on very noisy samples.</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1186">APT-1186</a>]
    - apt-copynumber-cyto - Added option to remove Mosaic Segment Calculation
    from the Y chomosome only for CytoScan 750k and HD arrays (default
  true).</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1174">APT-1174</a>]
    - apt-probeset-genotype - added new output file option:
    AxiomGT1.probabilities.txt.</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1125">APT-1125</a>]
    - apt-probeset-summarize added options to turn off probe caching for
    platforms with many duplicate probes (which caused performance issues with
    EC2).</li>
</ul>
<ul>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-334">OSC-334</a>] -
    apt-copynumber-onco-ssa - Fixed short, low marker-count LOH segments being
    reported on certain samples in regions with small number of markers with
    large separation.</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-335">OSC-335</a>] -
    apt-copynumber-onco-ssa - Fixed breakpoints for PAR region in chromosome
  X.</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-369">OSC-369</a>] -
    apt-copynunber-onco-ssa - Fixed non-default calibration parameters being
    ignored by LOH and Cychp Reporter nodes.</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-359">OSC-359</a>] -
    apt-copynumber-onco-ssa - Automated recentering/adjustment of log2 rations
    if rounded ploidy not equal to 2.</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-373">OSC-373</a>] -
    apt-copynumber-onco-ssa - Added reliability score for TuScan algorithm.</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-371">OSC-371</a>] -
    apt-copynumber-onco-ssa - Added breakpoints at centromeres in TuScan
    segmenter.</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/OSC-393">OSC-393</a>] -
    apt-copynumber-onco-ssa - Added tuscan-initial-ploidy to OSCHP header and
    Report file</li>
</ul>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-365">APTNG-365</a>] -
    APT2 framework - Removed some blank lines in console output</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-367">APTNG-367</a>] -
    APT2 framework - Updated help text for --arg-file option</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-374">APTNG-374</a>] -
    APT2 framework - Added checks for badly formatted int and float
  parameters.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-368">APTNG-368</a>] -
    APT2 framework - Indicate required parameters in help text.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-369">APTNG-369</a>] -
    APT2 framework - Set sockets parameters as (hidden) developer
  parameters.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-370">APTNG-370</a>] -
    APT2 framework - Improved error messages for unrecognized parameters.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-353">APTNG-353</a>] -
    APT2 framework - Applications output usage info if called with no or bad
    parameters.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-337">APTNG-337</a>] -
    APT2 framework - Dropped unnecessary _&amp;size columns from HDF5
  output.</li>
</ul>
<ul>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/APTNG-354">APTNG-354</a>] -
    apt2-dset-util - Added to release.</li>
  <li>[<a
    href="http://jira.affymetrix.com:8080/browse/SNPOL-258">SNPOL-258</a>] -
    Ps_Metrics and Ps_Classification added to release.</li>
</ul>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.15.2 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Fri Oct 30, 2013</dd>
</dl>

<p>This release adds support for the OncoScan array platforms through the
addition of the APT2 application apt-copynumber-onco-ssa,
apt-copynumber-onco-ref, apt-copynumber-onco-som-ssa and
apt-copynumber-onco-som-ref. (Note only the binrary distributions include the
new applications: a separte source code distribution will be made available in
order to compile the applications from source.)</p>

<h4>New Features and Bug Fixes</h4>
<ul>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1177">APT-1177</a>]
    - Added OncoScan applications.</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1125">APT-1125</a>]
    - Added option to turn on duplicate probe caching manually</li>
</ul>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.15.1 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Mon Feb 25, 2013</dd>
</dl>

<p>This is a minor release addressing the following features:</p>
<ul>
  <li>Adding mosaicism analysis to the apt-copynumber-cyto pipeline for the
    CytoScan family of arrays.</li>
  <li>Adding saturation reports to the apt-geno-qc application.</li>
</ul>

<h4>New Features and Bug Fixes</h4>
<ul>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1002">APT-1002</a>]
    - Mosaicism for Cytoscan pipeline</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1132">APT-1132</a>]
    - Added saturation report to apt-geno-qc and apt-cel-extract</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1131">APT-1131</a>]
    - Use a background value of 0 for probes that have no bg bins when using
    pm-gcbg.</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-1139">APT-1139</a>]
    - Add project file for apt-copynumber-gender for inclusion in Windows
    releases</li>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-414">APT-414</a>] -
    Updated documentation for CLF file format to address row_major vs col_major
    issue.</li>
</ul>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.15.0 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Thu May 24, 2012</dd>
</dl>

<p>This release features a number of improvements and changes. Highlights
include:</p>
<ul>
  <li>Migrating Windows binary releases to Visual Studio 2010</li>
  <li>Support in Windows for file paths longer than 240 characters</li>
  <li>Various improvements in Linux and OS X build procedures</li>
  <li>Reorganization of source code repositories and regression data to make it
    easier for users to compile and test APT code on different platforms</li>
  <li>Improvements in internal testing and reliability</li>
  <li>Various performance improvements and new features</li>
</ul>

<h4>Bug Fixes</h4>
<ul>
  <li>General 
    <ul>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-932">APT-932</a>] -
        Fixed problemswith HDF5 libraries compiling from source on ubuntu</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-939">APT-939</a>] -
        SQLite files read locking caused fatal errors - implemented retry for
        locked data</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-943">APT-943</a>] -
        Added missing apt-summary-genotype HTML documentation</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-966">APT-966</a>] -
        Fixed bugs in and FsPath::mkdir including errIfExists being ignored</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-1072">APT-1072</a>] -
        Fixed issue with SVN version not being set in APT compiled from source
        code distribution</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-1080">APT-1080</a>] -
        Removed references to win32 from readme files</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-1081">APT-1081</a>] -
        Fixed makefile to allow running "make check" without prior make (to
        make it consistent with "make integration" and "make regression"
      targets</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-1087">APT-1087</a>] -
        Modified configure script to allow re-running to change regression data
        location</li>
    </ul>
  </li>
  <li>Testing 
    <ul>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-948">APT-948</a>] -
        Negative test failures in unit tests not being caught</li>
    </ul>
  </li>
  <li>Cyto 
    <ul>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-953">APT-953</a>] -
        apt-copynumber-cyto: Added check to the high-pass filter to process
        only those probesets with process_flag non-zero n0</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-958">APT-958</a>] -
        apt-copynumber-cyto: analysis-genotype was not run when no analysis
        steps included in command line, causing segmentation fault (intended
        behavior is to run all steps)</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-1031">APT-1031</a>] -
        apt-copynumber-cyto: Added indication in log file to show if "allelic
        difference modulation correction" is turned on</li>
    </ul>
  </li>
  <li>Genotyping 
    <ul>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-627">APT-627</a>] -
        apt-probeset-genotype: Fix crash when inbred file doesn’t list all
        the CEL file names for the genotype batch run</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-938">APT-938</a>] -
        apt-probeset-genotype: Check for wrong header in probeset-ids file not
        working</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-945">APT-945</a>] -
        apt-probeset-genotype: Crash if there are insufficient valid calls on
        probeset - added error message</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-979">APT-979</a>] -
        apt-probeset-summarize: Error running a RMA with no MPS or PS file on a
        WT exon array</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-984">APT-984</a>] -
        apt-probset-summarize: Does not handle unexpected exceptions during
        normalization gracefully.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-1065">APT-1065</a>] -
        apt-probeset-genotype: Analysis-spec string in output files is
        reporting incorrect values for ArtifactReduction if defaults are not
        used (correct string is used in analysis)</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-1096">APT-1096</a>] -
        apt-probeset-genotype: Modified help text to indicate --sqlite-output
        option not available on Windows</li>
    </ul>
  </li>
  <li>DMET 
    <ul>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-970">APT-970</a>] -
        apt-dmet-genotype: Same file used to store reported and analyzed probe
        sets - changed to use different file names</li>
    </ul>
  </li>
</ul>

<h4>Improvement and New Features </h4>
<ul>
  <li>General 
    <ul>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-650">APT-650</a>] -
        The string "amd64" should be "x86_64" in our environmen.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-678">APT-678</a>] -
        Change the APT build system for Windows to MSBuild</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-679">APT-679</a>] -
        Upgrade APT Solution and projects for VS2010</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-784">APT-784</a>] -
        Clean up unused values and other gcc warnings.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-944">APT-944</a>] -
        apt-calvin-equivalent fails to identify NaN - number differences</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-947">APT-947</a>] -
        Edit documents to reflect unix only applications.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-987">APT-987</a>] -
        apt-file5-equivalent: Add "--sign-allow-negation" parameter for command
        line access to affx::File5_File::equivalent() argument
      "bAllowNegation"</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-904">APT-904</a>], [<a
        href="http://jira.affymetrix.com:8080/browse/APT-908">APT-908</a>], [<a
        href="http://jira.affymetrix.com:8080/browse/APT-913">APT-913</a>] -
        Static code analysis: Unitialized variables, unreachable code and other
        fixes</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-973">APT-973</a>] -
        Reorganize SVN repository for APT 1.15</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-1012">APT-1012</a>] -
        Adopt Klocwork static analysis for APT</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-1060">APT-1060</a>] -
        Rename apt/sdk/apt2 directory to avoid confusion with upcoming APT2
        release</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-1019">APT-1019</a>] -
        apt-cel-transform: refactor to use APT BaseEngine class</li>
    </ul>
  </li>
  <li>Testing/Regression Data 
    <ul>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-680">APT-680</a>] -
        Remove symbolic links in regression data</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-952">APT-952</a>] -
        Remove confusing output including references to FATAL ERROR in negative
        unit tests that pass</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-976">APT-976</a>] -
        Allow test results to be moved to different location (outside of source
        code tree)</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-988">APT-988</a>] -
        Missing output for some integration and regression tests fixed</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-1009">APT-1009</a>] -
        Reorganize regression data to allow selective customer downloads based
        on applications used</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-832">APT-832</a>] -
        Add a regression testcase that exercises sockets via the
      SocketEngine</li>
    </ul>
  </li>
  <li>Cyto 
    <ul>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-490">APT-490</a>] -
        apt-copynumber-cyto: AllelePeaks dump no-mans land and FLD filtering
        intermediate data for testing</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-886">APT-886</a>] -
        apt-copynumber-cyto: Add more intermediate results to assist with
        software verification</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-997">APT-997</a>] -
        apt-copynumber-cyto: clarify ref model annotation file in messaging</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-1048">APT-1048</a>] -
        apt-copynumber-cyto: Improve performance of covariate log2 ratio
        adjuster</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-534">APT-534</a>] -
        apt-copynumber-cyto: Add support for processing flag for probe sets for
        CytoScan* arrays</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-897">APT-897</a>] -
        apt-copynumber-gender: Application to check genders of CELs to
        determine valid to run reference model generation reference model
        generation.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-1085">APT-1085</a>] -
        apt-copynumber-cyto: Set algorithm version in CYCHP/DXCHP file to APT
        version instead of ChAS version</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-942">APT-942</a>] -
        apt-copynumber-cyto: Messaging to display major steps as an algorithm
        run progress for ChAS</li>
    </ul>
  </li>
  <li>Genotyping 
    <ul>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-937">APT-937</a>] -
        apt-probeset-genotype: Add error message for non-genotype probeset in
        probeset subset list</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-954">APT-954</a>] -
        apt-probeset-summarize: Added the ability to use the probe set name
        from the PGF file rather than the probe set id in the output of CHP
        files</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-955">APT-955</a>] -
        apt-probeset-summarize: Do not error in DABG when no GC bin data
      exists.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-956">APT-956</a>] -
        apt-probeset-summarize: Add ability to subtract a given amount from the
        degrees of freedom for the chi squared test in DABG (required for miRNA
        analysis)</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-1051">APT-1051</a>] -
        apt-probeset-summarize: Include Probeset Names (not IDs) in output
        TXT/CHP file</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-482">APT-482</a>] -
        apt-probeset-genotype: Suppress
        "affymetrix-algorithm-param-chip-view-md5sum" parameter when probeset
        file is not used</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-1005">APT-1005</a>] -
        Create vignette for miRNA analysis</li>
    </ul>
  </li>
  <li>DMET 
    <ul>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-998">APT-998</a>] -
        apt-md5-check: Create tool to verify MD5 values for DMET Console
      output.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-1006">APT-1006</a>],
        [<a
        href="http://jira.affymetrix.com:8080/browse/APT-1030">APT-1030</a>] -
        apt-metabolizer: Add apt-metabolizer to linux/os x/Windows builds</li>
    </ul>
  </li>
</ul>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.14.4 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Fri Jan 26, 2012</dd>
</dl>

<p>This release focusses on improvements to the apt-copynumber-cyto pipeline
for processing of CytoScanHD and CytoScan750K arrays. Highlights include:</p>
<ul>
  <li>Support for dropped probesets expected in na33 annotation libraries.</li>
  <li>Covariate adjusters to adjust for modulation in Allele Peaks data in
    certain rare samples.</li>
  <li>Improved performance.</li>
</ul>

<h4>Bug fixes and new features</h4>
<ul>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-960">APT-960</a>]
    - Improve confusing help text for apt-copynumber-cyto</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-965">APT-965</a>]
    - Some CNProbeSet state is passed to the next probe set when a bad
    annotation is found.</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-981">APT-981</a>]
    - Implement Allele Peaks covariate adjuster.</li>
  <li>[<a
    href="http://jira.ev.affymetrix.com:8080/browse/APT-1025">APT-1025</a>] -
    Add vignette for CytoScan750K support</li>
</ul>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.14.3.1 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Tue Nov 1, 2011</dd>
</dl>

<p>This is a minor release in support of Axiom avian arrays. Added support for
ZW gender calling (ZW female, ZZ male).</p>

<h4>Improvements and New Features</h4>
<ul>
  <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-535">APT-535</a>] -
    apt-probeset-genotype - added support for gender calling on avian arrays.
    Added options --chrZprobes, --chrWprobes and --zw-gender-calling.</li>
</ul>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.14.3 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Fri July 15, 2011</dd>
</dl>

<p>The following highlights the changes enhancements in this version of the
software.</p>
<ul>
  <li>Support for CytoScanHD array analysis using apt-copynumber-cyto</li>
  <li>Various new and improved algorithms for CytoScanHD analysis</li>
  <li>Various performance improvements</li>
  <li>apt-midas was restored to the release package</li>
</ul>

<p>Details below:</p>

<h4>Bug fixes</h4>
<ul>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-345">APT-345</a>]
    - CNAnalysisMethodGaussianSmooth - improved error messages and added a
    check for the minimum value of the smooth_bw parameter</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-434">APT-434</a>]
    - apt-probeset-summarize - improved checks and error messages for .bgp
    files lacking probes with necessary representative GC counts</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-517">APT-517</a>]
    - apt-probeset-genotype - added tests to verify the file format of priors
    files</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-727">APT-727</a>]
    - apt-copynumber-cyto - improved performance for SSA processing for
    CytoScanHD CEL files</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-728">APT-728</a>]
    - apt-copynumber-cyto - added "uncomputed gender" boolean flag to cychp
    file header to indicate when no gender calling was enabled</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-822">APT-822</a>]
    - apt-copynumber-cyto - updated the calvin file writer to not include empty
    segments tables</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-871">APT-871</a>]
    - apt-copynumber-cyto - standardized the naming of text files created by
    geno-qc</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-887">APT-887</a>]
    - apt-copynumber-cyto - fixed a bug causing LOH segments that extended
    across long gaps in marker coverage to be appropriately identified to gap
    boundaries</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-888">APT-888</a>]
    - apt-copynumber-cyto - fixed a bug where the LOH Segments algorithm did
    not always utilize respecting the minGenomicSpan parameter resulting in
    some LOH segments being miscalled</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-930">APT-930</a>]
    - Updated -help text in apt-copynumber-cyto</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-932">APT-932</a>]
    - Fixed problems compiling hdf5 libraries from source on Ubuntu</li>
</ul>

<h4>Improvements and New Features</h4>
<ul>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-40">APT-40</a>] -
    docs - The file format document has been updated</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-587">APT-587</a>]
    - apt-cdf-export - added the "--file-md5" option to compute the md5 of an
    entire file.</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-645">APT-645</a>]
    - CNReference api - updated the module to support CytoScanHD</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-763">APT-763</a>]
    - apt-copynumber-cyto - added a log message indicating if snp-qc-snp-list
    could not be read</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-789">APT-789</a>]
    - apt-copynumber-cyto - changed the default behaviour to require modules to
    be invoked explicitly. The following modules must now be explicitly called
    to use them as shown before: 
    <ul>
      <li>local-gc-background-correction-reference-method
      pdnn-reference-method</li>
      <li>wave-correction-reference-method wave-correction-reference-method</li>
      <li>local-gc-background-intensity-adjustment-method
        pdnn-intensity-adjustment-method</li>
      <li>image-correction-intensity-adjustment-method
        high-pass-filter-intensity-adjustment-method</li>
      <li>wave-correction-log2ratio-adjustment-method
        wave-correction-log2ratio-adjustment-method</li>
    </ul>
  </li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-810">APT-810</a>]
    - apt-copynumber-cyto - improved performance of wave correction</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-817">APT-817</a>]
    - apt-copynumber-cyto - Genotyping and LOH analysis on the X chromosome are
    now based on CN state and not on the presence/absence of the Y chromosome
    for the CytoScanHD pipeline</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-858">APT-858</a>]
    - apt-copynumber-cyto - made the additional-waves-reference-method module
    selected-qc parameter more generic</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-933">APT-933</a>]
    - Improved readme guidance for windows builds</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-534">APT-534</a>]
    - apt-copynumber-cyto - support for processing flags for CytoScanHD array
    annotations file</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-733">APT-733</a>]
    - apt-copynumber-cyto - implemented Generic Signal Covariate Adjuster</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-734">APT-734</a>]
    - apt-copynumber-cyto - implemented Generic log2 ratio Covariate
  Adjuster</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-752">APT-752</a>]
    - apt-copynumber-cyto - implemented high pass filtering</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-773">APT-773</a>]
    - apt-copynumber-cyto - implemented gradient correction applicable for all
    chips with copy number probes</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-776">APT-776</a>]
    - apt-copynumber-cyto - added Computed Call Rate to Report file and CYCHP
    header</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-778">APT-778</a>]
    - apt-copynumber-cyto - the proportion of the autosomes that exhibit LOH is
    now in the CYCHP header</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-785">APT-785</a>]
    - apt-copynumber-cyto - added support for separate HMM parameters for
    Autosomes, ChrX and ChrY</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-809">APT-809</a>]
    - apt-copynumber-cyto - added support for external covariates file</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-843">APT-843</a>]
    - apt-copynumber-cyto - changed name of the
    log2ratio-adjustment-high-pass-filter to log2ratio-adjustment-method</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-825">APT-825</a>]
    - apt-copynumber-cyto - implemented wavelet shrinkage estimator (signal
    restoration) for log2 ratios</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-884">APT-884</a>]
    - apt-copynumber-workflow - implemented a check to ensure the minimum
    number of cel files were provided for reference file creation</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-912">APT-912</a>]
    - Removed obsolete applications apt-copynumber-analysis,
    apt-copynumber-reference, and apt-copynumber-log2ratio from release.
    Restored apt-midas.</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-923">APT-923</a>]
    - Incorporated changes in Fusion SDK into APT package</li>
  <li>[<a href="http://jira.ev.affymetrix.com:8080/browse/APT-928">APT-928</a>]
    - Removed apt-dmet-copynumber from 1.14.3 release</li>
</ul>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.14.2 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Fri March 18, 2011</dd>
</dl>

<p>The following highlights the changes enhancements in this version of the
software.</p>
<ul>
  <li>Support for the new CDF file format version described in the section on
    the Fusion SDK Version 1.11.1 above.</li>
  <li>Support for various arrays as indicated above in the section on the new
    features included in GTC 4.1.</li>
  <li>New call rate metrics calculation using autosomal SNPs only</li>
  <li>Support for 3000 SNPQC probesets during genotyping</li>
  <li>New artifact reduction support in the genotyping workflow to support
    Axiom arrays with new design</li>
  <li>Performance improvements</li>
</ul>

<p>Details below:</p>

<h4>Improvements and New Features</h4>
<ul>
  <li>Genotyping 
    <ul>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-228">APT-228</a>] -
        apt-geno-qc aborts improved error reporting for corrupted cel
      files.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-252">APT-252</a>] -
        apt-probeset-genotype - fix description of --disk-cache option</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-104">APT-104</a>] -
        apt-probeset-genotype - Report sample call rate, sample percentage of
        AB calls (het_rate) and sample percentage of AA/AB calls (hom_rate)</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-148">APT-148</a>] -
        apt-snp-summary - break out snp summary table calculations in GTC into
        stand-alone app.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-159">APT-159</a>] -
        apt-data-subset - allow sample multichannel cel files for Axiom
      testing.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-213">APT-213</a>] -
        apt-probeset-genotype - More compact data output for genotyping for A5
        and txt formats.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-314">APT-314</a>] -
        apt-probeset-genotype - API for reading cluster location information
        from models files.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-297">APT-297</a>] -
        apt-probeset-summarize - Remove mas5 signal and detection.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-467">APT-467</a>] -
        apt-probeset-genotype - Implemented new QC and overall Call Rate
        metrics for Axiom Genotyping Runs.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-501">APT-501</a>] -
        apt-probeset-genotype - Implemented new Artifact Reduction
      algorithms.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-518">APT-518</a>] -
        apt-probeset-genotype - add option to strip off allele suffix from
        probeset id when writing feature effects file</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-535">APT-535</a>] -
        apt-probeset-genotype - add gender flip to handle avian arrays (Z/W
        chromosomes).`</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-568">APT-568</a>] -
        apt-geno-qc Implemented reagent version 1 and 2 discriminator.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-659">APT-659</a>] -
        Improved genotyping performance change from GTC 4.0 / APT 1.12 to GTC
        4.1 / APT 1.14.2</li>
    </ul>
  </li>
</ul>
<ul>
  <li>Cyto 
    <ul>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-158">APT-158</a>] -
        Cannot specify waviness-seg-count-cutoff=0 and
        use-high-waviness-seg-count=false for the same run.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-238">APT-238</a>] -
        Cyto enhancement: write SCARS even when info is 0</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-339">APT-339</a>] -
        Replace guid value from .Report.txt file of the apt-copynumber-cyto
        application by something that keeps the same value for gold and test
        runs or delete it.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-380">APT-380</a>] -
        Improve handling of output directory creation for Windows</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-150">APT-150</a>] -
        Change apt-copynumber-cyto wave correction code to default to the
        number of waves found in the reference if none defined on the command
        line. Also report the number of waves used in the cychp file.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-183">APT-183</a>] -
        Add median log2 ratio per chromosome to chromosome summary table in
        cychp files.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-237">APT-237</a>] -
        Add ability to configure cychp file name</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-250">APT-250</a>] -
        Support for new parameter for ref mode file creation in cyto that
        indicates if CEL files based on ATP or an manual process.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-336">APT-336</a>] -
        Add Meta-data support to Reference Model file creation and chp files
        for Cyto</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-353">APT-353</a>] -
        Creation details in model files (genotyping specifically)</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-180">APT-180</a>] -
        Adding waveSmooth analysis string option for cyto</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-267">APT-267</a>] -
        Cyto enhancement memory issue for more large numbers of cel files</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-309">APT-309</a>] -
        For APS/APG, when processing cel files, print the name rather than a
        dot to the log file.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-275">APT-275</a>] -
        Warning message change when number of waves is not specified</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-266">APT-266</a>] -
        Change apt implementation of SCAR values so that no more nan values for
        the 1498 SNP markers with Information Content = 0</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-636">APT-636</a>] -
        Check probeset lists for errors and report meaningful message if
      found</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-740">APT-740</a>] -
        apt-geno-qc handling of avian array QC (Z/W chromosomes).</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-344">APT-344</a>] -
        LOH parameters reported in cychp file.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-563">APT-563</a>] -
        for apt-copynumber-cyto need to be able to specify sketch size
        (override default) for SNP6 data where --cyto2 is false</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-182">APT-182</a>] -
        Cyto image correction to use single blocking</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-516">APT-516</a>] -
        Introduction of copy number calling module for CytoScanHD.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-528">APT-528</a>] -
        Enable Cyto HMM call for CytoScanHD arrays under
      apt-copynumber-cyto</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-529">APT-529</a>] -
        Gender calling issues for CytoScanHD array from apt-copynumber-cyto</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-530">APT-530</a>] -
        Enable confidence score calculation in apt-copynumber-cyto for
        CytoScanHD array</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-531">APT-531</a>] -
        Enable LOH result from the Chromosome summary table in
        apt-copynumber-cyto for CytoScanHD</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-544">APT-544</a>] -
        Add processed allelic difference to apt-workflow-cyto.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-652">APT-652</a>] -
        Implement new QC metric for SNP markers</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-775">APT-775</a>] -
        Implement a quantile normaliztion methodology which respects
        differences between autosomes and chromosome X/Y</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-546">APT-546</a>] -
        Add command line control of BRLMM-P parameters in cyto</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-656">APT-656</a>] -
        SNP6 CN performance improvement from GTC 4.0/APT 1.12 to GTC 4.1/APT
        Head</li>
    </ul>
  </li>
</ul>
<ul>
  <li>General 
    <ul>
      <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-55">APT-55</a>]
        - apt-cel-transform: work with multi channel cel files</li>
      <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-56">APT-56</a>]
        - apt-cel-transform: generate new GUID and propagate old in headers</li>
      <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-57">APT-57</a>]
        - apt-cel-transform: include command line in cel header</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-121">APT-121</a>] -
        Reduce disk usage by recycling intesity marts</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-122">APT-122</a>] -
        DiskIntensityMart to allow arbitrary data names.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-127">APT-127</a>] -
        Remove ChipLayout from AnalysisStream function calls.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-129">APT-129</a>] -
        Store probe level data outside of ChipLayout.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-140">APT-140</a>] -
        Remove references to raw_intensity_mean and raw_intensity_stddev from
        reports.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-155">APT-155</a>] -
        Write only analysis probe intensities instead of all intensities for
        final diskmart</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-156">APT-156</a>] -
        Initialize diskmart with CEL order instead of layout order</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-193">APT-193</a>] -
        Add a version of IntensityMart::getCelData that accepts a channel
        designation</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-206">APT-206</a>] -
        Add apt-copynumber-correlation a new app to generate a correlation
        matrix of log2 ratios using cychp files as input.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-216">APT-216</a>] -
        The --analysis-files-path search path is Unix-style colon-separated and
        it gets confused by Windows absolute paths.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-220">APT-220</a>] -
        Improve version reporting</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-224">APT-224</a>] - Do
        not include NaNs in correlation calculations for
        Calvin::equivalent(...) function.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-253">APT-253</a>] -
        Allow hints for a subset of CEL files/samples in the hints file</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-262">APT-262</a>] -
        apt-cel-extract error message when --target-sketch-file is given, but
        not --analysis</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-279">APT-279</a>] -
        Replace calls to quickSort and binarySearch by STL equivalents.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-350">APT-350</a>] -
        Allow brlmm-p posteriors to be used as priors without edits</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-358">APT-358</a>] -
        Configure - add example of setting sdk_output_top</li>
      <li>[<a href="http://jira.affymetrix.com:8080/browse/APT-10">APT-10</a>]
        - Allow use of sparse-mart (in-memory mart) with chipstream code</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-138">APT-138</a>] -
        File5 - Add char/short datatypes</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-198">APT-198</a>] -
        SummaryGenotypeEngine can input and output file5 format</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-212">APT-212</a>] -
        Output summaries, calls and confidences in sqlite format for Simon's
        group</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-234">APT-234</a>] -
        Support for providing metadata to write into the CHP file header</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-245">APT-245</a>] -
        Add capability of ignoring "Sets" in Calvin comparison</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-293">APT-293</a>] -
        Engines listen and prints on sockets for Windows integration</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-273">APT-273</a>] -
        Handle long filenames on windows</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-284">APT-284</a>] -
        apt-cel-transformer - add --cel-files option.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-360">APT-360</a>] -
        Consolidate file functions from Util into calvin_files FileUtils</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-686">APT-686</a>] -
        Use environment variables/macros for documentation version numbers,
      etc.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-436">APT-436</a>] -
        Converted all of APT to use FsPath libraries. (Consistent fix for
        MAX_PATH issues).</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-246">APT-246</a>] -
        Tsv support for shorts</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-688">APT-688</a>] -
        Added CDF GUID to the CHP header</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-685">APT-685</a>] -
        Remove in-place updates of source code from doxygen make process</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-696">APT-696</a>] -
        Fix formatting of html docs</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-697">APT-697</a>] -
        Move help docs out of .cpp files into separate .dox files</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-699">APT-699</a>] -
        Clean-up/simplify make dox task</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-458">APT-458</a>] -
        apt-cdf-to-spf - modify to handle multi-channel cdf input (CDF file
        format version 4)</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-485">APT-485</a>] -
        calvinlite should check for other celfile versions.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-731">APT-731</a>] -
        calvinlite should be able to select keys to print.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-482">APT-482</a>] -
        Added md5 values to CHP file headers.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-548">APT-548</a>] -
        Replaced comman and tab delimited file access with file/TsvFile.
        (Removed util/AffxFile).</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-672">APT-672</a>] -
        apt-annotation-converter added for all platforms.</li>
    </ul>
  </li>
</ul>

<h4>Bugs</h4>
<ul>
  <li>Genotyping 
    <ul>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-124">APT-124</a>] -
        Genotype chp check reports errors with confidences match instead of
        when they don't match</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-178">APT-178</a>] -
        Canary parameter help info printed on stdout when it should be on
      stderr</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-202">APT-202</a>] -
        Calvin GenericFileWriter generates empty error message if it cannot
        open the file.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-225">APT-225</a>] -
        Uncaught exception in apt-probeset-genotype if cel file does not
      exist.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-261">APT-261</a>] -
        apt-probeset-summarize unable to remove temp dir</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-370">APT-370</a>] -
        Application apt-probeset-summarize memory issue for PM only array.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-388">APT-388</a>] -
        apt-probset-summarize fails when running Adapter Type Normalization
        analysis method - Allowing annotation-file to be specified for
        apt-probeset-summarize.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-398">APT-398</a>] -
        apt-probeset-genotype - added a minimum value to CSepThr.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-393">APT-393</a>] -
        Using SNP list for genotyping produces correct txt files but CHP file
        does not contain calls only header</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-415">APT-415</a>] - QC
        performance improvements.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-488">APT-488</a>] -
        Error when insufficient space to run genotyping</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-512">APT-512</a>] -
        Rat array genotyping fails using APT windows (head and 1.12), ok in
        Linux (head and 1.12)</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-550">APT-550</a>] -
        Reading a poorly formed models file causes the application to abruptly
        exit</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-632">APT-632</a>] -
        Added error if user tries to specify both --read-genders and
        --set-gender-method on the commandline.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-664">APT-664</a>] -
        SNP6 copynumber - Gender column in 4.1 changed to "Unknown" for all
        CNCHP files</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-682">APT-682</a>] -
        apt-probeset-genotype fails to run for axiom array when library files
        are in E or F drive (rather than C)</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-714">APT-714</a>] -
        apt-geno-qc memory leak when reading cel files fixed.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-722">APT-722</a>] -
        Memory ios_base::failbit set error shows when using .cel files on
        network paths.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-799">APT-799</a>] -
        snpQC call rates show as percentage instead of fraction.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-207">APT-207</a>] -
        APT High Pass Filtering never runs when PDNN Background Correction is
        turned off</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-445">APT-445</a>] -
        SpectClust::findNLargestEvals always returns "false" for "converged"
        (even if did converge)</li>
    </ul>
  </li>
</ul>
<ul>
  <li>Cyto 
    <ul>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-154">APT-154</a>] -
        CNReferenceMethodWaveCorrection opens reference files as read/write
        when should be read only</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-162">APT-162</a>] -
        Cyto legacy bug fixes: util/AffyConvgetDouble(...) - Rounding errors on
        floating point output.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-164">APT-164</a>] -
        Cyto legacy bug fixes: CNAnalysisMethodChipstream::getSampleId(...) -
        No longer parsing "_NA" from sample name.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-165">APT-165</a>] -
        Cyto legacy bug fixes: CNAnalysisMethodChipstream Missing Last Genotype
        Call Override Sample</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-166">APT-166</a>] -
        Cyto legacy bug fixes: CNAnalysisMethodLog2Ratio Removed return when
        Median == 0. (Very rarely occuring bug, seen in some cancer samples
        only).</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-167">APT-167</a>] -
        Cyto legacy bug fixes: CNAnalysisMethodReference PDNN Median
      Intensity</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-169">APT-169</a>] -
        Cyto legacy bug fixes: CNLog2RatioAdjustmentMethodWaveCorrection -
        Application always uses the maximum number of waves in CN reference
        regardless of what user specifies for wave-count.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-170">APT-170</a>] -
        Cyto legacy bug fixes: CNProbeSet Underflow in GC Binning.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-171">APT-171</a>] -
        Cyto legacy bug fixes: CNReferenceMethodWaveCorrection. Changed order
        of processing from Probe Set Name to Genomic Order.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-172">APT-172</a>] -
        Cyto legacy bug fixes: CNSmoother Module - remove short segments in CN
        run time encoding.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-175">APT-175</a>] -
        CEL file index numbers are being printed to report files instead of CEL
        file names</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-241">APT-241</a>] -
        Cyto performance improvements.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-215">APT-215</a>] -
        Parameter --analysis-files-path is being ignored for the
        --snp-qc-snp-list file in certain cases</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-276">APT-276</a>] -
        Cyto CNAnalysis for reference model file uses wrong extension if not
        set in job file</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-294">APT-294</a>] -
        Improved CN/HMM</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-295">APT-295</a>] -
        Improve AllelePeaks detection</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-299">APT-299</a>] -
        Improve LOH calling for chromosomes 19 and 22</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-321">APT-321</a>] -
        Change in Behavior for --cels Argument: errAbort CNCytoEngine if CEL
        files are specified both as command line arguments and cel-file or cels
        list(s).</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-354">APT-354</a>] -
        Added probe-file option to the analysis-files-path prepend list in
        CNCytoEngine</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-391">APT-391</a>] -
        CN/SNP processing flags need to be handled properly - handling of flag
        2.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-419">APT-419</a>] -
        apt-copynumber-cyto --explain unexpected should throw warning
      message.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-456">APT-456</a>] -
        Marker Names Unavailable for SCAR Values from APT - now output all
        markers used in the reference creation to the ProbeSet section of the
        cychp file.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-466">APT-466</a>] -
        apt-calvinlite-util does not display values of boolean header
        parameters, only an empty string.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-489">APT-489</a>] -
        Allele peaks data differences between Cyto SSA runs</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-502">APT-502</a>] -
        Marker count of Mosaicism counts the SNPs that do not have a
      log2ratio</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-519">APT-519</a>] -
        apt-copynumber-cyto vs workflow Chr X LOH differences for Female
      samples</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-520">APT-520</a>] -
        apt-copynumber-cyto vs workflow extra records in reference
        feature_response and snp_posteriors table</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-522">APT-522</a>] -
        Numerical differences between estimated waves from
        apt-copynumber-workflow and apt-copynumber-cyto</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-523">APT-523</a>] -
        AGR probe id was not generated in CN reference build for SNP6 data
        using apt-copynumber-cyto</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-525">APT-525</a>] -
        LOH calls problem comparing results from apt-copynumber-cyto and
        apt-copynumber-workflow</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-540">APT-540</a>] -
        Scrambled CNReferenceHeader Header Values in Cyto and Workflow</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-541">APT-541</a>] -
        cyto generates single-sample mode output with rounding differences vs
        workflow when wave correction enabled</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-615">APT-615</a>] -
        apt-copynumber-cyto AllelePeaks algorithm 6X duplicate rows in PekSets
        dataset</li>
    </ul>
  </li>
</ul>
<ul>
  <li>General 
    <ul>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-142">APT-142</a>] -
        File5 - size not correct after write_array.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-260">APT-260</a>] -
        apt-chp-to-txt documentation bug</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-281">APT-281</a>] -
        apt-file5-util: no help reported when no args on command line</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-337">APT-337</a>] -
        Util::memInfo reports swapAvail=0 on Darwin OS-X - Documented.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-363">APT-363</a>] -
        CHP file headers slow to read - improved.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-369">APT-369</a>] -
        Fix bug in apt-mas5 where .cel was included in output chp file name</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-374">APT-374</a>] -
        TmpFileFactory ignoring user-defined file pre/suffixes</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-510">APT-510</a>] -
        APT will fail if batch name has "=" sign</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-612">APT-612</a>] -
        apt-calvinlite-util unable read short .chp file with missing data which
        Calvin Viewer can read</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-634">APT-634</a>] -
        Fixed Windows 260 character MAX_PATH limit. Consolidated all file
        access methods and routines in util/Util and util/AffxFile respectively
        to util/Fs to support windows MAX_PATH.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-669">APT-669</a>] -
        Function SnpModelConverter::getTsvArrayType fails to get chip type of
        posterior models - added substring matching to check chip types.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-674">APT-674</a>] -
        Added exception handling for SQLite libraries.</li>
      <li>[<a
        href="http://jira.affymetrix.com:8080/browse/APT-701">APT-701</a>] -
        FsPath constructor doesn't set internal path name.</li>
    </ul>
  </li>
</ul>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.12.a </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Tue Sept 23 2010 
      <ul>
        <li>Incremental release containing apt-geno-qc and
          apt-probeset-genotype only.</li>
      </ul>
    </dd>
</dl>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.12.0 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Monday Oct 12 2009 
      <ul>
        <li>Full Axiom ™ support</li>
        <li>Added support for multi channel cel files to apt-cel-extract</li>
      </ul>
    </dd>
</dl>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.11.0 (Internal Release Only) </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Sat Aug 29 2009 
      <ul>
        <li><b>Fixed bug in Windows x64 binary release where executables
          depended on debug runtime libraries</b></li>
        <li><b>Fixed a bug in AdapterTypeNormalization where transformed
          intensities were not passed through to next chipstream when
          initializing.</b></li>
        <li><b>Partial release</b> to support QC and genotype calling on the
          Axiom ™ platform. We recomend Staying with 1.10.2 for other
          applications of APT until the next full release (1.12.0).</li>
        <li><b>Fixed a problem with the segments in a CYCHP file. The
          segment_id should be a UINT, not a STRING.</b></li>
        <li>Cytogenetics Genome Wide Array Support</li>
        <li>Added apt-summary-genotype to windows binary releases</li>
        <li>Added apt-mas5</li>
        <li>Added support for SQLite annotation files to
        apt-copynumber-workflow</li>
        <li>Added artifact reduction chipstream node.</li>
        <li>Inbred tweaks to apt-probeset-genotype</li>
        <li>Added waviness amplitude metric to cychp summary metrics</li>
        <li>Added segments for CNNeutralLOH and NormalDiploid.
        [experimental]</li>
        <li>Added antigenomic ratio to CNExperiment [experimental]</li>
        <li>Added Mosaicism. [experimental]</li>
        <li>enable wave correction for SNP6 [experimental]</li>
        <li>refactor how feature effects are handled. Now match by probe ID AND
          probeset ID</li>
        <li>Switch Windows builds to use VS2008</li>
        <li>Added XML output options for unit and regression tests</li>
        <li>Added support for multiple intensity channels (eg multi color CEL
          files)</li>
        <li>Refactor BaseEngine and handling of option and state.</li>
        <li>Commandline programs which use a BaseEngine derived analysis, now
          accept a set of shared options</li>
        <li>remove use of tmp chp file when outputing CHP files</li>
        <li>removed in-memory mode for apt-probeset-genotype and
          apt-probeset-summarize</li>
        <li>reconcilled differences in same options between
          apt-probeset-genotype and apt-probeset-summarize</li>
      </ul>
    </dd>
</dl>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.10.2 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Tue Jan 27 2009 
      <ul>
        <li>Added support for DMET Plus: 
          <ul>
            <li>apt-dmet-genotype will generate DMET CHP files from DMET CEL
              files. The CHP files contain genotyping and copynumber
            results.</li>
            <li>apt-dmet-translation will generate DMET star allele translation
              reports from CHP files.</li>
          </ul>
        </li>
        <li>Fix bug in apt-probeset-summarize --probeset-ids option when
          processing genotype probesets using the expr.genotype=true modifier
          in the analysis string.</li>
        <li>Improve error handling when required columns in TSV files are
          missing.</li>
        <li>Fix CNCHP header so that files can be loaded into GTC 3.x</li>
        <li>Fixed problem in CN reference file generation so that APT generated
          reference files are backwards compatabile with GTC 3.x</li>
        <li>reconcilled differences in same options between
          apt-probeset-genotype and apt-probeset-summarize</li>
      </ul>
    </dd>
</dl>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.10.2 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Tue Jan 27 2009 
      <ul>
        <li>Added support for DMET Plus: 
          <ul>
            <li>apt-dmet-genotype will generate DMET CHP files from DMET CEL
              files. The CHP files contain genotyping and copynumber
            results.</li>
            <li>apt-dmet-translation will generate DMET star allele translation
              reports from CHP files.</li>
          </ul>
        </li>
        <li>Fix bug in apt-probeset-summarize --probeset-ids option when
          processing genotype probesets using the expr.genotype=true modifier
          in the analysis string.</li>
        <li>Improve error handling when required columns in TSV files are
          missing.</li>
        <li>Fix CNCHP header so that files can be loaded into GTC 3.x</li>
        <li>Fixed problem in CN reference file generation so that APT generated
          reference files are backwards compatabile with GTC 3.x</li>
        <li>Fix quantile normalization code so that values outside a predefined
          target sketch are capped. Without this fix infinite values (and
          negative values) were possible.</li>
        <li>Fix makeDir error in apt-geno-qc when output file specification
          (from --out-dir option) did not include a directory portion.</li>
        <li>Accomidate genotype probesets which do not have equal numbers of
          allele A and B probes.</li>
        <li>In apt-cel-extract, fix the ordering of probes in the on-disk
          intensity mart. Previous version was very slow due to non-optimal
          ordering of probe intensities on disk. Related to this fix is that
          the output order is different when using a probeset list. (It is now
          in the same order as the provided probeset list).</li>
        <li>Fixed a bug where target sketches provided to apt-cel-transformer
          where assumed to be in a5 format.</li>
        <li>Fixed a bug where apt-probeset-summarize would fail if an entire
          probeset was removed via the probe kill list when using a CDF
        file.</li>
        <li>Added utility apt-engine-wrapper which allows one to directly call
          analysis engines and query for what options an engine will
        accept.</li>
        <li>Added apt-file5-util which allows one to convert between a5 and
          text/tsv file formats.</li>
        <li>Drop binaries for Solaris/SPARC</li>
        <li>Change in OS-X binaries: 
          <ul>
            <li>dropped "un-tested" universal binaries</li>
            <li>providing stand alone 64-bit PowerPC binaries</li>
            <li>providing stand alone 64-bit Intel binaries</li>
          </ul>
        </li>
        <li>Changed APT guids to be RFC-4122 format guids</li>
      </ul>
    </dd>
</dl>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.10.1 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Mon Nov 3 2008 
      <ul>
        <li><b>Fixed bug in MAPD calculation by copynumber workflow</b>. In
          1.10.0, the MAPD values were computed before putting probesets in
          genome order. The implication is that MAPDs reported by 1.10.0 were
          inflated relative to the real MAPD values.</li>
        <li>Add genome GC content correction to copynumber workflow.</li>
        <li>Add a log file to apt-summary-genotype</li>
        <li>Add on-disk mode to apt-cel-extract (on by default)</li>
        <li>Add progress meters for chip layout file reading (ie CDF)</li>
        <li>Fixed bug where block size and memory check code was still being
          run even when on-disk mode was set. The implication in 1.10.0 was
          that multiple iterations (library file reads) were being done
          unnecessarily. Results were still correct.</li>
        <li>Add copynumber binaries to windows installer (missing in 1.10.0
          release)</li>
        <li>Changed the code to only read lib files only once when doing one
          iteration in apt-probeset-genotype and apt-probeset-summarize.</li>
        <li>Clean up of a5 options for apt-probeset-genotype</li>
        <li>Add a5 options to apt-probeset-summarize</li>
        <li>Refactored the analysis engines to inherit from a common engine
          base class.</li>
      </ul>
    </dd>
</dl>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.10.0 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Wed Jul 9 2008 
      <ul>
        <li><b>Fixed bug where pm-sum method was reporting values equivalent to
          pm-only when using SPF file rather than a CDF file. This is in the
          context of using apt-probeset-summarize with genotyping array via the
          "expr" addition to the analysis string.</b></li>
        <li><b>Fixed bug where incorrect "x" coordinate was reported in feature
          effect output files from apt-probeset-summarize and
          apt-probeset-genotype (all releases) and in apt-cel-extract (releases
          newer than 1.8.0) output. Incorrect values only show up when working
          with arrays where the number of rows and columns in the CEL file are
          not equal.</b></li>
        <li><b>Fixed bug in PLIER where feature effects were updated after the
          last update to target responses</b></li>
        <li>Added copynumber functionality for SNP6 (see
          apt-copynumber-workflow).</li>
        <li>Added Canary copynumber calling for SNP6 (see apt-canary).</li>
        <li>Complete refactor of apt-chp-to-txt. Now handles most chp file
          formats and types. Resequencing and Tag chp files are not handled by
          this tool.</li>
        <li>Added Vignettes: 
          <ul>
            <li>SNP6 Copynumber analysis using individual components</li>
            <li>Single sample genotyping (ie for QC purposes)</li>
          </ul>
        </li>
        <li>Removed apt-cel-to-pq. Supplanted by apt-cel-extract.</li>
        <li>Misc 
          <ul>
            <li>Added experimental HDF5 output file formats to
              apt-probeset-genotype</li>
            <li>Added on-disk intensity mart. Set as the default for
              apt-probeset-summarize and apt-probeset-genotype.</li>
            <li>Fixed memory capping on x64 windows and 64-bit OS-X.</li>
            <li>Added the ability to apply pre-computed feature effects when
              using median polish in apt-probeset-summarize and
              apt-probeset-genotype.</li>
            <li>Added an adaptor type normalization (used by copynumber
              workflow)</li>
            <li>apt-probeset-genotype will now report signal estimates for copy
              number probesets when the --all-types and --summaries option is
              given</li>
          </ul>
        </li>
      </ul>
    </dd>
</dl>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.8.6 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Mon Feb 3 2008 
      <ul>
        <li><b>Fix bug in IterPlier where one would get a segfault if the
          number of chips exceeded 100.</b></li>
        <li><b>Change from float to double in QuantBirdseed methods to improve
          cross platform portability. This may result in slight differences in
          the confidence value and in rare cases a change between making a
          genotype call versus a no-call.</b></li>
        <li><b>order of results in apt-probeset-genotype report file have
          changed due to some refactoring of the code.</b></li>
        <li>added birdseed-v2</li>
        <li>added ContrastQC qc metric to apt-geno-qc</li>
        <li>fix memory estimation (don't cap at 2G on Windows (x64) and OS-X
          (ppc64, x86_64)</li>
        <li>apt-cel-transformer: remove ability to change formats. add ability
          to handle agcc files</li>
        <li>Improvements to command line option handling</li>
        <li>Allow med-norm to be followed by other chip stream components when
          using a fixed target</li>
        <li>added a target and doavg options to quant-norm</li>
        <li>fix bug in --select-probes option of apt-probeset-genotype where
          probe IDs were not reported correctly</li>
      </ul>
    </dd>
</dl>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.8.5 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Wed Oct 31 2007</dd>
</dl>
<ul>
  <li><b>Change to the DM Implementation</b> DM calls/confidences can be
    reported by apt-geno-qc and are generated and used by apt-probeset-genotype
    to generate seed calls for BRLMM on 500k data. Differences were observed
    between different platforms (particularly 64-bit vs 32-bit) in a small
    number of SNPs (ie 0-5 SNPs out of 250,000 on 500K Sty Chip). Code changes
    were made which rectify this, however you may see some some changes in DM
    results because of this.</li>
</ul>
<ul>
  <li>Binary Releases 
    <ul>
      <li>Added 64-bit Windows Binaries (x64)</li>
    </ul>
  </li>
</ul>
<ul>
  <li>New documentation 
    <ul>
      <li>Vignettes 
        <ul>
          <li>Masking Out Probes in apt-probeset-summarize</li>
          <li>Use of APT to Analyze WT Based Exon and Gene Expression
          Arrays</li>
        </ul>
      </li>
      <li>Added new top level pages and new navigation for end user (binary
        release) documentation 
        <ul>
          <li>Change Log</li>
          <li>Vignettes</li>
          <li>FAQs</li>
          <li>File Formats</li>
          <li>Platforms</li>
        </ul>
      </li>
      <li>Updated Manuals 
        <ul>
          <li>Added new FAQ items to application specific manuals</li>
        </ul>
      </li>
    </ul>
  </li>
</ul>
<ul>
  <li>New Applications: 
    <ul>
      <li>apt-cel-convert: a tool for converting between various CEL file
        formats.</li>
    </ul>
  </li>
</ul>
<ul>
  <li>Changed Applications: 
    <ul>
      <li>Most Applications 
        <ul>
          <li>log messages on stderr rather than stdout</li>
        </ul>
      </li>
      <li>apt-cel-transformer: 
        <ul>
          <li>support PGF/CLF and SPF in addition to CDF</li>
        </ul>
      </li>
      <li>apt-cel-to-pq: 
        <ul>
          <li>support AGCC CEL files</li>
        </ul>
      </li>
      <li>apt-cel-extract: 
        <ul>
          <li>added support for AGCC CEL files</li>
          <li>complete rewrite of this application to use the core APT
            SparseMart framework. As a result this is really a new tool as far
            as command line arguements and behavior of the application are
            concerned.</li>
          <li>the tool can now scale to a large number of cel files (although
            you will need to set the --block-size manually)</li>
          <li>block number is reported (critical for assigning genotype probes
            to alleles)</li>
          <li>you can provide an analysis string (including pm-adjustor) to
            manipulate probe intensities</li>
          <li>you can report just the pm values, the pm and background (ie mm,
            gcbg, ...) values together (ie 132,22), or the result of the pm
            minus background</li>
          <li>you can provide both a probeset list and probe list (intersection
            will be extracted)</li>
          <li>probeset and probe lists now work with CDF files</li>
          <li>the probeset type is reported (useful for multi-type CDF files
            like SNP6)</li>
          <li>the tool will work without any library files (with reduced
            functionality of course)</li>
          <li>enable use of experimental spf library files</li>
        </ul>
      </li>
      <li>apt-geno-qc: 
        <ul>
          <li>added --chrX-probe and --chrY-probe options which enable CN probe
            gender calling</li>
          <li>added --probe-class-file option which will turn on the
            CelStatListener and report probe intensity means by group</li>
          <li>new column headers -- consistency with apt-probeset-genotype
            report.</li>
          <li>new ordering of columns -- this will continue to vary over time.
            You should rely on column labels rather than column position to
            parse out specific information. Same holds true for
            apt-probeset-genotype report file</li>
          <li>dropped --rpt/-r option, (replaced with --dm-out option)</li>
          <li>Added --dm-out, --dm-thresh, and --dm-het-mult options for
            enabling DM calls and confidence output and setting DM options.
            Note that the output is now in TSV format.</li>
          <li>complete refactor of the source code</li>
        </ul>
      </li>
      <li>apt-probeset-summarize: 
        <ul>
          <li><b>Fixed a bug in MAS5 background correction routine when using
            pgf file.</b> In short, the number of probe sets was set to 0 when
            using a pgf file.</li>
          <li>quality assessment metrics and QC probeset values are now written
            to the AGCC CHP file headers</li>
          <li>probesets which failed to be quantified (ie certain controls) are
            no longer written to the AGCC CHP file. In previous versions such
            probesets would typically show up with a 0.0 quantification
          value.</li>
          <li>changes to how the md5sum (gc background probes and normalization
            probes) is computed to ensure cross platform consistency</li>
          <li>syncronized meta header data between apt-probeset-genotype and
            apt-probeset summarize. IE, there will be some changes to the meta
            info going into CHP files and other output files.</li>
          <li>added support for SPF files</li>
          <li>fixed bug where mas50-bg chipstream failed if there were
            additional chip streams after it</li>
          <li>added additional boundary conditions in RMA::findBandWidth in
            order to match behavior of BioConductor. May fix errors with
            certain chips with low overall intensity.</li>
          <li>previous behavior was to not output mm_* quality metrics when
            bgrd probes were provided (via --bgp option). Now mm_* is reported
            if there are more than 5000 MM probes listed in the library
            file(s). Note that for the Human, Mouse, and Rat Exon 1.0 ST
            arrays, this will be dominated by the "genomic background" probes
            which are flagged as MM probes in the PGF file.</li>
          <li>removed the path from the cel file names listed in the report
            file.</li>
          <li>improved masking of probes (still experimental) 
            <ul>
              <li>--rm-probes to --kill-list (sync w/
              apt-probeset-genotype)</li>
              <li>take into account probeset when masking probes -- ie you can
                now mask out a probe that is in more than one probeset</li>
              <li>masking file can have probe_id (1 base) or x/y position. if
                both are present it will check to see that they are
              consistent.</li>
              <li>when probesets and meta probesets are completely masked out,
                a warning will be issued rather than having the software exit
                with an error message.</li>
            </ul>
          </li>
          <li>added --mem-usage option to set memory use in MB as alternative
            to setting block-size</li>
          <li>added --chip-type option. can be specified multiple times. This
            option will dictate the chip type in the output files and it will
            also specify what chip types the cel and lib files are checked
            against. Note that this will override the typcial check between
            chip types in lib files versus those in the cel files.</li>
          <li>When doing CDF based analysis, chip types based on substrings
            ending with "." are also considered a match. IE, CDF file
            foo.bar.v1.r10.cdf will be detected as matching the possible chip
            types foo.bar.v1.r10, foo.bar.v1, foo.bar, and foo.</li>
          <li>new --cc-md-chp-output option will generate experimental AGCC
            Multi Data CHP files.</li>
          <li>added a --set-analysis-name option which will override the name
            computed from the analysis string. This option will affect the
            prefix used for output file names and the labels in certain meta
            fields and/or report file metrics.</li>
        </ul>
      </li>
      <li>apt-probeset-genotype: 
        <ul>
          <li><b>fixed bug where chip-type in CHP file header was not set when
            using --force option</b></li>
          <li><b>removed --txt-output option, which had a bug where calls were
            not thresholded in text output file for birdseed</b></li>
          <li><b>fixed bug on solaris where use of birdseed method would cause
            a segfault</b></li>
          <li>Added documentation on --write-models option of
            apt-probeset-genotype</li>
          <li>fix reported analysis spec for all genotype calling methods. The
            genotype calling method (Quant Method) was not reported and in some
            cases incorrect algorithm parameters were reported.</li>
          <li>for BRLMM-P, cluster_distance_mean metric is now correctly
            reported. Previously it was just set to zero.</li>
          <li>fixed bug where -s option with --cc-chp-output would fail</li>
          <li>added a number of additional metrics to the report <a
            href="file:">file:</a> 
            <ul>
              <li>DM call rate (when DM listener is run -- ie BRLMM)</li>
              <li>Various gender calls based on methods which are suitable to
                the chips/analysis being run</li>
              <li>Various metrics underlying the gender calls</li>
            </ul>
          </li>
          <li>added a --probe-class-file option which will read in a file
            grouping probes into classes for use by the CelStatListener.</li>
          <li>added --read-genders option to read genders from a file.</li>
          <li>implement --set-gender-method option to force use of a particular
            gender method. Current options are: 
            <ul>
              <li>"cn-probe-chrXY-ratio"</li>
              <li>"dm-chrX-het-rate"</li>
              <li>"em-cluster-chrX-het-contrast"</li>
              <li>"user-supplied"</li>
              <li>"none"</li>
              <li>"supplied-genotypes-chrX-het-rate"</li>
            </ul>
          </li>
          <li>change chrXForce to --no-gender-force: ie run even if no suitable
            gender method is available</li>
          <li>removed labelz (brlmm-p) KXY option which did nothing.</li>
          <li>added birdseed-dev genotype calling method. "birdseed-v1" (alias
            for full analysis, and quant method name) will return the original
            birdseed implementation. "birdseed-dev" (alias for full analysis,
            and quant method name) will return the latest integrated method for
            birdseed from the Broad. This implementation will change over time.
            "birdseed" (alias for full analysis, and quant method name) will
            return the original birdseed-v1 implementation.</li>
          <li>An option, --set-analysis-name, will be added to
            apt-probeset-genotype. This will allow one to set the analysis name
            irregardless of what analysis method is used. IE,
            "--set-analysis-name %birdseed" would allow birdseed-dev to
            masquerade as birdseed. Note that the analysis spec field would
            still enumerate the exact method used.</li>
          <li>birdseed-dev has been updated to reflect birdseed version 2.6
            from the Broad.</li>
          <li>a column header for the cel file names is now present in report
            file output. Meta information is now included in the report file
            header. Additional column (call_rate) was added which matches that
            in the AGCC CHP file header. Also note that only the filename is
            now listed. The path, if present is removed.</li>
          <li>changes to how the md5sum (gc background probes and normalization
            probes) is computed to ensure cross platform consistency</li>
          <li>syncronized meta header data between apt-probeset-genotype and
            apt-probeset summarize. IE, there will be some changes to the meta
            info going into CHP files and other output files.</li>
          <li>added support for SPF files</li>
          <li>improved masking of probes (still experimental) 
            <ul>
              <li>see details under apt-probeset-summarize</li>
              <li>currently cannot remove an entire probeset -- will result in
                an error. use probeset list with -s option instead.</li>
            </ul>
          </li>
          <li>added --chip-type option. can be specified multiple times. This
            option will dictate the chip type in the output files and it will
            also specify what chip types the cel and lib files are checked
            against. Note that this will override the typcial check between
            chip types in lib files versus those in the cel files.</li>
          <li>When doing CDF based analysis, chip types based on substrings
            ending with "." are also considered a match. IE, CDF file
            foo.bar.v1.r10.cdf will be detected as matching the possible chip
            types foo.bar.v1.r10, foo.bar.v1, foo.bar, and foo.</li>
          <li>added a --set-analysis-name option which will override the name
            computed from the analysis string. This option will affect the
            prefix used for output file names and the labels in certain meta
            fields and/or report file metrics.</li>
        </ul>
      </li>
    </ul>
  </li>
</ul>
<ul>
  <li>Source Code 
    <ul>
      <li>refactor how genders are handled. Now GenderCalls is then generated
        and passed around in the genotype engine. 
        <ul>
          <li>refactor EM gender calling to implement ChipSummary into
          reporters</li>
          <li>refactor DM listener to implement ChipSummary into reporters</li>
        </ul>
      </li>
      <li>refactor PgOptions class to dynamicly allocate options. (See
        apt-probeset-genotype for example of the new interface.)</li>
      <li>integrated birdseed 2.4 from the Broad 
        <ul>
          <li>QuantMethods: birdseed-v1, birdseed-dev, birdseed (which is
            birdseed-v1)</li>
          <li>Analysis Strings: birdseed-v1, birdseed-dev, birdseed (which is
            birdseed-v1)</li>
          <li>refactor the CN probe gender calling method as a ChipSummary into
            reporters.</li>
          <li>change eps/epsilon alg params to use full number, not scientific
            notation which breaks the state propigation in analysis-spec</li>
          <li>moved SpecialSnps into chipstream folder</li>
          <li>fixed propigation of user set values into reported analysis spec
            for birdseed</li>
          <li>current birdseed folder renamed birdseed-dev</li>
          <li>created birdseed-v1 folder with previous birdseed implementation
            [GenderCaller and FitSNPGaussiansPriors3]</li>
          <li>The code in birdseed-v1 moved into a C++ namespace
          "birdseed::v1".</li>
          <li>The code in birdseed-dev moved into a C++ namespace
            "birdseed::dev".</li>
          <li>New analysis spec alias: birdseed-v1, birdseed-dev. birdseed-dev
            will always point to the latest Broad recomendations. birdseed-v2
            will be added when the birdseed v2 method is frozen.</li>
          <li>Analysis spec "birdseed" would remain as a pointer to
            "birdseed-v1".</li>
          <li>birdseed correction-factor was locked at 1. can now be set by
            user.</li>
        </ul>
      </li>
      <li>implement FileGenders for reading genotype genders -- also checks
        against cel file names in opts.celFiles (previous stub did not).</li>
      <li>removed unused legacy code in ProbesetGenotypeEngine.cpp</li>
      <li>minor clean up of some memory de-allocation in main compute
      method</li>
      <li>complete refactor of ChipLayout and CelListeners.</li>
      <li>The regression tests have been reconfigured to allow for running the
        regression tests outside of Affymetrix. Regression test data is now
        provided as a separate download from the APT source code.</li>
      <li>Refactored windows builds to use static libraries for core components
        (ie calvin, file, chipstream).</li>
      <li>Lots more regression tests</li>
      <li>Support for expression Chp file checking in CalvinChpCheck.h</li>
      <li>fix memory allocation in Pg::Options</li>
      <li>More checks regarding GC count info when using gc-bg, pm-gcbg, and
        dabg</li>
      <li>More use of Verbose class for log output (rather than using cout or
        cerr directly.</li>
      <li>More use of Err class (rather than exit() or throwing exceptions)</li>
      <li>Verbose::warn no default to cerr rather than cout.</li>
      <li>Moved chip-type checks out of the command line wrappers and into the
        engine code. (ProbesetSummarize and ProbesetGenotype) 
        <ul>
          <li>generally you do not want to fill in the chipType and
            chipTypeSupplied members of the engine options class.</li>
          <li>setting chipType will force all output to be reported as that
            chip type.</li>
          <li>setting chipTypeSupplied will force chip type checking against
            that list of chip types. If chipType is not set, it will be set to
            the first entry in chipTypeSupplied.</li>
        </ul>
      </li>
      <li>Refactored how chip summary metrics are propigated into output files.
        There is now a ChipSummary interface which various classes can
        implement. These classes can then be registered with certain reporters
        (ie Run Report and AGCC CHP reporters) as sources of chip summary
      info.</li>
    </ul>
  </li>
</ul>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.8.0 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Tue May 29 2007</dd>
</dl>
<ul>
  <li>New Applications: 
    <ul>
      <li>apt-geno-qc: a tool for computing single chip QC metrics for WGSA
        (500k, Genome Wide SNP5, and Genome Wide SNP6) genotyping chips</li>
    </ul>
  </li>
</ul>
<ul>
  <li>Changed Applications: 
    <ul>
      <li>apt-probeset-summarize: 
        <ul>
          <li><b>Fixed overflow error in accumulator for DABG values (previous
            behavior affected probesets with lots of probes where the DABG
            p-value with go to 1 rather than 0.</b></li>
          <li><b>fixed bug related to non-square CEL files</b> where the
            mapping between x/y coordinates and probe IDs was incorrect</li>
          <li>atom_id field in the feature response and feature residual files
            now contain "0". Atom IDs are no longer retained in the analysis
            software in order to reduce the memory requirements. As a result,
            the information is not available for reporting into this output
            file.</li>
          <li>added ability to treat genotyping and copy number probesets as
            expression probesets. See "--explain expr".</li>
          <li>added pca-select and spect-select feature selection methods
            (intended for use in computing gene level estimates from WT Gene
            and Exon arrays). See "--explain pca-select" and "--explain
            spect-select".</li>
          <li>enable use of meta probeset files (mps) with cdf files</li>
          <li>added PM+MM (pm-sum) adjustor. See "--explain pm-sum".</li>
          <li>added median and mean summarization methods. See "--explain
            median".</li>
        </ul>
      </li>
      <li>apt-probeset-genotype: 
        <ul>
          <li><b>fixed bug related to non-square CEL files</b> where the
            mapping between x/y coordinates and probe IDs was incorrect</li>
          <li>added birdseed method for genotyping calling (associated with
            SNP6)</li>
          <li>added brlmm-p-plus method for genotyping calling (associated with
            SNP6)</li>
          <li>added brlmm-p method for genotyping calling (associated with
          SNP5)</li>
        </ul>
      </li>
    </ul>
  </li>
</ul>
<ul>
  <li>Source Code 
    <ul>
      <li>reduce memory requirements of apt-probeset-summarize and
        apt-probeset-genotype by using a new internal structures for chip
        layout information</li>
      <li>added support for an experimental chip layout format (SPF)</li>
      <li>new build system with multiple static libraries</li>
    </ul>
  </li>
</ul>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.6.0 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Fri Sep 29 2006</dd>
</dl>
<ul>
  <li>General 
    <ul>
      <li>Upgraded to Newmat 10D</li>
      <li>Added preliminary support for Command Console Files (CEL and CHP)
        note that not all applications support CC CEL and CHP files yet. See
        application specific notes below for which applications now support CC
        files.</li>
    </ul>
  </li>
</ul>
<ul>
  <li><b>Change to the PLIER algorithm</b> 
    <ul>
      <li>The PLIER source code contained in this release is replacing the
        separate stand alone PLIER SDK.</li>
      <li>The PLIER code was changed by the addition of three new options: 
        <ul>
          <li>SafetyZero: Within the PLIER code zero values are now adjusted by
            a slight positive bias (default 0.000001) to prevent errors during
            log transformation. This is particularly relevant when feature
            response values are rounded to zero and feed into PLIER as
            precomputed feature effects.</li>
          <li>NumericalTolerance: Sets how hard the PLIER algorithm will work
            to resolve small differences between results when using precomputed
            vs non-precomputed feature responses. (Default is 0.1)</li>
          <li>FixPrecomputed: This option causes the PLIER code to recompute
            the target response after fitting the feature response. This
            improves concordance between PLIER results generated de novo versus
            those generated using a precomputed feature response.</li>
        </ul>
      </li>
      <li>These changes improve the stability of PLIER estimates when comparing
        those generated using precomputed feature responses versus those
        generated using full PLIER model fit. Note that in this release of APT
        you may still see some significant differences in cases where the
        likelihood space is relatively flat combined with rounding errors
        caused by writing out floating point values to the text feature
        response file.</li>
    </ul>
  </li>
</ul>
<ul>
  <li><b>Change sketch normalization default sketch size</b> 
    <ul>
      <li>Sketch normalization (ie when you request quant-norm and do not
        specify a sketch size) will now default to a sketch size of the 1% of
        the features or 50,000 - which ever is higher. If there are fewer than
        50,000 features, then all the features will be used (ie full quantile
        normalization).</li>
    </ul>
  </li>
</ul>
<ul>
  <li>New Applications: 
    <ul>
      <li>apt-cel-extract: extract intensities from CEL files</li>
      <li>apt-chp-to-txt: extract values from CHP files</li>
      <li>apt-cel-to-pq: dump out mapping genotype probeset intensities as
        quartet</li>
      <li>apt-dump-pgf: extract/dump out information in pgf and clf library
        files</li>
      <li>apt-summary-vis: create IGB and BED visualization files from
        summary.txt files</li>
      <li>apt-cdf-export: extract/dump out information in the cdf library
      files</li>
      <li>apt-tsv-join: merge text files</li>
      <li>apt-matrix-diff: compare two text files w/ matrix of values</li>
    </ul>
  </li>
</ul>
<ul>
  <li>Changed Applications: 
    <ul>
      <li>apt-midas: 
        <ul>
          <li><b>Changed -c option to --cel-files for consistency with other
            APT apps.</b></li>
          <li><b>Changed header in file grouping CEL files from cel_file to
            cel_files for consistency with other APT apps.</b></li>
        </ul>
      </li>
      <li>apt-probeset-summarize: 
        <ul>
          <li>meta probeset files are now specified using a separate option, -m
            or --meta-probesets rather than by using the -s option as
          before.</li>
          <li>change --chp-output option to --xda-chp-output</li>
          <li>change output/input files to use probe_id rather than feature_id
            for improved consistency with terminology used in various library
            files</li>
          <li>apt-probeset-summarize will now try and guess an appropriate
            block size given the amount of available memory to reduce the
            change of out of memory errors and disk swap</li>
          <li>support for Comand Console (CC) CHP file output and CEL file
          input</li>
          <li>rm-probe option for masking out probes</li>
          <li>transposed report file contents to be consistent with
            apt-probeset-genotype</li>
          <li>reports associated with each quantification method such that the
            use of multiple analysis parameters (-a) will result in multiple
            reports</li>
          <li>reports are always generated -- removed --report option</li>
          <li>added an avgdiff and sea methods</li>
          <li>added md5sum of probes used for GC background correction</li>
          <li>added md5sum of probes used for computing quantile normalization
            sketch</li>
          <li>refactored memory allocation to improve memory use on Windows
            where the address space is more fragmented</li>
          <li>reduced memory used by the quantile normalization (including
            sketch) code</li>
        </ul>
      </li>
    </ul>
  </li>
</ul>
<ul>
  <li>apt-probeset-genotype: 
    <ul>
      <li>change --chp-output option to --xda-chp-output</li>
      <li>rm-probe option for masking out probes</li>
      <li>automatic blocking of probesets to improve usability on low memory
        machines</li>
      <li>support for Comand Console (CC) CHP file output and CEL file input
        (CC genotype CHP file content is currently experimental and will
      change)</li>
      <li>changed Unknown to UnknownGender</li>
      <li>for xda CHP output, report median raw PM values for controls (in
        order to be consistent with CHP files generated by GTYPE)</li>
      <li>the report file now had a column header for the cel file name,
        "cel_files"</li>
    </ul>
  </li>
</ul>
<ul>
  <li>Binary Releases 
    <ul>
      <li>Added Mac OS-X Universal/Fat Binaries for i386/ppc/ppc64</li>
      <li>Removed dependency on .NET for windows binaries</li>
    </ul>
  </li>
</ul>
<ul>
  <li>Source Code 
    <ul>
      <li>draft file format documentation</li>
      <li>refactor common code out of the application specific engines</li>
      <li>refactor the QC reports to allow for probe and probeset level
      metrics</li>
      <li>formalizing structure of options that are used by chipstream via
        SelfDoc::Opt class which contains a name, description, default and
        range values for each available option.</li>
      <li>include Command Console file parsing code</li>
      <li>allow for serialization of engine parameters</li>
      <li>clean up code to quell gcc warnings</li>
      <li>enable building fat/universal binaries on OS-X</li>
      <li>cancel engines is now done through the ProgressHandlers</li>
      <li>refactor general enumeration and types not specific to any one class
        or method</li>
    </ul>
  </li>
</ul>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.4.0 </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Wed Apr 26 2006</dd>
</dl>

<p>This is the first stand alone release of the Analysis Power Tools (APT).
Previous versions were released as part of the Exon Array Computational Tool
(ExACT).</p>
<ul>
  <li>General 
    <ul>
      <li>The default APT license is now GPL. For commercial override contact
        <a href="mailto:devnet@affymetrix.com">devnet@affymetrix.com</a>.</li>
    </ul>
  </li>
</ul>
<ul>
  <li>New Applications 
    <ul>
      <li>apt-cel-transformer: create new CEL files with various data
        transformations applied (ie normalization)</li>
      <li>apt-snp-compare: simple tool for comparing matrix of SNP calls</li>
    </ul>
  </li>
</ul>
<ul>
  <li>Changed Applications 
    <ul>
      <li>apt-probeset-genotype: implements the BRLMM genotype calling
        algorithm for mapping arrays</li>
      <li>apt-probeset-summarize: generate probeset signal estimates from
        expression and exon arrays 
        <ul>
          <li>changed name to include apt prefix</li>
          <li>change feature response and residual output (and input) files now
            have probeset_id, atom_id, feature_id (aka probe_id), x, and y
            values.</li>
          <li>change default precision of output (0 places for most files, 5
            digits for dabg p values)</li>
        </ul>
      </li>
      <li>Broken out main probeset-genotype and probeset-summarize into a core
        engine and a separate thin command line wrapper.</li>
      <li>Refactor status and log reporting to facilitate GUI integration</li>
      <li>Migrate windows builds to Visual Studio 2005</li>
      <li>New build system for *NIX systems</li>
      <li>More doxygen high level and API documentation</li>
    </ul>
  </li>
</ul>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.2.1 (as part of ExACT 1.2.1) </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Mon Jan 30 2006</dd>
</dl>

<p>Minor changes on the C++ side to support re-applying a multi-chip PLIER
analysis to new chips.</p>
<ul>
  <li>Changed Applications 
    <ul>
      <li>probeset-summarize 
        <ul>
          <li>added option to use pre-computed normalization sketches and
            precomputed feature responses</li>
          <li>added option to save sketches and precomputed feature
          responses</li>
          <li><b>WARNING, because of the way the PLIER algorithm alternates
            optimizing the signal estimate and the feature responses, using
            precomputed feature responses will result in a slightly different
            answer when the same CEL files are used.</b></li>
        </ul>
      </li>
    </ul>
  </li>
</ul>
<dl class="version">
  <dt><b>Version:</b></dt>
    <dd>apt-1.2.0 (as part of ExACT 1.2.0) </dd>
</dl>
<dl class="date">
  <dt><b>Date:</b></dt>
    <dd>Mon Jan 02 2006</dd>
</dl>

<p>This was essentially the first release of APT. Most of the C++ code was
experimental and thus subject to change. You may try the <a
href="mailto:devnet@affymetrix.com">devnet@affymetrix.com</a> mail address or
the DevNet ExACT message forum on Affymetrix.com for unofficial support.</p>
<ul>
  <li>New Applications 
    <ul>
      <li>midas: compute the MiDAS F stat and P values for alternative
        splicing, the MiDAS method is described in a whitepaper available <a
        href="http://www.affymetrix.com/support/technical/byproduct.affx?product=exact">here</a>.</li>
      <li>apt-probeset-summarize: generate probeset signal estimates from
        expression and exon arrays 
        <ul>
          <li>does normalization and signal summarization in a single step</li>
          <li>includes RMA, IterPLIER, and PLIER</li>
          <li>experimental</li>
        </ul>
      </li>
    </ul>
  </li>
</ul>
<ul>
  <li>Source Code 
    <ul>
      <li>Added C++ DABG implementation. See the source code under
      sdk/dabg.</li>
      <li>Added experimental C++ implementation of the MiDAS altsplice
        detection method. See the source code under sdk/midas.</li>
      <li>Added experimental C++ ChipStream code. This code provides a
        framework to apply multiple transformations in a multichip analysis.
        See code under sdk/chipstream. 
        <ul>
          <li>A command line program using ChipStream, probeset-summarize, is
            also provided. This program can do normalization and probe
            summarization (ie PLIER, RMA, ...) in a single go and can do so
            with either a CDF file or a PGF/CLF files. See code under
            sdk/chipstream/probeset-summarize.</li>
        </ul>
      </li>
      <li>Added a separate build system for the C++ code. The makefile will
        build a simple Affymetrix SDK static library and the chip stream
        (probeset-summarize) and midas executables. If you have CPP Unit
        installed, some basic unit testing will also be done.</li>
    </ul>
  </li>
</ul>

<p><b></b></p>
<pre>Analysis Power Tools (APT) Release 2.10.2.2</pre>

<p></p>
</div>
<hr class="footer" />
<address class="footer">
  <small>Generated on Sun Jun 24 2018 13:29:25 for Analysis Power Tools by  <a
  href="http://www.doxygen.org/index.html"><img class="footer"
  src="doxygen.png" alt="doxygen" /></a> 1.7.1 </small> 
</address>
</div>
</div>
</body>
</html>
