<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Analysis Power Tools: MANUAL: ps-classification (2.10.2)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div class="header">
  <div class="headertitle">
<div class="title"><h1>MANUAL: ps-classification (2.10.2) </h1></div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><h2>Contents </h2>
<ul>
<li><a href="#intro">Introduction.</a></li>
<li><a href="#quickstart">Quick Start</a></li>
<li><a href="#options">Options</a></li>
</ul>
<p><a class="anchor" id="intro"></a> </p>
<h2>Introduction </h2>
<p><b>ps-classification</b> reads in the SNP QC metrics tables generated by ps-metrics and performs SNP classification based on customizable criteria, including call rate, FLD, HetSO, and HomRO. It classifies SNPs into seven major categories: "PolyHighResolution","NoMinorHom", "MonoHighResolution", "OffTargetVariant", "CallRateBelowThreshold", "Other", and "UnexpectedGenoFreq". The category "Hemizygous" appears only when hemizygous SNPs are in the metrics file but the special SNPs data is not present in the metrics file. ps-classification also identifies the best probeset per SNP. More details describing the use of ps-classification can be found in the  Axiom Genotyping Solution Data Analysis Guide, available at https://tools.thermofisher.com/content/sfs/manuals/axiom_genotyping_solution_analysis_guide.pdf..</p>
<p><a class="anchor" id="quickstart"></a> </p>
<h2>Quick Start </h2>
<p>This command runs ps-classification on a metrics file named metrics.txt  in the directory ./input for a human dataset and writes the classification output to the directory ./classification: </p>
<pre class="fragment">ps-classification --species-type human --metrics-file ./input/metrics.txt --output-dir ./classification
</pre><p><a class="anchor" id="details"></a> </p>
<h2>Details </h2>
<p>The input files and options for ps-classification are:</p>
<ul>
<li>metrics-file - The name of the biallelic metrics file generated by ps-metrics</li>
<li>multi-metrics-file - The name of the multiallelic metrics file generated by ps-metrics</li>
<li>ps2snp-file - The name of a file containing the mapping of probeset IDs to SNP IDs. The column names should be "probeset_id" and "snpid". The user must supply this file when classifying de novo SNPs or any SNPs that have more than one probeset. (Optional)</li>
<li>output-dir - Output directory for results files: defaults to current directory.</li>
<li>pid-file - The list of probesets that ps-classification should run on. (Optional)</li>


<p>The classification thresholds accepted by ps-classification are:</p>

<li>species-type - The type of organism genotyped. The current options are "human", non-human "diploid", and "polyploid". Default is "human".</li>
<li>output-recommended - Logical value indicating whether to output a list of recommended SNPs to recommended.ps. Default is TRUE.</li>
<li>recommended - Flag indicating whether to output a list of converted SNPs to a file recommened.ps: default is FALSE.</li>

<p>If a biallelic metrics file is provided, the following classification thresholds are used:</p>

<li>cr-cutoff - Threshold for call rate. If not specified, the default for human is 95 and for diploid and polyploid is 97.</li>
<li>fld-cutoff - Threshold for FLD: default=3.6.</li>
<li>het-so-cutoff - Threshold for HetSO: default=-0.1.</li>
<li>het-so-otv-cutoff - Threshold for OTV detection: default=-0.3.</li>
<li>hom-ro-1-cutoff - Threshold for HomRO when a SNP has one genotype: default=0.6.</li>
<li>hom-ro-2-cutoff - Threshold for HomRO when a SNP has two genotypes: default=0.3.</li>
<li>hom-ro-3-cutoff - Threshold for HomRO when a SNP has three genotypes: default=-0.9.</li>
<li>hom-ro - Flag indicating whether the metric HomRO is used in classification: default is TRUE.</li>
<li>hom-het - Flag indicating whether the metric HomHet is used in classification. The HomHet metric identifies two-cluster SNPs/probesets with one homozygote cluster and one heterozygote cluster. This checks if the minor homozygote cluster is missing, which is unreasonable for highly inbred species (e.g. wheat). This metric should be turned on when classifying SNPs/probesets in highly inbred species: default is TRUE.</li>
<li>num-minor-allele-cutoff - Threshold for the number of minor alleles: default=2.</li>
<li>priority-order - When performing probeset selection, the best probeset is selected according to the priority order of probeset conversion types. Valid values are sequences of the strings "PolyHighResolution", "NoMinorHom", "OTV", "MonoHighResolution","CallRateBelowThreshold", and "UnexpectedGenoFreq", separated by commas, with each string occuring exactly once: default 'PolyHighResolution,NoMinorHom,OTV,MonoHighResolution,CallRateBelowThreshold'</li>
<li>genotype-p-value-cutoff - minimum p-value for a probeset to be recommended instead of UnexpectedGenotypeFrequency: default=0.000001.</li>

<p>If the biallelic metrics file contains copy-number aware metrics, the following classification thresholds are used with SNPs with copy-number aware calls:</p>

<li>hom-ro-hap-1-cutoff - Threshold for HomRO when a SNP has one haploid genotype: default=0.4.</li>
<li>hom-ro-hap-2-cutoff - Threshold for HomRO when a SNP has one haploid genotype: default=0.2.</li>
<li>hom-hap-X-cutoff - Maximum distance diploid clusters can be outside of haploid clusters in the X dimension: default=-0.2.</li>
<li>hom-hap-Y-lower-cutoff - Maximum distance diploid clusters can be below haploid clusters in the Y dimension: default=1.</li>
<li>hom-hap-Y-upper-cutoff - Maximum distance haploid clusters can be above diploid clusters in the Y dimension: default=0.2.</li>
<li>CN0-hap-X-cutoff - Maximum distance CN0 clusters can be outside of haploid clusters in the X dimension: default=-0.2.</li>
<li>CN0-hap-Y-cutoff - Maximum distance CN0 clusters can be above haploid clusters in the Y dimension: default=0.</li>
<li>CN0-dip-X-cutoff - Maximum distance CN0 clusters can be outside of diploid clusters in the X dimension: default=0.2.</li>
<li>CN0-dip-Y-cutoff  - Maximum distance CN0 clusters can be above diploid clusters in the Y dimension: default=0.</li>

<p>If the supplemental filter is turned on, the following classification thresholds are used with SNPs with supplemental metrics:</p>

<li>use-supplemental - Flag indicating if the supplemental classification should be performed: default=FALSE.</li>
<li>homfld-cut - Minimum HomFLD value for a probeset to be PHR instead of HomHomResolution: default=6.5.</li>
<li>n-minor-hom-cut - minimum number of samples in the minor homozygous cluster to be recommended instead of nMinorHom: default=0.</li>
<li>hetv-maf-cut - cut-off for het rate versus MAF for UnexpectedHeterozygosity: default=10.</li>
<li>variance-class - category types that the supplemental workflow is run on. Takes a single category or a comma separated list: default='PolyHighResolution'</li>
<li>bb-varx-cut - minimum X variance value for the BB cluster to be recommended instead of BBvarianceX: default=10.</li>
<li>bb-vary-cut - minimum Y variance value for the BB cluster to be recommended instead of BBvarianceY: default=10.</li>
<li>ab-varx-cut - minimum X variance value for the AB cluster to be recommended instead of ABvarianceX: default=10.</li>
<li>ab-vary-cut - minimum Y variance value for the AB cluster to be recommended instead of ABvarianceY: default=10.</li>
<li>aa-varx-cut - minimum X variance value for the AA cluster to be recommended instead of AAvarianceX: default=10.</li>
<li>aa-vary-cut - minimum Y variance value for the AA cluster to be recommended instead of AAvarianceY: default=10.</li>
<li>bb-varx-z-cut - minimum z-score X variance value for the BB cluster to be recommended instead of BBvarianceX: default=3.</li>
<li>bb-vary-z-cut - minimum z-score Y variance value for the BB cluster to be recommended instead of BBvarianceY: default=3.</li>
<li>ab-varx-z-cut - minimum z-score X variance value for the AB cluster to be recommended instead of ABvarianceX: default=3.</li>
<li>ab-vary-z-cut - minimum z-score Y variance value for the AB cluster to be recommended instead of ABvarianceY: default=3.</li>
<li>aa-varx-z-cut - minimum z-score X variance value for the AA cluster to be recommended instead of AAvarianceX: default=3.</li>
<li>aa-vary-z-cut - minimum z-score Y variance value for the AA cluster to be recommended instead of AAvarianceY: default=3.</li>
<li>bb-varx-z-cut-enabled - flag indicating if the Z thresholds should be on the BB X variances used instead of the variance thresholds with the supplemental workflow: default=TRUE.</li>
<li>bb-vary-z-cut-enabled - flag indicating if the Z thresholds should be on the BB Y variances used instead of the variance thresholds with the supplemental workflow: default=TRUE.</li>
<li>ab-varx-z-cut-enabled - flag indicating if the Z thresholds should be on the AB X variances used instead of the variance thresholds with the supplemental workflow: default=TRUE.</li>
<li>ab-vary-z-cut-enabled - flag indicating if the Z thresholds should be on the AB Y variances used instead of the variance thresholds with the supplemental workflow: default=TRUE.</li>
<li>aa-varx-z-cut-enabled - flag indicating if the Z thresholds should be on the AA X variances used instead of the variance thresholds with the supplemental workflow: default=TRUE.</li>
<li>aa-vary-z-cut-enabled - flag indicating if the Z thresholds should be on the AA Y variances used instead of the variance thresholds with the supplemental workflow: default=TRUE.</li>

<p>If the SSPs file is turned on, ps-classification performs the supplemental workflow and the output the SSPs metrics along with the supplemental metrics:</p>

<li>use-ssp - flag indicating if the SSP workflow should be run: default=FALSE.</li>

<p>If the biallelic metrics file contains metrics for special SNPs, the following classifiation thresholds are used with non-PAR X, Y, and MT SNPs:</p>

<li>hom-ro-hap-1-MTChr-cutoff - minimum HomRO value for MT probesets with 1 haploid cluster: default=0.4.</li>
<li>hom-ro-hap-2-MTChr-cutoff - minimum HomRO value for MT probesets with 2 haploid clusters: default=0.2.</li>
<li>hom-ro-hap-1-YChr-cutoff - minimum HomRO value for Y probesets with 1 haploid cluster: default=0.4.</li>
<li>hom-ro-hap-2-YChr-cutoff - minimum HomRO value for Y probesets with 2 haploid clusters: default=0.2</li>
<li>hom-ro-hap-1-XChr-cutoff - minimum HomRO value for non-PAR X probesets with 1 haploid cluster (male samples only): default=0.1.</li>
<li>hom-ro-hap-2-XChr-cutoff - minimum HomRO value for non-PAR X probesets with 2 haploid clusters (male samples only): default=0.05.</li>
<li>homfld-YChr-cut - minimum HomFLD_hap value for a Y probeset to be PHR instead of Other (male samples only): default=6.5.</li>
<li>homfld-XChr-cut - minimum HomFLD_hap value for a non-PAR X probeset to be PHR instead of Other (male samples only): default=6.5.</li>
<li>fld-XChr-cut - minimum FLD should be larger than FLD_X.cut for non-PAR X probesets (female samples only): default=4.0.</li>
<li>aaf-XChr-cut - maximum difference in A-allele frequency between female and male samples for non-PAR X probesets: default=0.36</li>
<li>CN0-hap-X-YChr-cutoff - maximum distance CN0 clusters can be outside of haploid clusters in the X dimension for Y probesets: default=0.1.</li>
<li>CN0-hap-Y-YChr-cutoff - maximum distance CN0 clusters can be above haploid clusters in the Y dimension for Y probesets (deprecated): default=0.35.</li>
<li>het-so-XChr-cutoff - minimum HetSO should be larger than HetSO_X.cut for non-PAR X probesets (female samples only): default=-0.1.</li>
<li>min-YChr-samples-cut - minimum number of male and female samples needed to classify Y probesets using the Euclidean distances thresholds: default=15.</li>

<p>If a multiallelic metrics file is provided, the following classification thresholds are used:</p>

<li>use-multi-allele - flag indicating if multiallelic classification should be performed: default=FALSE</li>
<li>sign-diff-hom-1-cutoff - minimum minSigBgndDiffHom for probesets with 1 cluster: default=0.5.</li>
<li>sign-diff-hom-2-cutoff - minimum minSigBgndDiffHom for probesets with 2 clusters: default=0.4.</li>
<li>min-mean-cp2-cutoff - minimum MinMean_cp_2 value needed to be PHR: default=9.</li>
<li>max-mean-cp2-cutoff - maximum MinMean_cp_2 value needed to be PHR: default=15.</li>
<li>genotype-p-value-cutoff - minimum p-value for a probeset to be recommended instead of UnexpectedGenotypeFrequency: default=0.000001.</li>
<p>ps-classification produces a performance summary output file (Ps.performance.txt) and several output files, one for each category, with a list of probesets that have been classified in each category.  The category probeset files are named after the category (e.g. NoMinorHom.ps). The performance file contains the metrics for each probeset, the probeset’s category (ConversionType), and if the probeset is the best probeset for the SNP. If there are multiple probesets for a SNP, then a ps2snp file must be provided.</p>

<li>metrics-file</b> - The name of the file containing output generated by ps-metrics.</li>
<li>ps2snp-file</b> (Optional) - The name of a file containing the mapping of probeset IDs to SNP IDs. The column names should be "probeset_id" and "snpid". The user must supply this file when classifying de novo SNPs or any SNPs that have more than one probeset.</li>
<li>output-dir</b> - Output directory for results files: defaults to current directory.</li>
<li>species-type</b> - The type of organism genotyped. The current options are "human", non-human "diploid", and "polyploid".</li>
<li>converted</b> - Flag indicating whether to output a list of converted SNPs to a file <code>converted.ps</code>: default is FALSE.</li>
<li>cr-cutoff</b> - Threshold for call rate. If not specified, the default for human is 95 and for diploid and polyploid is 97.</li>
<li>fld-cutoff</b> - Threshold for FLD: default=3.6.</li>
<li>het-so-cutoff</b> - Threshold for HetSO: default=-0.1.</li>
<li>het-so-otv-cutoff</b> - Threshold for OTV detection: default=-0.3.</li>
<li>hom-ro-1-cutoff</b> - Threshold for HomRO when a SNP has one genotypes: default=0.6.</li>
<li>hom-ro-2-cutoff</b> - Threshold for HomRO when a SNP has two genotypes: default=0.3.</li>
<li>hom-ro-3-cutoff</b> - Threshold for HomRO when a SNP has three genotypes: default=-0.9.</li>
<li>hom-ro</b> - Flag indicating whether the metric HomRO is used in classification: default is TRUE.</li>
<li>hom-het</b> - Flag indicating whether the metric HomHet is used in classification. The HomHet metric identifies two-cluster SNPs/probesets with one homozygote cluster and one heterozygote cluster. This checks if the minor homozygote cluster is missing, which is unreasonable for highly inbred species (e.g. wheat). This metric should be turned on when classifying SNPs/probesets in highly inbred species: default is TRUE.</li>
<li>num-minor-allele-cutoff</b> - Threshold for the number of minor alleles: default=2.</li>
<li>priority-order</b> - When performing probeset selection, the best probeset is selected according to the priority order of probeset conversion types. Valid values are sequences of the strings <em>"PolyHighResolution"</em>, <em>"NoMinorHom"</em>, <em>"OTV"</em>, <em>"MonoHighResolution"</em> and <em>"CallRateBelowThreshold"</em>, separated by commas, with each string occuring exactly once: default <code>'PolyHighResolution,NoMinorHom,OTV,MonoHighResolution,CallRateBelowThreshold'</code></li>
</ul>
<p>The metric <code>hom-ro</code> shoud be turned off for organisms where the homozygote clusters may center at zero in the contrast space, particularly polyploid species. If a <code>ps2snp-file</code> is provided, then the best performance probeset will be selected to represent SNPs with multiple probesets. There are seven output files, one for each category, containing lists of probeset IDs belonging to each category. Each file contains a list of probeset IDs belonging to the category. The performance output file summarizes the categorization results across all SNPs.</p>
<p><a class="anchor" id="options"></a> </p>
<h2>Options: </h2>
<pre class="fragment">
   'ps-classification'
   2.10.2 stash.amer.thermo.com/ma-alg/apt2-snppolisher.git/release/APT_2.10.2@5b62fc9457(75b176a197) 2018-08-25-06:31


USAGE:
     ps-classification --species-type [human|polyploid|diploid] --metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; [OPTIONS]
	 
	OR
	 
	ps-classification --species-type [human|polyploid|diploid] --metrics-file &lt;FILE&gt --multi-metrics-file &lt;FILE&gt --output-dir &lt;DIR&gt [OPTIONS]
	
	OR
	
	ps-classification --species-type [human|polyploid|diploid] --multi-metrics-file &lt;FILE&gt --output-dir &lt;DIR&gt [OPTIONS]
USER HELP ----------

  --version (flag) = Display the version info for this program.

  --user-help (flag) = Display help intended for end-users.
    Aliases: --help, -h

  --log-file (string8) = The name of the log file.  Defaults to 'PROGRAMNAME-NUMBER.log'.
    Aliases: --lf

  --console-add-select (string8) = Add selectors for console messages.
    e.g. to include all debug messages: --console-add-select debug
         to include all messages of level 3 or higher: --console-add-select '*:3'


  --console-add-neg-select (string8) = Add selectors to be excluded from console messages.
    e.g. to exclude all warning messages and errors summary: --console-add-neg-select WARNING,summary


  --arg-file (filename) = Read arguments from this file.
    File should be a valid XML file, and should contain elements of the form
    <Parameter name="PARAMETER-NAME" analysis="NODE-NAME" currentValue="VALUE" />.

  --check-param (flag) = Stop the program after the command parameters have been checked.

  --metrics-file (filename) = Full path name for metrics file generated by ps-metrics.

  --species-type (string8,required) = Type of organism to be genotyped. Valid options are 'Human','Diploid' or 'Polyploid'.

  --report-file-ext (string8) = File extension for the ps report files
    default value: 'ps'

  --ps2snp-file (filename) = Full path name of file mapping probeset IDs to SNP IDs (Recommended).

  --output-dir (dirname) = Path to output directory.

  --output-recommended (bool) = Logical value indicating whether to output a list of recommended SNPs to recommended.ps. Default is TRUE. See the SNPolisher User Guide for more information about SNP conversion.
    default value: 'True'

  --recommended (vec_string8) = List of categories whose SNPs will be output as recommended. If output.recommended is TRUE, the default value for Human is PolyHighResolution, MonoHighResolution, NoMinorHom, Hemizygous, and for Diploid or Polyploid the default is PolyHighResolution.

  --cr-cutoff (float64) = Threshold for call rate. Default is 95.0 for Human, 97.0 for Diploid or Polyploid.
    default value: '95.000000'

  --fld-cutoff (float64) = Threshold for FLD
    default value: '3.600000'

  --het-so-cutoff (float64) = Threshold for HetSO
    default value: '-0.100000'

  --het-so-otv-cutoff (float64) = Threshold for HetSO OTV
    default value: '-0.300000'

  --hom-ro-1-cutoff (float64) = Threshold for HomRO when SNP has 1 genotype
    default value: '0.600000'

  --hom-ro-2-cutoff (float64) = Threshold for HomRO when SNP has 2 genotype
    default value: '0.300000'

  --hom-ro-3-cutoff (float64) = Threshold for HomRO when SNP has 3 genotype
    default value: '-0.900000'

  --hom-ro (flag) = True if HomRO metric to be used in classification

  --hom-het (flag) = True if Hom Het metric to be used in classification

  --num-minor-allele-cutoff (float64) = Threshold for number of minor alleles
    default value: '2.000000'

  --hom-ro-hap-1-cutoff (float64) = Threshold for HomRO Haploid 1
    default value: '0.400000'

  --hom-ro-hap-2-cutoff (float64) = Threshold for HomRO Haploid 2
    default value: '0.200000'

  --hom-hap-X-cutoff (float64) = Threshold for Hom-X  Haploid
    default value: '-0.200000'

  --hom-hap-Y-lower-cutoff (float64) = Threshold for Hom-Y  Haploid  lower
    default value: '1.000000'

  --hom-hap-Y-upper-cutoff (float64) = Threshold for Hom-Y  Haploid upper
    default value: '0.200000'

  --CN0-hap-X-cutoff (float64) = Threshold for CN0-X  Haploid
    default value: '-0.200000'

  --CN0-hap-Y-cutoff (float64) = Threshold for CN0-Y  Haploid
    default value: '0.000000'

  --CN0-dip-X-cutoff (float64) = Threshold for CN0-X  Diploid
    default value: '0.200000'

  --CN0-dip-Y-cutoff (float64) = Threshold for CN0-Y  Diploid
    default value: '0.000000'

  --priority-order (vec_string8) = Priority order of probeset conversion types when performing probeset selection.
    default value: 'PolyHighResolution,NoMinorHom,OTV,MonoHighResolution,UnexpectedGenotypeFreq,CallRateBelowThreshold'

  --pid-file (filename) = Full path name of pid file.
    default value: 'pid.ps'

  --use-supplemental (bool) = Turn on supplemental filters. default: false, else: true.
    default value: 'False'

  --homfld-cut (float64) = HomFLD.cut
    default value: '6.500000'

  --n-minor-hom-cut (float64) = nMinorHom.cut
    default value: '0.000000'

  --hetv-maf-cut (float64) = HetvMAF.cut
    default value: '10.000000'

  --variance-class (vec_string8) = List of variance.class categories. Allows a single category or a comma separated list
    default value: 'PolyHighResolution'

  --bb-varx-cut (float64) = BB.varX.cut
    default value: '10.000000'

  --bb-vary-cut (float64) = BB.varY.cut
    default value: '10.000000'

  --ab-varx-cut (float64) = AB.varX.cut
    default value: '10.000000'

  --ab-vary-cut (float64) = AB.varY.cut
    default value: '10.000000'

  --aa-varx-cut (float64) = AA.varX.cut
    default value: '10.000000'

  --aa-vary-cut (float64) = AA.varY.cut
    default value: '10.000000'

  --bb-varx-z-cut (float64) = BB.varX.Z.cut
    default value: '3.000000'

  --bb-vary-z-cut (float64) = BB.varY.Z.cut
    default value: '3.000000'

  --ab-varx-z-cut (float64) = AB.varX.Z.cut
    default value: '3.000000'

  --ab-vary-z-cut (float64) = AB.varY.Z.cut
    default value: '3.000000'

  --aa-varx-z-cut (float64) = AA.varX.Z.cut
    default value: '3.000000'

  --aa-vary-z-cut (float64) = AA.varY.Z.cut
    default value: '3.000000'

  --aa-vary-z-cut-enabled (bool) = AA.varY.Z.cut.enabled
    default value: 'True'

  --ab-vary-z-cut-enabled (bool) = AB.varY.Z.cut.enabled
    default value: 'True'

  --bb-vary-z-cut-enabled (bool) = BB.varY.Z.cut.enabled
    default value: 'True'

  --aa-varx-z-cut-enabled (bool) = AA.varX.Z.cut.enabled
    default value: 'True'

  --ab-varx-z-cut-enabled (bool) = AB.varX.Z.cut.enabled
    default value: 'True'

  --bb-varx-z-cut-enabled (bool) = BB.varX.Z.cut.enabled
    default value: 'True'

  --use-ssp (bool) = Turn on SSP metrics. default: false, else: true.
    default value: 'False'

  --hom-ro-hap-1-MTChr-cutoff (float64) = Threshold for HomRO MT chromosome 1
    default value: '0.400000'

  --hom-ro-hap-2-MTChr-cutoff (float64) = Threshold for HomRO MT chromosome 2
    default value: '0.200000'

  --hom-ro-hap-1-YChr-cutoff (float64) = Threshold for HomRO Y chromosome 1
    default value: '0.400000'

  --hom-ro-hap-2-YChr-cutoff (float64) = Threshold for HomRO Y chromosome 2
    default value: '0.200000'

  --hom-ro-hap-1-XChr-cutoff (float64) = Threshold for HomRO X chromosome 1
    default value: '0.100000'

  --hom-ro-hap-2-XChr-cutoff (float64) = Threshold for HomRO X chromosome 2
    default value: '0.050000'

  --homfld-YChr-cut (float64) = HomFLD.cut for Y chromosome
    default value: '6.500000'

  --homfld-XChr-cut (float64) = HomFLD.cut for X chromosome
    default value: '6.500000'

  --fld-XChr-cut (float64) = FLD.cut for X chromosome
    default value: '4.000000'

  --aaf-XChr-cut (float64) = aaf.cut for X chromosome
    default value: '0.360000'

  --CN0-hap-X-YChr-cutoff (float64) = Threshold for CN0-X  Y chromosome
    default value: '0.100000'

  --CN0-hap-Y-YChr-cutoff (float64) = Threshold for CN0-Y  Y chromosome
    default value: '0.350000'

  --het-so-XChr-cutoff (float64) = Threshold for HetSO X Non-Par chromosome probesets
    default value: '-0.100000'

  --multi-metrics-file (filename) = Full path name for multi-allele Metrics file generated by ps-metrics.

  --use-multi-allele (bool) = Turn on multi-allele filters to support metrics file multi-allele version. default: false, else: true.
    default value: 'False'

  --sign-diff-hom-1-cutoff (float64) = signal background difference for homozygous clusters first cutoff
    default value: '0.500000'

  --sign-diff-hom-2-cutoff (float64) = signal background difference for homozygous clusters second cutoff
    default value: '0.400000'

  --min-mean-cp2-cutoff (float64) = minimal min-mean-cp-2 value cutoff
    default value: '9.000000'

  --max-mean-cp2-cutoff (float64) = maximal min-mean-cp-2 value cutoff
    default value: '15.000000'

  --genotype-p-value-cutoff (float64) = p-value clusters cutoff
    default value: '0.000001'

  --@help       = print developer help

</pre> </div></div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated for Analysis Power Tools by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
