<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Analysis Power Tools: MANUAL: ps-metrics (2.10.2)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div class="header">
  <div class="headertitle">
<div class="title"><h1>MANUAL: ps-metrics (2.10.2) </h1></div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><h2>Contents </h2>
<ul>
<li><a href="#intro">Introduction.</a></li>
<li><a href="#quickstart">Quick Start</a></li>
<li><a href="#details">Details</a></li>
<li><a href="#options">Options</a></li>
</ul>
<p><a class="anchor" id="intro"></a> </p>
<h2>Introduction </h2>
<p><b>ps-metrics</b> calculates quality control (QC) metrics for SNPs genotyped using Axiom arrays. ps-metrics takes standard genotyping outputs from Axiom arrays and generates a file containing QC metrics for all input SNPs. One metrics file is generated for each type of posteriors file: a biallelic posteriors file results in a biallelic metrics file, and a multiallelic posteriors file results in a multiallelic posteriors file. If the user provides both types of posteriors files, two metrics files are generated.  More details describing the use of ps-metrics and detailed explanations for each of the metrics reported can be found in the  Axiom Genotyping Solution Data Analysis Guide, available at https://tools.thermofisher.com/content/sfs/manuals/axiom_genotyping_solution_analysis_guide.pdf.</p>
<p><a class="anchor" id="quickstart"></a> </p>
<h2>Quick Start </h2>
<p>This command runs ps-metrics on the posteriors and calls files in directory ./input and generates the output file metrics.txt: </p>
<pre class="fragment">ps-metrics --posterior-file ./input/AxiomGT1.snp-posteriors.txt --call-file ./input/AxiomGT1.calls.txt --metrics-file metrics.txt
</pre><p><a class="anchor" id="details"></a> </p>
<h2>Details </h2>
<p>The following parameters are required to run <code>ps-metrics</code>:</p>
<ul>
<li><b>posterior-file</b> - The file output from Axiom genotyping which contains posterior summaries of clusters for SNPs (usually named "AxiomGT1.snp-posteriors.txt" or "AxiomGT1.snp-posteriors.multi.txt")..</li>
<li><b>metrics-file</b> - The name of the output metrics file. The user must supply a file name in order to save the output as a separate file, which is read in to other functions such as ps-classification (http://media.affymetrix.com/support/developer/powertools/changelog/Ps_Classification.html).</li>
</ul>
<p>In addition, one of the following two parameters is required:</p>
<ul>
<li><b>call-file</b> - The file output by GTC/APT which contains genotype calls for SNPs (usually named "AxiomGT1.calls.txt").</li>
<li><b>chp-files</b> - A text file containing a list of CHP files from which to take calls (one file path per line).</li>
</ul>
<p>Either the <code>--call-file</code> or the <code>--chp-files</code> parameter must be provided in order to run ps-metrics.</p>
<p>The biallelic output of <code>ps-metrics</code> is a tab delimited text file containing a single table with the following columns:</p>
<table class="doxtable">
<tr>
<th>Column Name </th><th>Values  </th></tr>
<tr>
<td>probeset_id </td><td>Probeset ID </td></tr>
<tr>
<td>CR </td><td>Call Rate </td></tr>
<tr>
<td>FLD </td><td>Fisherâ€™s Linear Discriminant (a metric which measures the separation of cluster centers) </td></tr>
<tr>
<td>HomFLD </td><td>Homozygote FLD (excluding the Heterozygote cluster) </td></tr>
<tr>
<td>HetSO </td><td>Heterozygous Cluster Strength Offset </td></tr>
<tr>
<td>HomRO </td><td>Homozygote Ratio Offset </td></tr>
<tr>
<td>nMinorAllele </td><td>Number of Minor Alleles </td></tr>
<tr>
<td>Nclus </td><td>Number of clusters </td></tr>
<tr>
<td>n_AA </td><td>Number of genotypes AA </td></tr>
<tr>
<td>n_AB </td><td>Number of genotypes AB </td></tr>
<tr>
<td>n_BB </td><td>Number of genotypes BB </td></tr>
<tr>
<td>n_NC </td><td>Number of null calls </td></tr>
<tr>
<td>hemizygous </td><td>Indicates if SNP is hemizygous (1 = hemizygous, 0 = not) </td></tr>
<td>H.W.statistic </td><td>Indicates if the Hardy-Weinberg p-value was calculated </td></tr>
<td>H.W.p-Value </td><td>The p-value generated from testing the Hardy-Weinberg statistic </td></tr>
</table>

<p>If copy-number aware genotype calls were assigned, the metrics file contains the additional columns:</p>
<table class="doxtable">
<tr>
<th>Column Name </th><th>Values  </th></tr>
<tr>
<td>HomFLD_hap </td><td>HomFLD calculated on haploid samples </td></tr>
<tr>
<td>HomRO_hap </td><td>HomRO calculated on haploid samples </td></tr>
<tr>
<td>n_A </td><td>Number of A genotypes</td></tr>
<tr>
<td>n_B </td><td>Number of B genotypes</td></tr>
<tr>
<td>n_CN0 </td><td>Number of copy-number zero genotypes</td></tr>
<tr>
<td>A.meanX</td><td>mean of the Contrast locations of the A genotypes</td></tr>
<tr>
<td>A.meanY</td><td>mean of the Size  locations of the A genotypes</td></tr>
<tr>
<td>A.varX</td><td>variance of the Contrast locations of the A genotypes</td></tr>
<tr>
<td>A.varY</td>	<td>variance of the Size locations of the A genotypes</td></tr>
<tr>
<td>B.meanX</td><td>mean of the Contrast locations of the B genotypes</td></tr>
<tr>
<td>B.meanY</td><td>mean of the Size  locations of the B genotypes</td></tr>
<tr>
<td>B.varX</td><td>variance of the Contrast locations of the B genotypes</td></tr>
<tr>
<td>B.varY</td><td>variance of the Size locations of the B genotypes</td></tr>
<tr>
<td>CN0.meanX</td><td>mean of the Contrast locations of the CN0 genotypes</td></tr>
<tr>
<td>CN0.meanY</td><td>mean of the Size  locations of the CN0 genotypes</td></tr>
<tr>
</table>

<p>If the summary file, special SNPs file, and report file were provided, the metrics file contains additional metrics with values calculated from the summary data only on the non-PAR X, Y, and MT SNPs. Non-PAR X and Y probesets have two row so metrics, one row for metrics calculated only on male samples and one row for metrics calculated only on female samples.:</p>
<table class="doxtable">
<tr>
<th>Column Name </th><th>Values  </th></tr>
<tr>
<td>AA.meanX </td><td>mean of the Contrast locations of the AA genotypes</td></tr>
<tr>
<td>AA.meanY </td><td>mean of the Size  locations of the AA genotypes</td></tr>
<tr>
<td>AA.varX </td><td>variance of the Contrast locations of the AA genotypes (non-PAR X  and Y only)</td></tr>
<tr>
<td>AA.varY </td><td>variance of the Size locations of the AA genotypes (non-PAR X  and Y only)</td></tr>
<tr>
<td>A.varX </td><td>variance of the Contrast locations of the A genotypes (non-PAR X  and Y only)</td></tr>
<tr>
<td>AB.meanX</td><td>mean of the Contrast locations of the AB genotypes</td></tr>
<tr>
<td>AB.meanY</td><td>mean of the Size  locations of the AB genotypes</td></tr>
<tr>
<td>AB.varX</td><td>variance of the Contrast locations of the AB genotypes (non-PAR X  only) (female samples only)</td></tr>
<tr>
<td>AB.varY</td><td>variance of the Size locations of the AB genotypes (non-PAR X  only) (female samples only)</td></tr>
<tr>
<td>BB.meanX</td><td>mean of the Contrast locations of the BB genotypes</td></tr>
<tr>
<td>BB.meanY</td><td>mean of the Size  locations of the BB genotypes</td></tr>
<tr>
<td>BB.varX</td><td>variance of the Contrast locations of the BB genotypes (non-PAR X  and Y only)</td></tr>
<tr>
<td>BB.varY</td><td>variance of the Size locations of the BB genotypes (non-PAR X  and Y only)</td></tr>
<tr>
<td>B.varX</td><td>variance of the Contrast locations of the B genotypes (non-PAR X  and Y only)</td></tr>
<tr>
<td>CN0.meanX</td><td>mean of the Contrast locations of the CN0 genotypes</td></tr>
<tr>
<td>CN0.meanY</td><td>mean of the Size  locations of the CN0 genotypes</td></tr>
<tr>
<td>NC.meanX</td><td>mean of the Contrast locations of the NC genotypes</td></tr>
<tr>
<td>NC.meanY</td><td>mean of the Size  locations of the NC genotypes</td></tr>
<tr>
<td>specialSNP_chr</td><td>Indicates the type of chromosome (X,  Y, MT, CHL, or autosomal)</td></tr>
<tr>
</table>
<p>Note: any clusters with fewer than clustermin samples are assigned a value of 0.03 for the variances.</p>
<p>If the summary file was provided and the supplemental option was used, the metrics file contains additional metrics with values calculated from the summary data:</p>
<table class="doxtable">
<tr>
<th>Column Name </th><th>Values  </th></tr>
<tr>
<td>AA.meanX </td><td>mean of the Contrast locations of the AA genotypes</td></tr>
<tr>
<td>AA.meanY </td><td>mean of the Size  locations of the AA genotypes</td></tr>
<tr>
<td>AA.varX </td><td>variance of the Contrast locations of the AA genotypes (non-PAR X  and Y only)</td></tr>
<tr>
<td>AA.varY </td><td>variance of the Size locations of the AA genotypes (non-PAR X  and Y only)</td></tr>
<tr>
<td>A.varX </td><td>variance of the Contrast locations of the A genotypes (non-PAR X  and Y only)</td></tr>
<tr>
<td>AB.meanX</td><td>mean of the Contrast locations of the AB genotypes</td></tr>
<tr>
<td>AB.meanY</td><td>mean of the Size  locations of the AB genotypes</td></tr>
<tr>
<td>AB.varX</td><td>variance of the Contrast locations of the AB genotypes (non-PAR X  only) (female samples only)</td></tr>
<tr>
<td>AB.varY</td><td>variance of the Size locations of the AB genotypes (non-PAR X  only) (female samples only)</td></tr>
<tr>
<td>BB.meanX</td><td>mean of the Contrast locations of the BB genotypes</td></tr>
<tr>
<td>BB.meanY</td><td>mean of the Size  locations of the BB genotypes</td></tr>
<tr>
<td>BB.varX</td><td>variance of the Contrast locations of the BB genotypes (non-PAR X  and Y only)</td></tr>
<tr>
<td>BB.varY</td><td>variance of the Size locations of the BB genotypes (non-PAR X  and Y only)</td></tr>
<tr>
</table>
<p>If the summary file was provided and the SSP option was used, the metrics file contains additional metrics with values calculated from the summary data:</p>
<table class="doxtable">
<tr>
<th>Column Name </th><th>Values  </th></tr>
<tr>
<td>BB_dis_x_adj</td><td>the distance between the point in the BB cluster closest to the AB cluster, restricted to a maximum height difference from the AB cluster (y_restrict)</td></tr>
<tr>
<td>AA_dis_x_adj</td><td>the distance between the point in the AA cluster closest to the AB cluster, restricted to a maximum height difference from the AB cluster (y_restrict)</td></tr>
<tr>
<td>freq_diff</td><td>difference in the frequency of assigned calls between female and male samples</td></tr>
<tr>
</table>
<p>The multiallelic output of ps-metrics is a tab delimited text file containing a single table with the following columns:</p>
<table class="doxtable">
<tr>
<th>Column Name </th><th>Values  </th></tr>
<tr>
<td>count_ma </td><td>number of alleles that appear per probeset for each allele. There is one "count_ma" variable per allele.</td></tr>
<tr>
<td>CR </td><td>Call rate</td></tr>
<tr>
<td>Nclus </td><td>number of clusters</td></tr>
<tr>
<td>nCalls </td><td>number of assigned calls</td></tr>
<tr>
<td>NC </td><td>number of NoCalls</td></tr>
<tr>
<td>nAllelesTested</td><td>number of total possible alleles for a SNP</td></tr>
<tr>
<td>nAllelesDetected</td><td>number of alleles appeared in assigned genotype calls for a SNP</td></tr>
<tr>
<td>NHetClus</td><td>number of heterozygous clusters</td></tr>
<tr>
<td>nMajorAlleles</td><td>maximum of all allele counts (count_ma metrics)</td></tr>
<tr>
<td>maxMinorAllele</td><td>second largest allele count</td></tr>
<tr>
<td>nMinorAlleles</td><td>sum of all allele counts without nMajorAlleles</td></tr>
<tr>
<td>HomCount</td><td>number of samples in all homozogyous clusters</td></tr>
<tr>
<td>MajorHomCount</td><td>number of samples in the largest homozygous cluster</td></tr>
<tr>
<td>MinorHomCount</td><td>number of samples in all homozygous clusters without MajorHomCount</td></tr>
<tr>
<td>HetCount</td><td>number of samples in all heterozygous clusters</td></tr>
<tr>
<td>nSamples</td><td>total number of samples with diploid calls</td></tr>
<tr>
<td>MAFall</td><td>minor allele frequency across all non-major alleles  (the number of minor alleles is nMinorAlleles)</td></tr>
<tr>
<td>MAFmax</td><td>minor allele frequency using the second largest allele count (the number of minor alleles is maxMinorAllele)</td></tr>
<tr>
<td>MinMean_cp_0</td><td>minimum of all copy number 0 means</td></tr>
<tr>
<td>MaxMean_cp_0</td><td>maximum of all copy number 0 means</td></tr>
<tr>
<td>MaxMeanDiff_cp_0 </td><td>maximum of all differences of Mean_cp_0 values</td></tr>
<tr>
<td>MinMean_cp_1 </td><td>minimum of all copy number 1 means</td></tr>
<tr>
<td>MaxMean_cp_1 </td><td>maximum of all copy number 1 means</td></tr>
<tr>
<td>MaxMeanDiff_cp_1 </td><td>maximum of all differences of Mean_cp_1 values</td></tr>
<tr>
<td>MinMean_cp_2 </td><td>minimum of all copy number 2 means</td></tr>
<tr>
<td>MaxMean_cp_2</td><td>maximum of all copy number 2 means</td></tr>
<tr>
<td>MaxMeanDiff_cp_2</td><td>maximum of all differences of Mean_cp_2 values</td></tr>
<tr>
<td>MinDiff_0to1</td><td>minimum difference between Mean_cp_0 and Mean_cp_1</td></tr>
<tr>
<td>MaxDiff_0to1</td><td>maximum difference between Mean_cp_0 and Mean_cp_1</td></tr>
<tr>
<td>MinDiff_1to2</td><td>minimum difference between Mean_cp_1 and Mean_cp_2</td></tr>
<tr>
<td>MaxDiff_1to2</td><td>maximum difference between Mean_cp_1 and Mean_cp_2</td></tr>
<tr>
<td>MinDiff_0to2</td><td>minimum difference between Mean_cp_0 and Mean_cp_2</td></tr>
<tr>
<td>MaxDiff_0to2</td><td>maximum difference between Mean_cp_0 and Mean_cp_2</td></tr>
<tr>
<td>MinFLD_0to1</td><td>minimum across all FLD values for copy numbers 0 and 1</td></tr>
<tr>
<td>MinFLD_1to2</td><td>minimum across all FLD values for copy numbers 1 and 2</td></tr>
<tr>
<td>MinFLD_0to2</td><td>minimum across all FLD values for copy numbers 0 and 2</td></tr>
<tr>
<td>FLD0to1_FLD0to2</td><td>ratio between MinFLD_1to2 and MinFLD_0to2</td></tr>
<tr>
<td>FLD1to2_FLD0to2</td><td>ratio between MinFLD_1to2 and MinFLD_0to2</td></tr>
<tr>
<td>FLD1to2_FLD0to1</td><td>ratio between MinFLD_1to2 and MinFLD_0to1</td></tr>
<tr>
<td>minSigBgndDiffHet</td><td>minimum of the differences between the signal and the background across all alleles for the heterozygous clusters</td></tr>
<tr>
<td>minSigBgndDiffHom</td><td>minimum of the differences between the signal and the background across all alleles for the homozygous clusters</td></tr>
<tr>
<td>AlleleDiffHom</td><td>difference between the background (copy number 0) and signal (copy number 2) for all homozygous clusters</td></tr>
<tr>
<td>AlleleDiffHet</td><td>difference between the allele mean for clusters with one copy of the allele and the non-allele means for clusters with one copy of the allele</td></tr>
<tr>
<td>maxAvgAlleleDiffHom</td><td>maximum of the averages of the AlleleDiffHom across all homozygous clusters</td></tr>
<tr>
<td>maxAvgAlleleDiffHet</td><td>maximum of the averages of the AlleleDiffHet across all heterozygous clusters.</td></tr>
<tr>
<td>specialSNP_chr</td><td>Indicates the type of chromosome (X,  Y, MT, or autosomal)</td></tr>
<tr>
</table>

<p><a class="anchor" id="options"></a> </p>
<h2>Options: </h2>
<pre class="fragment">
   'ps-metrics'
   2.10.2 stash.amer.thermo.com/ma-alg/apt2-snppolisher.git/release/APT_2.10.2@5b62fc9457(75b176a197) 2018-08-25-06:31
	
USAGE:
ps-metrics --posterior-file &lt;FILE&gt --call-file &lt;FILE&gt; --pid-file &lt;FILE&gt; --metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; [OPTIONS]
OR
ps-metrics --posterior-file &lt;FILE&gt; --call-file &lt;FILE&gt; --metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; [OPTIONS]
OR
ps-metrics --posterior-file &lt;FILE&gt; --call-file &lt;FILE&gt; --pid-file &lt;FILE&gt; --summary-file &lt;FILE&gt; --special-snps &lt;FILE&gt; --metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; [OPTIONS]
OR
ps-metrics --posterior-file &lt;FILE&gt; --call-file &lt;FILE&gt; --summary-file &lt;FILE&gt; --special-snps &lt;FILE&gt; --metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; [OPTIONS]
OR
ps-metrics --posterior-file &lt;FILE&gt; --chp-files &lt;FILE&gt; --metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; [OPTIONS]
OR
ps-metrics --posterior-file &lt;FILE&gt; --chp-files &lt;FILE&gt; --pid-file &lt;FILE&gt; --metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; [OPTIONS]
OR
ps-metrics --batch-folder &lt;FILE&gt; --metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; [OPTIONS]
OR
ps-metrics --posterior-file &lt;FILE&gt; --batch-folder &lt;FILE&gt; --metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; [OPTIONS]
OR
ps-metrics --batch-folder &lt;FILE&gt; --pid-file &lt;FILE&gt; --metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; [OPTIONS]
OR
ps-metrics --posterior-file &lt;FILE&gt; --batch-folder &lt;FILE&gt; --pid-file &lt;FILE&gt; --metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; [OPTIONS]
OR
ps-metrics --batch-folder &lt;FILE&gt; --special-snps &lt;FILE&gt; --metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; [OPTIONS]
OR
ps-metrics --posterior-file &lt;FILE&gt; --batch-folder &lt;FILE&gt; --special-snps &lt;FILE&gt; --metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; [OPTIONS]
OR
ps-metrics --batch-folder &lt;FILE&gt; --pid-file &lt;FILE&gt; --special-snps &lt;FILE&gt; --metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; [OPTIONS]
OR
ps-metrics --posterior-file &lt;FILE&gt; --batch-folder &lt;FILE&gt; --pid-file &lt;FILE&gt; --special-snps &lt;FILE&gt; --metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; --use-multi-allele <true> [OPTIONS]
OR
ps-metrics --multi-posterior-file &lt;FILE&gt; --batch-folder &lt;FILE&gt; --special-snps &lt;FILE&gt; --multi-metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; --use-multi-allele <true> [OPTIONS]
OR
ps-metrics --multi-posterior-file &lt;FILE&gt; --batch-folder &lt;FILE&gt; --multi-metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; --use-multi-allele <true> [OPTIONS]
OR
ps-metrics --multi-posterior-file &lt;FILE&gt; --posterior-file &lt;FILE&gt; --batch-folder &lt;FILE&gt; --metrics-file &lt;FILE&gt; --multi-metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; --use-multi-allele <true> [OPTIONS]
OR
ps-metrics --multi-posterior-file &lt;FILE&gt; --posterior-file &lt;FILE&gt; --special-snps &lt;FILE&gt; --batch-folder &lt;FILE&gt; --metrics-file &lt;FILE&gt; --multi-metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; --use-multi-allele <true> [OPTIONS]
OR
ps-metrics --multi-posterior-file &lt;FILE&gt; --batch-folder &lt;FILE&gt; --pid-file &lt;FILE&gt; --multi-metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; --use-multi-allele <true> [OPTIONS]
OR
ps-metrics --multi-posterior-file &lt;FILE&gt; --batch-folder &lt;FILE&gt; --pid-file &lt;FILE&gt; --special-snps &lt;FILE&gt; --multi-metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; --use-multi-allele <true> [OPTIONS]
OR
ps-metrics --multi-posterior-file &lt;FILE&gt; --posterior-file &lt;FILE&gt; --batch-folder &lt;FILE&gt; --pid-file &lt;FILE&gt; --metrics-file &lt;FILE&gt; --multi-metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; --use-multi-allele <true> [OPTIONS]
OR
ps-metrics --multi-posterior-file &lt;FILE&gt; --posterior-file &lt;FILE&gt; --batch-folder &lt;FILE&gt; --pid-file &lt;FILE&gt; --special-snps &lt;FILE&gt; --metrics-file &lt;FILE&gt; --multi-metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; --use-multi-allele <true> [OPTIONS]
OR
ps-metrics --multi-posterior-file &lt;FILE&gt; --posterior-file &lt;FILE&gt; --call-file &lt;FILE&gt; --metrics-file &lt;FILE&gt; --multi-metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; --use-multi-allele <true> [OPTIONS]
OR
ps-metrics --multi-posterior-file &lt;FILE&gt; --posterior-file &lt;FILE&gt; --call-file &lt;FILE&gt; --special-snps &lt;FILE&gt; --metrics-file &lt;FILE&gt; --multi-metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; --use-multi-allele <true> [OPTIONS]
OR
ps-metrics --multi-posterior-file &lt;FILE&gt; --posterior-file &lt;FILE&gt; --call-file &lt;FILE&gt; --pid-file &lt;FILE&gt; --metrics-file &lt;FILE&gt; --multi-metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; --use-multi-allele <true> [OPTIONS]
OR
ps-metrics --multi-posterior-file &lt;FILE&gt; --posterior-file &lt;FILE&gt; --call-file &lt;FILE&gt; --pid-file &lt;FILE&gt; --special-snps &lt;FILE&gt; --metrics-file &lt;FILE&gt; --multi-metrics-file &lt;FILE&gt; --output-dir &lt;DIR&gt; --use-multi-allele <true> [OPTIONS]

USER HELP ----------

  --version (flag) = Display the version info for this program.

  --user-help (flag) = Display help intended for end-users.
    Aliases: --help, -h

  --log-file (string8) = The name of the log file.  Defaults to 'PROGRAMNAME-NUMBER.log'.
    Aliases: --lf

  --console-add-select (string8) = Add selectors for console messages.
    e.g. to include all debug messages: --console-add-select debug
         to include all messages of level 3 or higher: --console-add-select '*:3'


  --console-add-neg-select (string8) = Add selectors to be excluded from console messages.
    e.g. to exclude all warning messages and errors summary: --console-add-neg-select WARNING,summary


  --arg-file (filename) = Read arguments from this file.
    File should be a valid XML file, and should contain elements of the form
    <Parameter name="PARAMETER-NAME" analysis="NODE-NAME" currentValue="VALUE" />.

  --check-param (flag) = Stop the program after the command parameters have been checked.

  --posterior-file (filename) = Full path name of posterior file. File must exist and file is required.
    default value: 'AxiomGT1.snp-posteriors.txt'

  --multi-posterior-file (filename) = Full path name of multi-posterior file. File must exist and file is required with the parameter use-multi-allele.
    default value: 'AxiomGT1.snp-posteriors.multi.txt'

  --call-file (filename) = Full path name of calls file. If do not used chp-files parameter, file must exist and file is required.
    default value: 'AxiomGT1.calls.txt'

  --chp-files (filename) = CHP files list file full path name. If do not used call-file parameter, file must exist and file is required. File should contain paths to CHP files, one per line.

  --batch-folder (filename) = Path to batch folder. If do not used txt version parameter is required.
    default value: 'batch.folder'

  --batch-folder-data-dir (filename) = Name of analysis data directory in batch folder.
    default value: 'AxiomAnalysisSuiteData'

  --pid-file (filename) = Full path name of pid file.
    default value: 'OffTargetVariant.ps'

  --metrics-file (filename) = Name of the output metrics file.
    default value: 'metrics.txt'

  --batch-folder-posterior-file (filename) = Name of posterior file in batch-folder.
    default value: 'AxiomGT1.snp-posteriors.txt'

  --batch-folder-blob-file (filename) = Part of blob file name.
    default value: 'AxiomAnalysisSuiteData/All_genotypes_by_snps.CHP'

  --batch-folder-multi-posterior-file (filename) = Name of multi-posterior file in batch-folder.
    default value: 'AxiomGT1.snp-posteriors.multi.txt'

  --output-dir (dirname) = Path to output directory
    default value: 'output'

  --do-pHW (bool) = Calculate Hardy-Weinberg Probability and add new column to metrics file. default: true, else: false.
    default value: 'True'

  --precision (int32) = The precision of the calculation results saved into metrics file. default: 4.
    default value: '4'

  --summary-file (filename) = Full path name of summary file. In CN version if do not used batch folder version file must exist and file is required for CN0 mean calculation.
    default value: 'AxiomGT1.summary.txt'

  --special-snps (filename) = Full path name of specialSnp file. In CN version file must exist and file is required for hemizygous define.
    default value: 'AxiomGT1.specialSNPs'
    Aliases: --specialSnp-file

  --clustermin (float64) = clustermin
    default value: '5.000000'

  --y-restrict (float64) = y-restrict
    default value: '0.200000'

  --use-supplemental (bool) = Calculate supplemental metrics and add it to metrics file. default: false, else: true. For supplemental metrics calculation user need to set summary file or batch-folder.
    default value: 'False'

  --use-ssp (bool) = Calculate SSP metrics and add it to metrics file. default: false, else: true. For SSP metrics calculation user need to set summary file or batch folder, report file (--report-file <FILE>) and special-snps.
    default value: 'False'

  --report-file (filename) = Full path name of report file. File is required for SSP metrics calculation.
    default value: 'AxiomGT1.report.txt'

  --use-multi-allele (bool) = Calculate metrics for multi-allele probesets. default: false, else: true.
    default value: 'False'

  --multi-metrics-file (filename) = Name of the output metrics file.
    default value: 'multi-metrics.txt'

  --genotype-freq-file (filename) = Full path name of genotype frequency file. File is required for clusters p-value calculation.
    default value: 'genotype_freq.txt'

  --min-genotype-freq-samples (int32) = Minimal count of samples to support clusters p-value calculation based on genotype frequency.
    default value: '20'

  --ces-k (int32) = CES transformation coefficient. Possible values are: 1, 2, 3, 4, 5.
    default value: '2'

  --use-eureka (bool) = Support eureka summary data. default: false, else: true.
    default value: 'False'

  --@help       = print developer help

</pre> </div></div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated for Analysis Power Tools by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
