<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Analysis Power Tools: VIGNETTES: Using Post-Genotyping Functionality in APT</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>VIGNETTES: Using Post-Genotyping Functionality in APT </h1>  </div>
</div>
<div class="contents">
<dl class="date"><dt><b>Date:</b></dt><dd>2018-06-22</dd></dl>
<h2><a class="anchor" id="content"></a>
Contents</h2>
<ul>
<li><a href="#intro">Introduction</a></li>
<li><a href="#psmetrics">Ps Metrics</a></li>
<li><a href="#psclassification">Ps Classification</a></li>
<li><a href="#otvcaller">OTV Caller</a></li>
</ul>
<h2><a class="anchor" id="intro"></a>
Introduction</h2>
<p>APT contains three functions for performing QC analysis on the results of genotyping: ps-metrics, ps-classification, and otv-caller.</p>

<h2><a class="anchor" id="psmetrics"></a>
Ps Metrics</h2>
<p>ps-metrics is run using a posteriors file and either a CHP file or on TXT a call file produced from an Axiom genotyping run. The following commands will create the metrics file.</p>

<p>ps-metrics --posterior-file 'file' --call-file 'file' --metrics-file 'file output location'</p>
<p>Or</p>
<p> ps-metrics --posterior-file 'file' --chp-files 'file' --metrics-file 'file output location' </p>
<p>The following command will run ps-metrics with biallelic and multiallelic posteriors files:</p>
<p> ps-metrics --multi-posterior-file 'file' --posterior-file 'file' --call-file 'file' --metrics-file 'file' --multi-metrics-file 'file' --output-dir 'file output location' --use-multi-allele 'true'</p>

<h2><a class="anchor" id="psclassification"></a>
PS Classification</h2>
<p>ps-classification is run using the output from ps-metrics. The user must specify the ploidy of the samples. To run ps-classification use the following command:</p>
<p>ps-classification --species-type [human|polyploid|diploid] --metrics-file 'file' --output-dir 'dir' --ps2snp-file 'file'</p>
<p>To change any of the default values of the biallelic thresholds, the following optional parameters can be used. Please refer to the Axiom Data Analysis Guide for more information on these metrics.</p>
<p>cr-cutoff = Threshold for call rate. Default is 95.0 for Human, 97.0 for Diploid or Polyploid.</p>
<p>fld-cutoff = Threshold for FLD, default value: 3.60.</p>
<p>het-so-cutoff = Threshold for HetSO, default value: -0.10</p>
<p>het-so-otv-cutoff = Threshold for HetSO OTV, default value: -0.30</p>
<p>hom-ro-1-cutoff = Threshold for HomRO when SNP has 1 genotype, default value: 0.60</p>
<p>hom-ro-2-cutoff = Threshold for HomRO when SNP has 2 genotypes, default value: 0.30</p>
<p>hom-ro-3-cutoff = Threshold for HomRO when SNP has 3 genotypes, default value: -0.90</p>
<p>hom-ro = True if HomRO metric to be used in classification</p>
<p>hom-het = True if Hom Het metric to be used in classification</p>
<p>num-minor-allele-cutoff = Threshold for number of minor alleles, default value: 2.0</p>
<p>priority-order = Priority order of probeset conversion types when performing probeset selection, default value: PolyHighResolution, NoMinorHom, OTV, MonoHighResolution, CallRateBelowThreshold'</p>
<p>ps-classification contains additional filters that may be required for polyploidy organisms, those with highly repetitive or complex genomes, datasets with special SNPs (non-PAR X, Y, MT), or for studying inbred populations. To access the additional filters, make sure that the input file from ps-metrics contains the additional special SNPs, supplemental, or SSP metrics.</p>

<h2><a class="anchor" id="otvcaller"></a>
OTV Caller</h2>

<p>otv-caller identifies Off Target Variants in genotyping data and updates the genotyping call to OTV for any samples that are placed in the OTV cluster. otv-caller can be run by specifying text files generated by ps-classification and ps-metrics, or by specifying an output directory generated by the Axiom Analysis Suite. Note that otv-caller can take any input list of probesets (pid file), and that the otv.ps file generated by ps-classification contains a list of probesets with clusters that may be OTV and which should be used with otv-caller.</p>

<p>otv-caller --pid-file 'file' --summary-file 'file' --posterior-file 'file' --call-file 'file' --confidence-file 'file' --output-dir 'dir' </p>

<p>OR</p>

<p>otv-caller --pid-file 'file' --batch-folder 'dir' --output-dir 'dir'</p>

<p>OR</p>

<p>otv-caller --pid-file 'file' --posterior-file 'file' --batch-folder 'dir' --output-dir 'dir'</p>
<p><b></p>
<pre>Analysis Power Tools (APT) Release 2.10.0</pre><p></b> </p>
</div>
<hr class="footer"/><address class="footer"><small>Generated on Sun Jun 24 2018 13:29:04 for Analysis Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
